---
title: "pedcancer_blood_aml"
author: "dz"
date: "2/24/2021"
output:
  word_document: default
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r load packages}

library(tidyverse)
library(stringr)
library(reshape2)
library(xlsx)
library(hexbin)
library("readxl")
library(factoextra)
library(ggpubr)
library(ggrepel)
library(limma)
library(pheatmap)
library(XML)

if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("org.hs.eg.db")

library("org.Hs.eg.db")

```


```{r read data}

achilles_avana <- read.csv("~/Documents/R/ddh/data/achilles_gene_effect.csv")

cell_info <- read.csv("~/Documents/R/ddh/data/ccle_cell_info.csv")
achilles_cell_info <- merge(achilles_avana, cell_info, by.x = "DepMap_ID", all.x = TRUE) 
expression <- read.csv("~/Documents/R/ddh/data/ccle_expression.csv")
expression_cell_info <- merge(expression, cell_info, by.x = "X", by.y = "DepMap_ID")
mirna <- read.csv("~/Documents/R/ddh/data/miRNA_expression.csv")
mirna_info <- merge(mirna, cell_info, by.x = ,1, by.y = "DepMap_ID")
metabolites <- read.csv("~/Documents/R/ddh/data/ccle_metabolomics.csv")


expression_long <- expression %>%
  pivot_longer(cols = 2:19178,
               values_to = "expression", 
               names_to = "gene_id")
achilles_info_long <- achilles_cell_info %>%
  pivot_longer(cols = 2:18018,
               values_to = "depscore",
               names_to = "gene_id")
expression_v_depscore <- left_join(achilles_info_long, expression_long, by = c("gene_id" = "gene_id", "DepMap_ID" = "X"))

expression_v_depscore_2 <- expression_v_depscore %>% mutate(gene_id = str_replace_all(gene_id, "\\..*", ""))

# TARGET AML data

# Discovery cohort -> clinical info
target_discovery_clinical <- read.xlsx("~/Documents/R/pedcancer/target_data/TARGET_AML_ClinicalData_Discovery_20211201.xlsx", 1)  # read first sheet

target_aml1031_clinical <- read.xlsx("~/Documents/R/pedcancer/target_data/TARGET_AML_ClinicalData_AML1031_20211201.xlsx", 1)  # read first sheet

# target sample matrix
target_discovery_sample_matrix <- read.xlsx("~/Documents/R/pedcancer/target_data/TARGET_AML_SampleMatrix_Discovery_20190807_v2.xlsx", 1)  # read first sheet

# methylation data

# need to find tss probes from illumina website for Infinium MethylationEPIC v1.0 B5 Manifest File (CSV Format)
# https://support.illumina.com/array/array_kits/infinium-methylationepic-beadchip-kit/downloads.html

illumina_manifest <- read.csv("~/Documents/R/pedcancer/target_data/infinium-methylationepic-v-1-0-b5-manifest-file.csv", header = TRUE)

# pull out all Probe.IDs labeled with 'tss'
# some of the results were also labeled with 'body'

illumina_tss <- pull(illumina_manifest %>%
  filter(grepl('TSS', UCSC_RefGene_Group)) %>%
  dplyr::select(Name))

# number of total cp probes that contain tss
illumina_tss <- illumina_manifest %>%
  filter(grepl('TSS', UCSC_RefGene_Group)) %>%
  dplyr::select(Name)

# number of total cp probes that contain tss but NOT Body, Exon, 5'UTR, 3'UTR...
illumina_tss_2 <- pull(illumina_manifest %>%
  filter(grepl('TSS', UCSC_RefGene_Group) &
           !grepl('Body', UCSC_RefGene_Group) &
           !grepl('Exon', UCSC_RefGene_Group) &
           !grepl('3.UTR', UCSC_RefGene_Group) &
           !grepl('5.UTR', UCSC_RefGene_Group)) %>%
  dplyr::select(Name))





```


```{r blood goals}

# what is the representation of aml in the depmap database? 
#   DONE

# what are the characteristics of these cancers in terms of age, aml subtype, cytogenetics, etc?
#   DONE

### are there broad patterns in the data that cluster aml by age group vs dependency scores? 
### what would be the best test to determine this? K-nearest neighbors?

### what are the top dependent genes in pediatric leukemia ? childhood vs aya vs adult ?
#   DONE

#### do the top dependent genes in leukemia differ by age group ? 
#   DONE

##### what are the statistically significant gene dependency differences by age ? 
#   


### extra: what are the top dependent mitochondrial proteins in pediatric leukemia ?
#can depmap data be compared with other datasets to find dependencies in aml based on patterns in other datasets, such as expression ? -> tcga, beatAML, etc

```



```{r cancer cell and lineage distribution by age}

# filter achilles_cell_info by age
# plot histogram of x=lineage, y=counts


info_cancer <- achilles_cell_info %>%
  select(stripped_cell_line_name, age, lineage, lineage_subtype, lineage_sub_subtype)

achilles_cell_info_age <- achilles_cell_info %>%
  mutate(child = ifelse(age <=20, "yes", "no")) %>%
  mutate(aya = ifelse(age >20 & age < 40, "yes", "no")) %>%
  mutate(adult = ifelse(age >= 40, "yes", "no"))  # beginning to add columns to simplify subsequent age-related data analysis.



# ped cancers

child_cancer <- info_cancer %>%
  filter(age < 20) %>%
  arrange(lineage) %>%       #### Need to reorder barplot for age <20
  mutate(lineage = factor(lineage, levels = c("peripheral_nervous_system", "blood", "soft_tissue", "bone", "lymphocyte", "central_nervous_system", "kidney", "eye", "liver", "ovary", "skin", "uterus"))) %>%
  ggplot(child_cancer, 
         mapping = aes(x=lineage, fill = lineage)) + 
         geom_bar()


print(child_cancer + 
        labs(title = "Childhood (age 0-19 yr)", 
             x = "Lineage", 
             y = "No. cell lines") + 
        scale_x_discrete(labels = c("PNS", "Blood", "Soft tissue", "Bone", "Lymphocyte", "CNS", "Kidney", "Eye", "Liver", "Ovary", "Skin", "Uterus")) +       #### Need to reorder barplot for age <20
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle =45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
              legend.position = 'none'))


# aya cancers

aya_cancer <- info_cancer %>%
  filter(age >20 & age < 40) %>%
  arrange(lineage) %>%
  mutate(lineage = factor(lineage, levels = c("blood", "lung", "skin", "central_nervous_system", "lymphocyte", "bone", "gastric", "soft_tissue", "breast", "colorectal", "liver", "upper_aerodigestive", "embryo", "ovary", "pancreas", "plasma_cell", "uterus"))) %>%
  ggplot(aya_cancer,
       mapping= aes(x=lineage, fill = lineage)) +
         geom_bar()


print(aya_cancer + 
        labs(title = "Adolescent and young adult (age 21-39 yr)", 
             x = "Lineage", 
             y = "No. cell lines") + 
        scale_x_discrete(labels = c("Blood", "Lung", "Skin", "CNS", "Lymphocyte", "Bone", "Gastric", "Soft tissue", "Breast", "Colorectal", "Liver", "Upper GI", "Embryo", "Ovary", "Pancreas", "Plasma cell", "Uterus")) + 
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle =45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
              legend.position = 'none'))


# child + aya

child_aya_cancer <- info_cancer %>%
  filter(age < 40) %>%
  arrange(lineage) %>%
  mutate(lineage = factor(lineage, levels = c("blood", "peripheral_nervous_system", "bone", "lymphocyte", "skin", "central_nervous_system", "soft_tissue", "lung", "gastric", "breast", "colorectal", "ovary", "upper_aerodigestive", "liver", "eye", "kidney", "uterus", "embryo", "pancreas", "plasma_cell"))) %>%
  ggplot(child__aya_cancer, 
         mapping = aes(x=fct_rev(lineage), fill = lineage)) + 
         geom_bar() +
  coord_flip()

print(child_aya_cancer + 
        labs(title = "Cancer Cell Lineage Disribution (age 0-39 yr)", 
             x = "Lineage", 
             y = "No. cell lines") + 
        scale_x_discrete(labels = c("Plasma Cel", "Pancreas", "Embryo", "Uterus", "Kidney", "Eye", "Liver", "Upper GI", "Ovary", "Colorectal", "Breast", "Gastric", "Lung", "Soft Tissue", "CNS", "Skin", "Lymphocyte", "Bone", "PNS", "Blood")) + 
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 0, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
              legend.position = 'none'))


# adult cancers

adult_cancer <- info_cancer %>%
  filter(age >= 40) %>%
  mutate(lineage = factor(lineage, levels = c("lung", "breast", "central_nervous_system", "pancreas", "colorectal", "skin", "upper_aerodigestive", "ovary", "liver", "esophagus", "gastric", "uterus", "plasma_cell", "blood", "urinary_tract", "cervix", "kidney", "bile_duct", "thyroid", "soft_tissue", "bone", "eye", "fibroblast", "prostate"))) %>%
  ggplot(adult_cancer,
       mapping= aes(x=lineage, fill = lineage)) +
         geom_bar()

print(adult_cancer + 
        labs(title = "Adult (age >40 yr)", 
             x = "Lineage", 
             y = "No. cell lines") + 
        scale_x_discrete(labels = c("Lung", "Breast", "CNS", "Pancreas", "Colorectal", "Skin", "Upper GI", "Ovary", "Liver", "Esophagus", "Gastric", "Uterus", "Plasma cell", "Blood", "Urinary tract", "Cervix", "Kidney", "Bile duct", "Thyroid", "Soft tissue", "Bone", "Eye", "Fibroblast", "Prostate")) + 
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
              legend.position = 'none'))

all_cancer <- achilles_cell_info %>%
  mutate(lineage = factor(lineage, levels = c("lung", "breast", "central_nervous_system", "pancreas", "colorectal", "skin", "upper_aerodigestive", "ovary", "liver", "esophagus", "gastric", "uterus", "plasma_cell", "blood", "urinary_tract", "cervix", "kidney", "bile_duct", "thyroid", "soft_tissue", "bone", "eye", "fibroblast", "prostate"))) %>%
  ggplot(,
       mapping= aes(x=lineage, fill = lineage)) +
         geom_bar()

achilles_cell_info %>%
  group_by(lineage) %>%
  count() %>%
  arrange(desc(n))

all_cells <- achilles_cell_info %>%
    mutate(lineage = factor(lineage, levels = c("lung", "central_nervous_system", "skin", "upper_aerodigestive", "blood", "ovary", "colorectal", "pancreas", "breast", "soft_tissue", "bone", "bile_duct", "lymphocyte", "peripheral_nervous_system", "gastric", "uterus", "esophagus", "kidney", "urinary_tract", "liver", "plasma_cell", "cervix", "thyroid", "eye", "prostate", "engineered", "embryo", "engineered_kidney", "epidermoid_carcinoma", "fibroblast"))) %>%
  ggplot(mapping = aes(x = lineage, fill = lineage)) + 
  geom_bar()

print(all_cells + 
        labs(title = "Cancer Representation in the DepMap Database", 
             x = "", 
             y = "No. cell lines") + 
             scale_x_discrete(labels = c("Lung", "CNS", "Skin", "Upper GI", "Blood", "Ovary", "Colorectal", "Pancreas", "Breast", "Soft Tissue", "Bone", "Bile Duct", "Lymphoma", "PNS", "Liver", "Plasma Cell", "Cervix", "Thyroid", "Eye", "Prostate", "Engineered", "Embryo", "Engineered Kidney", "Epidermoid", "Fibroblast")) +
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
              legend.position = 'none'))

print(all_cancer + 
        labs(title = "", 
             x = "", 
             y = "No. cell lines") + 
        scale_x_discrete(labels = c("Lung", "Breast", "CNS", "Pancreas", "Colorectal", "Skin", "Upper GI", "Ovary", "Liver", "Esophagus", "Gastric", "Uterus", "Plasma cell", "Blood", "Urinary tract", "Cervix", "Kidney", "Bile duct", "Thyroid", "Soft tissue", "Bone", "Eye", "Fibroblast", "Prostate")) + 
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
              legend.position = 'none'))

```



```{r blood cancers}
## goal is to make pie charts for blood cancers from each age groups

# ped
child_blood <- info_cancer %>%
  filter(age < 20, 
         lineage == "blood")


child_blood_dist <- data.frame(
  group = c("AML", "B-ALL", "T-ALL", "CML"), 
  value = c(5, 5, 3, 1))

child_blood_pie <- child_blood_dist %>%
  mutate(prop = child_blood_dist$value / sum(value)) %>%
  arrange(desc(group)) %>%
  mutate(ypos = cumsum(prop) - 0.5*prop)

ggplot(child_blood_pie, 
  mapping = aes(x = "", y = prop, fill = group)) +
  geom_bar(width = 1, stat = "identity", color = "white") +
  coord_polar("y", start=0) +
  theme_void() +
  theme(legend.position = "none") +
  geom_text(aes(y = ypos, 
                label = group), 
                color = "white", 
                size = 12) +
  scale_fill_brewer(palette = "Set1")


# aya
aya_blood <- info_cancer %>%
  filter(age >= 15 & age <40, 
         lineage == "blood")

aya_blood_dist <- data.frame(
  group = c("AML", "B-ALL", "CML"), 
  value = c(9, 2, 4))
  
aya_blood_pie <- aya_blood_dist %>%
  mutate(prop = aya_blood_dist$value / sum(value)) %>%
  arrange(desc(group)) %>%
  mutate(ypos = cumsum(prop) - 0.5*prop)

ggplot(aya_blood_pie, 
       mapping = aes(x = "", y = prop, fill = group)) +
  geom_bar(width = 1, stat = "identity", color = "white") +
  coord_polar("y", start=0) +
  theme_void() +
  theme(legend.position = "none") +
  geom_text(aes(y = ypos, 
                label = group), 
                color = "white", 
                size = 12) +
  scale_fill_brewer(palette = "Set1")


#child + aya

info_cancer %>%
  filter(age <40, 
         lineage == "blood") %>%
  arrange(lineage_subtype) %>%
  arrange(lineage_sub_subtype)

child_aya_blood_dist <- data.frame(
  group = c("B-ALL", "T-ALL", "AML", "CML"), 
  value = c(9, 4, 15, 5))
  
child_aya_blood_pie <- child_aya_blood_dist %>%
  mutate(prop = child_aya_blood_dist$value / sum(value)) %>%
  arrange(desc(group)) %>%
  mutate(ypos = cumsum(prop) - 0.5*prop)

ggplot(child_aya_blood_pie, 
       mapping = aes(x = "", y = prop, fill = group)) +
  geom_bar(width = 1, stat = "identity", color = "white") +
  coord_polar("y", start=0) +
  theme_void() +
  theme(legend.position = "none") +
  geom_text(aes(y = ypos, 
                label = group), 
                color = "white", 
                size = 12) +
  scale_fill_brewer(palette = "Set1")


# adult
adult_blood <- info_cancer %>%
  filter(age >= 40,
         lineage == "blood")

adult_blood_dist <- data.frame(
  group = c("AML", "CML", "CLL", "Unspecified"), 
  value = c(7, 2, 1, 3))

adult_blood_pie <- adult_blood_dist %>%
  mutate(prop = adult_blood_dist$value / sum(value)) %>%
  arrange(desc(group)) %>%
  mutate(ypos = cumsum(prop) - 0.5*prop)

ggplot(adult_blood_pie, 
       mapping = aes(x = "", y = prop, fill = group)) +
  geom_bar(width = 1, stat = "identity", color = "white") +
  coord_polar("y", start=0) + 
  theme_void() +
  theme(legend.position = "none") +
  geom_text(aes(y = ypos, 
                label = group), 
                color = "white", 
                size = 12) +
  scale_fill_brewer(palette = "Set1")



# all blood cells

achilles_cell_info %>%
  filter(lineage == "blood") %>%
  group_by(lineage_subtype) %>%
  summarise(n())
  select(stripped_cell_line_name, age, lineage, lineage_subtype)

all_blood <- achilles_cell_info %>%
  filter(lineage == "blood")

all_blood_dist <- data.frame(
  group = c("AML", "ALL", "CML", "CLL"), 
  value = c(26, 17, 8, 4))

all_blood_pie <- all_blood_dist %>%
  mutate(prop = all_blood_dist$value / sum(value)) %>%
  arrange(desc(group)) %>%
  mutate(ypos = cumsum(prop) - 0.5*prop)

ggplot(all_blood_pie, 
       mapping = aes(x = "", y = prop, fill = group)) +
  geom_bar(width = 1, stat = "identity", color = "white") +
  coord_polar("y", start=0) + 
  theme_void() +
  theme(legend.position = "none") +
  geom_text(aes(y = ypos, 
                label = group), 
                color = "white", 
                size = 12) +
  scale_fill_brewer(palette = "Set1")

# ages
achilles_cell_info %>%
  filter(lineage_subtype == "AML") %>%
  select(lineage_subtype, age)

aml_age <- achilles_cell_info %>%
  filter(lineage_subtype == "AML")

aml_age_dist <- data.frame(
  group = c("Child", "AYA", "Adult"), 
  value = c(5, 9, 8))

aml_age_pie <- aml_age_dist %>%
  mutate(prop = aml_age_dist$value / sum(value)) %>%
  arrange(desc(group)) %>%
  mutate(ypos = cumsum(prop) - 0.5*prop)

ggplot(aml_age_pie, 
       mapping = aes(x = "", y = prop, fill = group)) +
  geom_bar(width = 1, stat = "identity", color = "white") +
  coord_polar("y", start=0) + 
  theme_void() +
  theme(legend.position = "none") +
  geom_text(aes(y = ypos, 
                label = group), 
                color = "white", 
                size = 12) +
  scale_fill_brewer(palette = "Set1")


```


```{r}
# AML cell lines

# what do we know about the representation of AML in the depmap database?
# interested in sublineage (AML), sub-sublineage (M-class), cytogenetics, age of patient

cell_info %>%
  filter(lineage_subtype == "AML",
         age <= 20) %>%
  select(stripped_cell_line_name, age, sex, lineage_subtype, lineage_sub_subtype, lineage_molecular_subtype) %>%
  arrange(-desc(age)) 

cell_info %>%
  filter(lineage_subtype == "AML",
         age >20 & age < 40) %>%
  select(stripped_cell_line_name, age, sex, lineage_subtype, lineage_sub_subtype, lineage_molecular_subtype) %>%
  arrange(-desc(age)) 

cell_info %>%
  filter(lineage_subtype == "AML",
         age >= 40) %>%
  select(stripped_cell_line_name, age, sex, lineage_subtype, lineage_sub_subtype, lineage_molecular_subtype) %>%
  arrange(-desc(age)) 

```


```{r top dependencies ?}
# what are the top dependencies in ped/aya aml? Are they different vs adult ?

# ped aml mean df
child_aml <- achilles_cell_info %>%
  filter(lineage_subtype == "AML",
         age <= 20) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(child_aml_mean = mean(dep_score))

# aya aml mean df
aya_aml <- achilles_cell_info %>%
  filter(lineage_subtype == "AML",
         age > 20 | age < 40) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(aya_aml_mean = mean(dep_score))

# adult aml mean df
adult_aml <- achilles_cell_info %>%
  filter(lineage_subtype == "AML",
         age >= 40) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(adult_aml_mean = mean(dep_score))

# merge dataframes

## child vs adult df
## age merge df = all 3 groups

child_vs_adult_aml <- merge(child_aml, adult_aml, by = "gene_id") %>%
  mutate(child_adult_delta = child_aml_mean - adult_aml_mean) %>%
  arrange(-desc(child_adult_delta))

age_merge_aml <- merge(child_vs_adult_aml, aya_aml, by = "gene_id") %>%
  mutate(child_adult_delta = child_aml_mean - adult_aml_mean,
         aya_adult_delta = aya_aml_mean - adult_aml_mean,
         child_aya_delta = child_aml_mean - aya_aml_mean)

head(age_merge_aml)


age_merge_aml %>%
  filter(child_aml_mean > -1, 
         aya_aml_mean < -1, 
         adult_aml_mean > -1)

```


```{r top dependencies ?}

## top dependency score differences using age merged file

arrange(age_merge_aml,
        -desc(child_adult_delta)) %>%
  head(300)

filter(age_merge_aml,
       child_adult_delta < -0.33 | child_adult_delta > 0.33) %>%
  arrange(-desc(child_aml_mean))

top_ped_aml_v2 <- child_vs_adult_aml %>%
  filter(child_adult_delta < -0.33) %>%
  arrange(-desc(child_aml_mean)) %>%
  arrange(-desc(child_adult_delta))
top_ped_aml_v2 <- top_ped_aml_v2$gene_id
top_ped_aml_v2split <- as.data.frame(top_ped_aml_v2 %>%
  str_split("\\..", n=2, simplify = TRUE))
top_ped_aml_v2_gene_id <- top_ped_aml_v2split$V1
write.csv(top_ped_aml_v2_gene_id, "~/Documents/R/pedcancer/top_ped_aml_v2_gene_id.csv", row.names = FALSE, col.names = FALSE)



## top differences among essential childhood AML genes and not adult
## this generated a df with gene id used in GO analyses

child_vs_adult_aml %>%
  filter(child_aml_mean < -1 & adult_aml_mean > -1) %>%
  arrange(-desc(child_adult_delta)) 

top_ped_aml <- child_vs_adult_aml %>%
  filter(child_aml_mean < -1 & adult_aml_mean > -1) %>%
  arrange(-desc(child_adult_delta))
top_ped_aml <- top_ped_aml$gene_id
top_ped_aml_split <- as.data.frame(top_ped_aml %>%
  str_split("\\..", n=2, simplify = TRUE))
top_ped_aml_gene_id <- top_ped_aml_split$V1
write.csv(top_ped_aml_gene_id, "~/Documents/R/pedcancer/top_ped_aml_gene_id.csv", row.names = FALSE, col.names = FALSE)



# distribution of dependency differences

ggplot(age_merge_aml,
       mapping = aes(x = child_adult_delta)) +
  geom_histogram(binwidth = 0.01, color = "black", fill = "white")


# correlation of dependencies score means

ggplot(age_merge_aml,
       mapping = aes(x = child_aml_mean, y = adult_aml_mean)) +
  geom_point(alpha = 0.2) +
  geom_smooth(color="red", fill="#69b3a2", se=TRUE)


```


```{r top dependencies vs other ped cancers}

# pull out essential genes vs other pediatric cancers

# by lineage

ped_lineage <- achilles_cell_info %>%
    filter(age <= 20) %>%
  select(lineage, lineage_subtype, A1BG..1.:ZZZ3..26009.) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(lineage) %>%
  group_by(gene_id, .add = TRUE) %>%
  summarise(mean = mean(dep_score)) %>%
  pivot_wider(names_from = lineage, 
              values_from = mean) %>%
  filter(blood <= -1 & 
           bone > -1 & 
           central_nervous_system > -1 &
           eye > -1 &
           kidney > -1 &
           liver > -1 &
           ovary > -1 &
           peripheral_nervous_system > -1 &
           skin > -1 &
           soft_tissue > -1 &
           uterus > -1) %>%
  arrange(-desc(blood))

top_ped_lineage <- ped_lineage$gene_id
top_ped_lineage <- as.data.frame(top_ped_lineage %>%
  str_split("\\..", n=2, simplify = TRUE))
top_ped_lineage_gene_id <- top_ped_lineage$V1
write.csv(top_ped_lineage_gene_id, "~/Documents/R/pedcancer/top_ped_lineage_gene_id.csv", row.names = FALSE, col.names = FALSE)


# by blood lineage_subtype

pedblood_sublineage <- achilles_cell_info %>%
    filter(age <= 20, 
           lineage == "blood") %>%
  select(lineage_subtype, A1BG..1.:ZZZ3..26009.) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(lineage_subtype) %>%
  group_by(gene_id, .add = TRUE) %>%
  summarise(mean = mean(dep_score)) %>%
  pivot_wider(names_from = lineage_subtype, 
              values_from = mean) %>%
  filter(AML < -1 &
         ALL > -1 &
         CML > -1) %>%
  arrange(-desc(AML))

top_pedblood_sublineage <- pedblood_sublineage$gene_id
top_pedblood_sublineage <- as.data.frame(top_pedblood_sublineage %>%
  str_split("\\..", n=2, simplify = TRUE))
top_pedblood_sublineage_gene_id <- top_pedblood_sublineage$V1
write.csv(top_pedblood_sublineage_gene_id, "~/Documents/R/pedcancer/top_pedblood_sublineage_gene_id.csv", row.names = FALSE, col.names = FALSE)


```

```{r}

#volcano plot for AML vs ped solid tumors

aml_cells <- pull(achilles_cell_info %>%
                    filter(lineage_subtype == "AML") %>%
                    select(stripped_cell_line_name))
aml_cells

solid_cells <- pull(achilles_cell_info %>%
                      filter(age < 20, 
                             lineage != "blood" &
                               lineage != "lymphocyte") %>%
                      select(stripped_cell_line_name))

solid_cells

aml_depscores <- achilles_cell_info %>%
  filter(stripped_cell_line_name %in% aml_cells) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009., 
               names_to = "gene_id",
               values_to = "depscore") %>%
  select(stripped_cell_line_name, gene_id, depscore) %>%
  pivot_wider(names_from = stripped_cell_line_name, 
              values_from = depscore)

solid_depscores <- achilles_cell_info %>%
  filter(stripped_cell_line_name %in% solid_cells) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009., 
               names_to = "gene_id",
               values_to = "depscore") %>%
    select(stripped_cell_line_name, gene_id, depscore) %>%
  pivot_wider(names_from = stripped_cell_line_name, 
              values_from = depscore)

merge_depscores <- merge(aml_depscores, solid_depscores, by = "gene_id")

#write.csv(merge_depscores, "~/Documents/R/pedcancer/merge_depscores.csv", row.names = FALSE)

volcano <- read.csv("~/Documents/R/pedcancer/merge_depscores_volcano.csv")

v <- volcano %>%
  ggplot(mapping = aes(x = delta, y = neglogP)) + 
  geom_point() +
  xlim(-1,1) +
  ylim(0,40) +
  geom_point(data = volcano %>%
               filter(delta < -0.57, neglogP >10 ),
             mapping = aes(x= delta, y= neglogP), color = "red") +
  geom_label_repel(data = volcano %>%
               filter(delta < -0.57, neglogP >10),
             aes(label = gene_id), 
             nudge_x = 0.35)


v + xlab("Essentiality Difference") +
  ylab("Significance (-logP)") +
  theme(
    axis.title.x = element_text(size = 18), 
    axis.title.y = element_text(size = 18), 
    axis.text.y = element_text(size = 16), 
    axis.text.x = element_text(size =16),
    aspect.ratio = 5/7)
  


vpre <- volcano %>%
  ggplot(mapping = aes(x = delta, y = neglogP)) + 
  geom_point() +
  xlim(-1,1) +
  ylim(0,40)

vpre + xlab("Essentiality Difference") +
  ylab("Significance (-logP)") +
  theme(
    axis.title.x = element_text(size = 18), 
    axis.title.y = element_text(size = 18), 
    axis.text.y = element_text(size = 16), 
    axis.text.x = element_text(size =16),
    aspect.ratio = 5/7)

volcano %>%
  filter(delta < -0.33, neglogP >5) %>%
  arrange(-desc(delta))
```

```{r}
#volcano plot for AML vs ped solid tumors

child_aml_cells <- pull(achilles_cell_info %>%
                    filter(age < 15, 
                           lineage_subtype == "AML") %>%
                    select(stripped_cell_line_name))
child_aml_cells

adult_aml_cells <- pull(achilles_cell_info %>%
                      filter(age >=40,
                             lineage_subtype == "AML") %>%
                      select(stripped_cell_line_name))

solid_cells

child_aml_depscores <- achilles_cell_info %>%
  filter(stripped_cell_line_name %in% child_aml_cells) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009., 
               names_to = "gene_id",
               values_to = "depscore") %>%
  select(stripped_cell_line_name, gene_id, depscore) %>%
  pivot_wider(names_from = stripped_cell_line_name, 
              values_from = depscore)

adult_aml_depscores <- achilles_cell_info %>%
  filter(stripped_cell_line_name %in% adult_aml_cells) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009., 
               names_to = "gene_id",
               values_to = "depscore") %>%
    select(stripped_cell_line_name, gene_id, depscore) %>%
  pivot_wider(names_from = stripped_cell_line_name, 
              values_from = depscore)

merge_aml_depscores <- merge(child_aml_depscores, adult_aml_depscores, by = "gene_id")

write.csv(merge_aml_depscores, "~/Documents/R/pedcancer/merge_aml_depscores.csv", row.names = FALSE)

volcano_merged <- read.csv("~/Documents/R/pedcancer/merge_aml_depscores_volcano.csv")

v_merged <- volcano_merged %>%
  ggplot(mapping = aes(x = delta, y = neglogP)) + 
  geom_point() +
  xlim(-1,1) +
  ylim(0,5) +
  geom_point(data = volcano_merged %>%
               filter(delta < -0.4 | delta > 0.4, neglogP >2 ),
             mapping = aes(x= delta, y= neglogP), color = "red") +
  geom_label_repel(data = volcano_merged %>%
               filter(delta < -0.4 | delta > 0.4, neglogP >2),
             aes(label = gene_id), 
             nudge_x = 0.35)


v_merged + xlab("Essentiality Difference") +
  ylab("Significance (-logP)") +
  theme(
    axis.title.x = element_text(size = 18), 
    axis.title.y = element_text(size = 18), 
    axis.text.y = element_text(size = 16), 
    axis.text.x = element_text(size =16),
    aspect.ratio = 5/7)
  


vpre <- volcano %>%
  ggplot(mapping = aes(x = delta, y = neglogP)) + 
  geom_point() +
  xlim(-1,1) +
  ylim(0,40)

vpre + xlab("Essentiality Difference") +
  ylab("Significance (-logP)") +
  theme(
    axis.title.x = element_text(size = 18), 
    axis.title.y = element_text(size = 18), 
    axis.text.y = element_text(size = 16), 
    axis.text.x = element_text(size =16),
    aspect.ratio = 5/7)

volcano %>%
  filter(delta < -0.33, neglogP >5) %>%
  arrange(-desc(delta))

age <- achilles_cell_info %>%
  ggplot(mapping = aes(x = age)) + 
  geom_histogram(binwidth = 1, fill = "blue", color = "black")

print(age + 
        labs(title = "", 
             x = "", 
             y = "No. cell lines") + 
        theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 18, color = "black", angle = 0, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 14),
              legend.position = 'none'))

dist <- achilles_cell_info %>%
  filter(age <20) %>% 
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id", 
               values_to = "depscore") %>%
  
  ggplot(mapping = aes(x = ))

```



```{r expression and dependency score correlation}

## goal: learn more about gene expression and dependency values by correlating data from childhood AML cell lines


## make mean gene expression dataframe for pediatric AML cell lines

expression_cell_info <- merge(expression, cell_info, by = "DepMap_ID")

child_aml_expression <- expression_cell_info %>%
  filter(lineage_subtype == "AML", 
         age <= 20) %>%
  pivot_longer(cols = 2:16384,
               names_to = "gene_id",
               values_to = "expression") %>%
  select(stripped_cell_line_name, gene_id, expression) %>%
  group_by(gene_id) %>%
  summarise(child_aml_expression = mean(expression))

ggplot(child_aml_expression, 
       mapping = aes(x = child_aml_expression)) +
  geom_histogram(binwidth = 0.1)


## merge dependency score data with gene expression data

depscore_by_expression <- merge(child_aml, child_aml_expression, by = "gene_id")

ggplot(depscore_by_expression,
       mapping = aes(x = child_aml_mean)) +
  geom_histogram(binwidth = 0.05)

ggplot(depscore_by_expression,
       mapping = aes(x = child_aml_mean, y = child_aml_expression)) +
  geom_hex(bins = 100) +
  scale_fill_continuous(type = "viridis") +
  theme_bw()

depscore_by_expression %>%
  filter(child_aml_mean < -1) %>%
  arrange(-desc(child_aml_mean))


## what are the clusters of genes with middle-level and high-level expression that have essential dependency scores ?

toplevel_exp <- depscore_by_expression %>%
  filter(child_aml_mean <= -1,
         child_aml_expression >= 9)

toplevel_genes <- toplevel_exp$gene_id

toplevel_genes_split <- as.data.frame(toplevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

toplevel_gene_id <- toplevel_genes_split$V1

write.csv(toplevel_gene_id, "~/Documents/R/pedcancer/toplevel_gene_id.csv", row.names = FALSE, col.names = FALSE)

midlevel_exp <- depscore_by_expression %>%
  filter(child_aml_mean <= -1) %>%
  filter(child_aml_expression >= 2.5 & child_aml_expression < 9)

midlevel_genes <- midlevel_exp$gene_id

midlevel_genes_split <- as.data.frame(midlevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

midlevel_gene_id <- midlevel_genes_split$V1

write.csv(midlevel_gene_id, "~/Documents/R/pedcancer/midlevel_gene_id.csv", row.names = FALSE)




```

```{r}
# does gene expression correlate with dependency score in adult AML ?
# same code, just with different age filter
# do essential adult low-, mid- and -top level expression genes represent the same GO pathways as childhood AML ?


adult_aml_expression <- expression_cell_info %>%
  filter(lineage_subtype == "AML", 
         age >= 40) %>%
  pivot_longer(cols = TSPAN6..7105.:AC113348.1..102724657.,
               names_to = "gene_id",
               values_to = "expression") %>%
  select(stripped_cell_line_name, gene_id, expression) %>%
  group_by(gene_id) %>%
  summarise(adult_aml_expression = mean(expression))

ggplot(adult_aml_expression, 
       mapping = aes(x = adult_aml_expression)) +
  geom_histogram(binwidth = 0.1)

adult_depscore_by_expression <- merge(adult_aml, adult_aml_expression, by = "gene_id")

ggplot(adult_depscore_by_expression,
       mapping = aes(x = adult_aml_mean, y = adult_aml_expression)) +
  geom_hex(bins = 100) +
  scale_fill_continuous(type = "viridis") +
  theme_bw()

adult_toplevel_exp <- adult_depscore_by_expression %>%
  filter(adult_aml_mean <= -1,
         adult_aml_expression >= 9)

adult_toplevel_genes <- adult_toplevel_exp$gene_id

adult_toplevel_genes_split <- as.data.frame(adult_toplevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

adult_toplevel_gene_id <- adult_toplevel_genes_split$V1

write.csv(adult_toplevel_gene_id, "~/Documents/R/pedcancer/adult_toplevel_gene_id.csv", row.names = FALSE, col.names = FALSE)

adult_midlevel_exp <- adult_depscore_by_expression %>%
  filter(adult_aml_mean <= -1) %>%
  filter(adult_aml_expression >= 2.5 & adult_aml_expression < 9)

adult_midlevel_genes <- adult_midlevel_exp$gene_id

adult_midlevel_genes_split <- as.data.frame(adult_midlevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

adult_midlevel_gene_id <- adult_midlevel_genes_split$V1

write.csv(adult_midlevel_gene_id, "~/Documents/R/pedcancer/adult_midlevel_gene_id.csv", row.names = FALSE)


adult_lowlevel_exp <- adult_depscore_by_expression %>%
  filter(adult_aml_mean <= -1,
         adult_aml_expression < 2.5)

adult_lowlevel_genes <- adult_lowlevel_exp$gene_id

adult_lowlevel_genes_split <- as.data.frame(adult_lowlevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

adult_lowlevel_gene_id <- adult_lowlevel_genes_split$V1

write.csv(adult_lowlevel_gene_id, "~/Documents/R/pedcancer/adult_lowlevel_gene_id.csv", row.names = FALSE)




```

```{r}
# goal: PILOT GRANT
# Top ALL gene programs per analysis above


# make mean gene expression dataframe for pediatric *ALL cell lines

expression_cell_info <- merge(expression, cell_info, by.x = "X", by.y = "DepMap_ID")
child_all <- achilles_cell_info %>%
  filter(lineage_subtype == "ALL",
         age < 15) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(child_all_mean = mean(dep_score))

child_all %>%
  arrange(-desc(child_all_mean))

child_pns <- achilles_cell_info %>%
  filter(lineage == "peripheral_nervous_system",
         age < 15) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(child_pns_mean = mean(dep_score))

child_soft <- achilles_cell_info %>%
  filter(lineage == "soft_tissue",
         age < 15) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(child_soft_mean = mean(dep_score))

child_bone <- achilles_cell_info %>%
  filter(lineage == "bone",
         age < 15) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(child_bone_mean = mean(dep_score))

child_cns <- achilles_cell_info %>%
  filter(lineage == "central_nervous_system",
         age < 15) %>%
  pivot_longer(cols = A1BG..1.:ZZZ3..26009.,
               names_to = "gene_id",
               values_to = "dep_score") %>%
  group_by(gene_id) %>%
  summarise(child_cns_mean = mean(dep_score)) %>%
  arrange(-desc(child_cns_mean))

child_all_pns <- merge(child_all, child_pns, by = "gene_id")
child_all_pns_soft <- merge(child_all_pns, child_soft, by = "gene_id")
child_all_pns_soft_bone <- merge(child_all_pns_soft, child_bone, by = "gene_id")
child_merge <- merge(child_all_pns_soft_bone, child_cns, by = "gene_id")

#top ALL genes
child_merge %>%
  filter(child_all_mean < -1 & child_pns_mean > -0.75 & child_soft_mean > -0.75 & child_bone_mean > -0.75 & child_cns_mean > -0.75) %>%
  arrange(-desc(child_all_mean))

#top PNS genes
child_merge %>%
  filter(child_pns_mean < -1 & child_all_mean > -0.75 & child_soft_mean > -0.75 & child_bone_mean > -0.75 & child_cns_mean > -0.75) %>%
  arrange(-desc(child_pns_mean))

#top CNS genes
child_merge %>%
  filter(child_cns_mean < -1 & child_all_mean > -0.75 & child_soft_mean > -0.75 & child_bone_mean > -0.75 & child_pns_mean > -0.75) %>%
  arrange(-desc(child_cns_mean))

#top soft tissue genes
child_merge %>%
  filter(child_soft_mean < -1 & child_all_mean > -0.75 & child_cns_mean > -0.75 & child_bone_mean > -0.75 & child_pns_mean > -0.75) %>%
  arrange(-desc(child_soft_mean))

#top bone genes
child_merge %>%
  filter(child_bone_mean < -1 & child_all_mean > -0.75 & child_cns_mean > -0.75 & child_soft_mean > -0.75 & child_pns_mean > -0.75) %>%
  arrange(-desc(child_bone_mean))


#other

child_all_bone_cns_delta <- child_all_bone_cns %>%
  mutate(all_bone_delta = child_all_mean - child_bone_mean,
         all_cns_delta = child_all_mean - child_cns_mean) %>%
  filter(child_all_mean < -1 & child_cns_mean > -0.75 & child_bone_mean > -0.75) %>%
  arrange(-desc(child_all_mean))
view(child_all_bone_cns_delta)

head(child_all_bone_cns_delta)

child_all_expression <- expression_cell_info %>%
  filter(lineage_subtype == "ALL", 
         age < 15) %>%
  pivot_longer(cols = TSPAN6..7105.:AC113348.1..102724657.,
               names_to = "gene_id",
               values_to = "expression") %>%
  select(stripped_cell_line_name, gene_id, expression) %>%
  group_by(gene_id) %>%
  summarise(child_all_expression = mean(expression))

ggplot(child_all_expression, 
       mapping = aes(x = child_all_expression)) +
  geom_histogram(binwidth = 0.1)
```


```{r}
# merge dependency score data with gene expression data

all_depscore_by_expression <- merge(child_all, child_all_expression, by = "gene_id")

all_depscore_by_expression %>%
  filter(gene_id == "TYMS..7298." |
         gene_id == "CTPS1..1503." | 
  gene_id == "MYB..4602." | 
  gene_id == "DHODH..1723." | 
  gene_id == "CCND3..896." | 
  gene_id == "DHFR..1719.") %>%
  ggplot(mapping = aes(x = child_all_mean, y = child_all_expression)) +
  geom_point()


ggplot(all_depscore_by_expression,
       mapping = aes(x = child_all_mean)) +
  geom_histogram(binwidth = 0.05)

ggplot(all_depscore_by_expression,
       mapping = aes(x = child_all_expression)) +
  geom_histogram(binwidth = 0.1)

all_depscore_by_expression %>%
  filter(child_all_expression > 3.6 & child_all_expression < 8.9)

ggplot(all_depscore_by_expression,
       mapping = aes(x = child_all_mean, y = child_all_expression)) +
  geom_hex(bins = 100) +
  scale_fill_continuous(type = "viridis") +
  theme_bw()

all_depscore_by_expression %>%
  filter(child_all_mean < -1) %>%
  arrange(-desc(child_all_mean))

#list of top ALL genes without regard to expression level

all_top <- all_depscore_by_expression %>%
  filter(child_all_mean <= -1) %>%
  arrange(-desc(child_all_mean))

all_toplevel_genes <- all_toplevel_exp$gene_id

all_toplevel_genes_split <- as.data.frame(all_toplevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

all_toplevel_gene_id <- all_toplevel_genes_split$V1

write.csv(all_toplevel_gene_id, "~/Documents/R/pedcancer/all_toplevel_gene_id.csv", row.names = FALSE, col.names = FALSE)


# what are the clusters of genes with middle-level and high-level expression that have essential dependency scores ?

all_toplevel_exp <- all_depscore_by_expression %>%
  filter(child_all_mean <= -1,
         child_all_expression >= 9)

all_toplevel_genes <- all_toplevel_exp$gene_id

all_toplevel_genes_split <- as.data.frame(all_toplevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

all_toplevel_gene_id <- all_toplevel_genes_split$V1

write.csv(all_toplevel_gene_id, "~/Documents/R/pedcancer/all_toplevel_gene_id.csv", row.names = FALSE, col.names = FALSE)

all_midlevel_exp <- all_depscore_by_expression %>%
  filter(child_all_mean <= -1) %>%
  filter(child_all_expression >= 2.5 & child_all_expression < 9)

all_midlevel_genes <- all_midlevel_exp$gene_id

all_midlevel_genes_split <- as.data.frame(all_midlevel_genes %>%
  str_split("\\..", n=2, simplify = TRUE))

all_midlevel_gene_id <- all_midlevel_genes_split$V1

write.csv(all_midlevel_gene_id, "~/Documents/R/pedcancer/all_midlevel_gene_id.csv", row.names = FALSE)


# all top- ALL genes

top_all_genes <- all_depscore_by_expression %>%
  filter(child_all_mean <= -1) %>%
  arrange(-desc(child_all_mean))

top_all_genes__depscore_name <- top_all_genes$gene_id

top_all_genes_depscore_name_split <- as.data.frame((top_all_genes__depscore_name) %>%
  str_split("\\..", n=2, simplify = TRUE))

top_all_genes_depscore_names <- top_all_genes_depscore_name_split$V1

write.csv(top_all_genes_depscore_names, "~/Documents/R/pedcancer/top_all_genes_depscore_names.csv", row.names = FALSE, col.names = FALSE)

```



```{r mitocarta}

# goal is to merge AML dependency scores with mitocarta database to identify top mito genes

head(child_aml)

string <- child_aml$gene_id
new_string <- as.data.frame(str_remove_all(string, "\\..\\d+\\."))
head(new_string)

child_aml <- as.data.frame(child_aml %>%
  str_remove_all("\\..\\d+\\."))



child_aml <- as.data.frame(child_aml$child_aml_mean, row.names = c(new_string))  
head(child_aml)

child_aml_split <- as.data.frame(child_aml$gene_id %>%
  str_split("\\..", simplify = FALSE))

merge(child_aml_split, mitocarta, by.x = "V1", by.y = "Symbol") %>%
  left_join((child_aml$child_aml_mean))

```

```{r}

# gene expression heterogeneity in blood cancer cells

aml_exp_IQR <- expression_v_depscore %>%
  filter(lineage_subtype == "AML") %>%
  select(stripped_cell_line_name, lineage_subtype, age, gene_id, expression) %>%
  group_by(gene_id) %>%
  mutate(IQR = IQR(expression, na.rm = TRUE)) %>%
  summarise(IQR = median(IQR)) %>%
  select(gene_id, IQR) %>%
  mutate(gene_id = str_replace_all(gene_id, "\\..*", ""))

AML_IQR_plot <- aml_exp_IQR %>% ggplot(mapping = aes(x = fct_reorder(gene_id, IQR), y = IQR)) + 
  geom_point(alpha = 0.05) +
  geom_point(data = aml_exp_IQR %>% filter(IQR >= 2.5), mapping = aes(x = gene_id, y = IQR, color = "red")) +
   geom_label_repel(data = aml_exp_IQR %>% filter(IQR >= 2.5), mapping = aes(x = gene_id, y = IQR, label = gene_id), max.overlaps = 70, size = 5) +
 labs(title = "Variability of gene expression in AML cell lines", 
             x = "Ranked genes", 
             y = "IQR of Gene Expression") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_text(size = 16),
         axis.ticks.x = element_blank(),
         axis.title.y = element_text(size = 16),
         axis.text.y = element_text(size = 12),
         axis.line = element_line(size = 1, color = "black", linetype = 1),
         legend.position = "none")
            
med_exp <- expression_v_depscore %>%
  filter(lineage_subtype == "AML") %>%
  select(stripped_cell_line_name, lineage_subtype, age, gene_id, expression) %>%
  group_by(gene_id) %>%
  summarise(median = median(expression)) %>%
  mutate(gene_id = str_replace_all(gene_id, "\\..*", ""))

exp_by_IQR <- merge(aml_exp_IQR, med_exp, by = "gene_id")
ggplot(exp_by_IQR, mapping = aes(x = median, y = mIQR)) + 
  geom_point(alpha = 0.2) 

IQR_top <- pull(aml_exp_IQR %>% arrange(desc(IQR)) %>% filter(IQR >= 2.5) %>% dplyr::select(gene_id))




```




```{r}

# leveraging transcriptional heterogeneity to predict essential gene targets

ccle_cells <- expression_cell_info %>%
  mutate(overlap = ifelse(stripped_cell_line_name %in% achilles_cell_info$stripped_cell_line_name, 'yes', 'no')) %>%
  filter(overlap == 'no') %>%
  select(stripped_cell_line_name, lineage, lineage_subtype, lineage_sub_subtype, age)

ccle_cells_expression <- expression_cell_info %>%
  mutate(overlap = ifelse(stripped_cell_line_name %in% achilles_cell_info$stripped_cell_line_name, 'yes', 'no')) %>%
  filter(overlap == 'no') %>%
  select(2:19178, lineage, lineage_subtype, stripped_cell_line_name, overlap) %>%
  pivot_longer(cols = 1:19177,
               names_to = "gene_id", 
               values_to = "expression")

ccle_lineage <- ccle_cells %>%
  group_by(lineage) %>%
  summarise(length(lineage))

ccle_ped <- ccle_cells %>%
  filter(age < 25) %>%
  group_by(lineage) %>%
  summarise(length(lineage))

ccle_cells %>%
  filter(age < 39, 
         lineage == "blood")

blood_cells <- ccle_cells %>% filter(lineage == "blood") 

ccle_aml <- blood_cells %>% filter(lineage_subtype == "AML")

expression_ccle_aml <- ccle_cells_expression %>% 
  filter(stripped_cell_line_name %in% ccle_aml$stripped_cell_line_name) %>% 
  select(stripped_cell_line_name, gene_id, expression) %>%
  mutate(gene_id = str_replace_all(gene_id, "\\..*", ""))

expression_ccle_aml %>%
  pivot_wider(names_from = stripped_cell_line_name, 
              values_from = expression) %>%
  unnest() %>%
  mutate(med = median(cols = 2:19))

expression_ccle_aml %>%
  group_by(gene_id) %>%
  summarise(med = median(expression), 
            sd = sd(expression),
            quant = quantile(expression), 
            min = min(expression), 
            max = max(expression)) %>%
  filter(med != 0, 
         min != 0) %>%
  mutate(range = max-min) %>%
  filter(range != 0) %>%
  arrange(desc(range)) %>%
            head(100) %>%
            ggplot(mapping = aes(x = gene_id, y = range)) +
            geom_col() +
    xlab("") +
    ylab("Expression Range") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 18),
          axis.text.x = element_text(size = 12, angle = 30, hjust = 1, vjust = 1, color = "black"),
          axis.text.y = element_text(size = 14))  
    
  

expression_ccle_aml %>%
  filter(gene_id == "CTSG") %>%
  ggplot(mapping = aes(x = expression)) + geom_his


```

```{r}
# read TARGET-AML gene expression data

pt1 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PABGKN-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PABGKN = RPKM)

pt2 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PABLDZ-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PABLDZ = RPKM)

pt3 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PACDZR-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PACDZR = RPKM)

pt4 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PADYIR-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PADYIR = RPKM)

pt5 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PADZCG-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PADZCG = RPKM)

pt6 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PADZKD-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PADZKD = RPKM)

pt7 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEAFC-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEAFC = RPKM)

pt8 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEAKL-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEAKL = RPKM)

pt9 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAECBZ-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAECBZ = RPKM)

pt10 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAECCE-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAECCE = RPKM)

pt11 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEENN-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEENN = RPKM)

pt12 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEERJ-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEERJ = RPKM)

pt13 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEEYP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEEYP = RPKM)

pt12 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEFGR-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEFGR = RPKM)

pt13 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEFHC-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEFHC = RPKM)

pt14 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEIKD-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEIKD = RPKM)

pt15 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEJBT-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAEJBT = RPKM)

pt16 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAERAH-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAERAH = RPKM)

pt17 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAESVN-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAESVN = RPKM)

pt18 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKERZ-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKERZ = RPKM)

pt19 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKHBF-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKHBF = RPKM)

pt20 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKIWK-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKIWK = RPKM)

pt21 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKIYW-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKIYW = RPKM)

pt22 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKKBK-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKKBK = RPKM)

pt23 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKLPD-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKLPD = RPKM)

pt24 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKRUP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKRUP = RPKM)

pt25 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKTCX-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKTCX = RPKM)

pt26 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKVGI-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKVGI = RPKM)

pt27 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKVYM-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAKVYM = RPKM)

pt28 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALBYP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PALBYP = RPKM)

pt29 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALFVW-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PALFVW = RPKM)

pt30 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALGKX-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PALGKX = RPKM)

pt31 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALHVV-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PALHVV = RPKM)

pt32 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALVKV-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PALVKV = RPKM)

pt33 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMVKZ-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAMVKZ = RPKM)

pt34 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMVVP-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAMVVP = RPKM)

pt35 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMYAS-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAMYAS = RPKM)

pt36 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMYGX-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAMYGX = RPKM)

pt37 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMYJX-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAMYJX = RPKM)

pt38 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANAEV-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANAEV = RPKM)

pt39 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANBWF-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANBWF = RPKM)

pt40 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANBYS-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANBYS = RPKM)

pt41 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANBZH-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANBZH = RPKM)

pt42 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANCSC-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANCSC = RPKM)

pt43 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANDER-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANDER = RPKM)

pt44 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANDIX-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANDIX = RPKM)

pt45 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANFMG-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANFMG = RPKM)

pt46 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANGDN-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANGDN = RPKM)

pt47 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANGJY-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANGJY = RPKM)

pt48 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANGTF-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANGTF = RPKM)

pt49 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANHYK-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANHYK = RPKM)

pt50 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANINI-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANINI = RPKM)

pt51 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANJGR-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANJGR = RPKM)

pt52 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKFG-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANKFG = RPKM)

pt53 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKFZ-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANKFZ = RPKM)

pt54 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKKE-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANKKE = RPKM)

pt55 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKNB-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANKNB = RPKM)

pt56 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLIR-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLIR = RPKM)

pt57 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLIZ-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLIZ = RPKM)

pt58 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLJN-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLJN = RPKM)

pt59 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLKB-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLKB = RPKM)

pt60 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLLX-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLLX = RPKM)

pt61 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLRE-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLRE = RPKM)

pt62 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLXK-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLXK = RPKM)

pt63 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLXM-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANLXM = RPKM)

pt64 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANNHB-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANNHB = RPKM)

pt65 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANNJI-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANNJI = RPKM)

pt66 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANPKN-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANPKN = RPKM)

pt67 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANPTM-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANPTM = RPKM)

pt68 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANSBH-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANSBH = RPKM)

pt69 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANSJB-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANSJB = RPKM)

pt70 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTNA-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANTNA = RPKM)

pt71 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTPW-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANTPW = RPKM)

pt72 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUTB-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANUTB = RPKM)

pt73 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUUA-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANUUA = RPKM)

pt74 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANVBP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANVBP = RPKM)

pt75 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANVGE-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANVGE = RPKM)

pt76 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANWHP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANWHP = RPKM)

pt77 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANYGP-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANYGP = RPKM)

pt78 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANZKA-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANZKA = RPKM)

pt79 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANZWB-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PANZWB = RPKM)

pt80 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPAEG-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPAEG = RPKM)

pt81 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPAWN-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPAWN = RPKM)

pt82 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPBEJ-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPBEJ = RPKM)

pt83 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPUNR-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPUNR = RPKM)

pt84 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVCN-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPVCN = RPKM)

pt85 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVGE-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPVGE = RPKM)

pt86 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVZK-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPVZK = RPKM)

pt87 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWHS-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPWHS = RPKM)

pt88 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWIU-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPWIU = RPKM)

pt89 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWYK-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPWYK = RPKM)

pt90 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWZR-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPWZR = RPKM)

pt91 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPXRJ-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPXRJ = RPKM)

pt92 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPXUF-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPXUF = RPKM)

pt93 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPXWI-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPXWI = RPKM)

pt94 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZIZ-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPZIZ = RPKM)

pt95 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZLT-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PAPZLT = RPKM)

pt96 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARAJX-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARAJX = RPKM)

pt97 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARANT-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARANT = RPKM)

pt98 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBFJ-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARBFJ = RPKM)

pt99 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBIU-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARBIU = RPKM)

pt100 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBRA-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARBRA = RPKM)

pt101 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBTV-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARBTV = RPKM)

pt102 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBXE-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARBXE = RPKM)

pt103 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARCUK-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARCUK = RPKM)

pt104 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARCZL-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARCZL = RPKM)

pt105 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARDDY-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARDDY = RPKM)

pt106 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARDMG-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARDMG = RPKM)

pt107 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARENB-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARENB = RPKM)

pt108 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARFAL-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARFAL = RPKM)

pt109 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARHVK-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARHVK = RPKM)

pt110 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARJCR-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARJCR = RPKM)

pt111 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARKCX-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARKCX = RPKM)

pt112 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARMZF-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARMZF = RPKM)

pt113 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARPDS-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARPDS = RPKM)

pt114 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARPWL-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARPWL = RPKM)

pt115 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARTAL-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARTAL = RPKM)

pt116 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUBT-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARUBT = RPKM)

pt117 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUNX-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARUNX = RPKM)

pt118 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUTH-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARUTH = RPKM)

pt119 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUUB-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARUUB = RPKM)

pt120 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARWDZ-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARWDZ = RPKM)

pt121 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARWXU-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARWXU = RPKM)

pt122 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARXMP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARXMP = RPKM)

pt123 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUWX-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARUWX = RPKM)

pt124 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARXNG-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARXNG = RPKM)

pt125 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARYGA-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARYGA = RPKM)

pt126 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARYVW-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARYVW = RPKM)

pt127 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARZUU-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARZUU = RPKM)

pt128 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARZWH-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PARZWH = RPKM)

pt129 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASBHI-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASBHI = RPKM)

pt130 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASBPW-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASBPW = RPKM)

pt131 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASCCS-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASCCS = RPKM)

pt132 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASCFW-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASCFW = RPKM)

pt133 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASDGX-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASDGX = RPKM)

pt134 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASECW-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASECW = RPKM)

pt135 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASEFD-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASEFD = RPKM)

pt136 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASFEW-09A-05R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASFEW = RPKM)

pt137 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASFNP-09A-02R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASFNP = RPKM)

pt138 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASGMZ-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASGMZ = RPKM)

pt139 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASGWH-09A-04R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASGWH = RPKM)

pt140 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASHBI-09A-01R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASHBI = RPKM)

pt141 <- read.table(url("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASHYZ-09A-03R.gene.quantification.txt"), header = TRUE) %>% dplyr::select(gene, RPKM) %>% dplyr::rename(PASHYZ = RPKM)




pt1_2 <- merge(pt1, pt2, by = "gene")
pt1_3 <- merge(pt1_2, pt3, by = "gene")
pt1_4 <- merge(pt1_3, pt4, by = "gene")
pt1_5 <- merge(pt1_4, pt5, by = "gene") 
pt1_6 <- merge(pt1_5, pt6, by = "gene")
pt1_7 <- merge(pt1_6, pt7, by = "gene")
pt1_8 <- merge(pt1_7, pt8, by = "gene")
pt1_9 <- merge(pt1_8, pt9, by = "gene")
pt1_10 <- merge(pt1_9, pt10, by = "gene")
pt1_11 <- merge(pt1_10, pt11, by = "gene")
pt1_12 <- merge(pt1_11, pt12, by = "gene")
pt1_13 <- merge(pt1_12, pt13, by = "gene")
pt1_14 <- merge(pt1_13, pt14, by = "gene")
pt1_15 <- merge(pt1_14, pt15, by = "gene")
pt1_16 <- merge(pt1_15, pt16, by = "gene")
pt1_17 <- merge(pt1_16, pt17, by = "gene")
pt1_18 <- merge(pt1_17, pt18, by = "gene")
pt1_19 <- merge(pt1_18, pt19, by = "gene")
pt1_20 <- merge(pt1_19, pt20, by = "gene")
pt1_21 <- merge(pt1_20, pt21, by = "gene")
pt1_22 <- merge(pt1_21, pt22, by = "gene")
pt1_23 <- merge(pt1_22, pt23, by = "gene")
pt1_24 <- merge(pt1_23, pt24, by = "gene")
pt1_25 <- merge(pt1_24, pt25, by = "gene")
pt1_26 <- merge(pt1_25, pt26, by = "gene")
pt1_27 <- merge(pt1_26, pt27, by = "gene")
pt1_28 <- merge(pt1_27, pt28, by = "gene")
pt1_29 <- merge(pt1_28, pt29, by = "gene")
pt1_30 <- merge(pt1_29, pt30, by = "gene")
pt1_31 <- merge(pt1_30, pt31, by = "gene")
pt1_32 <- merge(pt1_31, pt32, by = "gene")
pt1_33 <- merge(pt1_32, pt33, by = "gene")
pt1_34 <- merge(pt1_33, pt34, by = "gene")
pt1_35 <- merge(pt1_34, pt35, by = "gene")
pt1_36 <- merge(pt1_35, pt36, by = "gene")
pt1_37 <- merge(pt1_36, pt37, by = "gene")
pt1_38 <- merge(pt1_37, pt38, by = "gene")
pt1_39 <- merge(pt1_38, pt39, by = "gene")
pt1_40 <- merge(pt1_39, pt40, by = "gene")
pt1_41 <- merge(pt1_40, pt41, by = "gene")
pt1_42 <- merge(pt1_41, pt42, by = "gene")
pt1_43 <- merge(pt1_42, pt43, by = "gene")
pt1_44 <- merge(pt1_43, pt44, by = "gene")
pt1_45 <- merge(pt1_44, pt45, by = "gene")
pt1_46 <- merge(pt1_45, pt46, by = "gene")
pt1_47 <- merge(pt1_46, pt47, by = "gene")
pt1_48 <- merge(pt1_47, pt48, by = "gene")
pt1_49 <- merge(pt1_48, pt49, by = "gene")
pt1_50 <- merge(pt1_49, pt50, by = "gene")
pt1_51 <- merge(pt1_50, pt51, by = "gene")
pt1_52 <- merge(pt1_51, pt52, by = "gene")
pt1_53 <- merge(pt1_52, pt53, by = "gene")
pt1_54 <- merge(pt1_53, pt54, by = "gene")
pt1_55 <- merge(pt1_54, pt55, by = "gene")
pt1_56 <- merge(pt1_55, pt56, by = "gene")
pt1_57 <- merge(pt1_56, pt57, by = "gene")
pt1_58 <- merge(pt1_57, pt58, by = "gene")
pt1_59 <- merge(pt1_58, pt59, by = "gene")
pt1_60 <- merge(pt1_59, pt60, by = "gene")
pt1_61 <- merge(pt1_60, pt61, by = "gene")
pt1_62 <- merge(pt1_61, pt62, by = "gene")
pt1_63 <- merge(pt1_62, pt63, by = "gene")
pt1_64 <- merge(pt1_63, pt64, by = "gene")
pt1_65 <- merge(pt1_64, pt65, by = "gene")
pt1_66 <- merge(pt1_65, pt66, by = "gene")
pt1_67 <- merge(pt1_66, pt67, by = "gene")
pt1_68 <- merge(pt1_67, pt68, by = "gene")
pt1_69 <- merge(pt1_68, pt69, by = "gene")
pt1_70 <- merge(pt1_69, pt70, by = "gene")
pt1_71 <- merge(pt1_70, pt71, by = "gene")
pt1_72 <- merge(pt1_71, pt72, by = "gene")
pt1_73 <- merge(pt1_72, pt73, by = "gene")
pt1_74 <- merge(pt1_73, pt74, by = "gene")
pt1_75 <- merge(pt1_74, pt75, by = "gene")
pt1_76 <- merge(pt1_75, pt76, by = "gene")
pt1_77 <- merge(pt1_76, pt77, by = "gene")
pt1_78 <- merge(pt1_77, pt78, by = "gene")
pt1_79 <- merge(pt1_78, pt79, by = "gene")
pt1_80 <- merge(pt1_79, pt80, by = "gene")
pt1_81 <- merge(pt1_80, pt81, by = "gene")
pt1_82 <- merge(pt1_81, pt82, by = "gene")
pt1_83 <- merge(pt1_82, pt83, by = "gene")
pt1_84 <- merge(pt1_83, pt84, by = "gene")
pt1_85 <- merge(pt1_84, pt85, by = "gene")
pt1_86 <- merge(pt1_85, pt86, by = "gene")
pt1_87 <- merge(pt1_86, pt87, by = "gene")
pt1_88 <- merge(pt1_87, pt88, by = "gene")
pt1_89 <- merge(pt1_88, pt89, by = "gene")
pt1_90 <- merge(pt1_89, pt90, by = "gene")
pt1_91 <- merge(pt1_90, pt91, by = "gene")
pt1_92 <- merge(pt1_91, pt92, by = "gene")
pt1_93 <- merge(pt1_92, pt93, by = "gene")
pt1_94 <- merge(pt1_93, pt94, by = "gene")
pt1_95 <- merge(pt1_94, pt95, by = "gene")
pt1_96 <- merge(pt1_95, pt96, by = "gene")
pt1_97 <- merge(pt1_96, pt97, by = "gene")
pt1_98 <- merge(pt1_97, pt98, by = "gene")
pt1_99 <- merge(pt1_98, pt99, by = "gene")
pt1_100 <- merge(pt1_99, pt100, by = "gene")
pt1_101 <- merge(pt1_100, pt101, by = "gene")
pt1_102 <- merge(pt1_101, pt102, by = "gene")
pt1_103 <- merge(pt1_102, pt103, by = "gene")
pt1_104 <- merge(pt1_103, pt104, by = "gene")
pt1_105 <- merge(pt1_104, pt105, by = "gene")
pt1_106 <- merge(pt1_105, pt106, by = "gene")
pt1_107 <- merge(pt1_106, pt107, by = "gene")
pt1_108 <- merge(pt1_107, pt108, by = "gene")
pt1_109 <- merge(pt1_108, pt109, by = "gene")
pt1_110 <- merge(pt1_109, pt110, by = "gene")
pt1_111 <- merge(pt1_110, pt111, by = "gene")
pt1_112 <- merge(pt1_111, pt112, by = "gene")
pt1_113 <- merge(pt1_112, pt113, by = "gene")
pt1_114 <- merge(pt1_113, pt114, by = "gene")
pt1_115 <- merge(pt1_114, pt115, by = "gene")
pt1_116 <- merge(pt1_115, pt116, by = "gene")
pt1_117 <- merge(pt1_116, pt117, by = "gene")
pt1_118 <- merge(pt1_117, pt118, by = "gene")
pt1_119 <- merge(pt1_118, pt119, by = "gene")
pt1_120 <- merge(pt1_119, pt120, by = "gene")
pt1_121 <- merge(pt1_120, pt121, by = "gene")
pt1_122 <- merge(pt1_121, pt122, by = "gene")
pt1_123 <- merge(pt1_122, pt123, by = "gene")
pt1_124 <- merge(pt1_123, pt124, by = "gene")
pt1_125 <- merge(pt1_124, pt125, by = "gene")
pt1_126 <- merge(pt1_125, pt126, by = "gene")
pt1_127 <- merge(pt1_126, pt127, by = "gene")
pt1_128 <- merge(pt1_127, pt128, by = "gene")
pt1_129 <- merge(pt1_128, pt129, by = "gene")
pt1_130 <- merge(pt1_129, pt130, by = "gene")
pt1_131 <- merge(pt1_130, pt131, by = "gene")
pt1_132 <- merge(pt1_131, pt132, by = "gene")
pt1_133 <- merge(pt1_132, pt133, by = "gene")
pt1_134 <- merge(pt1_133, pt134, by = "gene")
pt1_135 <- merge(pt1_134, pt135, by = "gene")
pt1_136 <- merge(pt1_135, pt136, by = "gene")
pt1_137 <- merge(pt1_136, pt137, by = "gene")
pt1_138 <- merge(pt1_137, pt138, by = "gene")
pt1_139 <- merge(pt1_138, pt139, by = "gene")
pt1_140 <- merge(pt1_139, pt140, by = "gene")
pt1_141 <- merge(pt1_140, pt141, by = "gene")



# redo this at the end for all samples
depmap_ids <- expression_v_depscore_2$gene_id

pt1_141$gene_id <- mapIds(org.Hs.eg.db, keys = pt1_141$gene, keytype = "ENSEMBL", column = "SYMBOL")
pt1_141 %>% head(10) %>% view()

target_141 <- pt1_137[, c(129, 1:128, 130:139)]
target_137 %>% head(10) %>% view()

target_137_depmap <- target_137 %>% dplyr::select(-gene) %>%
                    filter(gene_id %in% depmap_ids) 

target_expression_pilot <- as.data.frame(target_137_depmap %>%   # need to have unique = TRUE !
  mutate(gene_id = make.names(gene_id, unique = TRUE)) %>%
  column_to_rownames("gene_id") %>%
  t()) %>%
    rownames_to_column("sample")

write.csv(target_expression_pilot, "~/Documents/R/pedcancer/target_data/TARGET_100/target_expression_pilot.csv", row.names = FALSE)

expression_samples <- pull(target_expression_pilot %>% dplyr::select(sample))

####



```


```{r}
## TRANSCRIPTIONAL VARIABILITY -- IQR

pt1_100_IQR <- pt1_100 %>% 
  pivot_longer(cols = 2:101, 
               names_to = "sample", 
               values_to = "expression") %>%
  group_by(gene) %>% 
  summarise(IQR = IQR(expression)) %>%
  arrange(desc(IQR))

pt1_100_IQR %>% top_frac(0.1) %>% ggplot(mapping = aes(x = fct_reorder(gene, IQR), y = IQR)) + geom_point()


pt1_100_IQR$gene_id <- mapIds(org.Hs.eg.db, keys = pt1_100_IQR$gene, keytype = "ENSEMBL", column = "SYMBOL")



target_depmap_aml_iqr <- merge(pt1_100_IQR, aml_exp_IQR, by = "gene_id")

target_iqr_depmap_plot <- target_depmap_aml_iqr %>% 
  ggplot(mapping = aes(x = fct_reorder(gene_id, IQR.x), y = IQR.x)) + 
  geom_point(alpha = 0.2) +
  geom_point(data = target_depmap_aml_iqr %>% filter(gene_id %in% IQR_top & IQR.x < 200), 
             mapping = aes(x = gene_id, y = IQR.x, 
             color = "green")) +  
  geom_point(data = target_depmap_aml_iqr %>% filter(gene_id %in% IQR_top & IQR.x >= 200), 
             mapping = aes(x = gene_id, y = IQR.x, 
             color = "red")) +
  geom_label_repel(data = target_depmap_aml_iqr %>% filter(gene_id %in% IQR_top & IQR.x >= 200), 
             mapping = aes(x = gene_id, y = IQR.x, 
             label = gene_id), max.overlaps = 40) +
   labs(title = "Variability of gene expression in pediatric AML", 
             x = "Ranked genes", 
             y = "IQR of Gene Expression") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_text(size = 16),
         axis.ticks.x = element_blank(),
         axis.title.y = element_text(size = 16),
         axis.text.y = element_text(size = 12),
         axis.line = element_line(size = 1, color = "black", linetype = 1),
         legend.position = "none") + 
  geom_hline(yintercept = 200)

target_iqr_depmap_plot_2 <- target_depmap_aml_iqr %>% 
  ggplot(mapping = aes(x = fct_reorder(gene_id, IQR.x), y = IQR.x)) + 
  geom_point(alpha = 0.2) +
  geom_point(data = target_depmap_aml_iqr %>% filter(gene_id %in% IQR_top), 
             mapping = aes(x = gene_id, y = IQR.x, 
             color = "red")) +
   labs(title = "Variability of gene expression in pediatric AML", 
             x = "Ranked genes", 
             y = "IQR of Gene Expression") +
   theme(axis.text.x = element_blank(),
         axis.title.x = element_text(size = 16),
         axis.ticks.x = element_blank(),
         axis.title.y = element_text(size = 16),
         axis.text.y = element_text(size = 12),
         axis.line = element_line(size = 1, color = "black", linetype = 1),
         legend.position = "none") + 
  geom_hline(yintercept = 200)


target_top_iqr <- target_depmap_aml_iqr %>% filter(IQR.x >= 200) %>% filter(gene_id %in% IQR_top)

pt1_100$gene_id <- mapIds(org.Hs.eg.db, keys = pt1_100_exp$gene, keytype = "ENSEMBL", column = "SYMBOL")

pt1_100_exp <- pt1_100 %>% pivot_longer(cols = 2:101,
                         names_to = "sample", 
                         values_to = "expression") %>%
  group_by(gene) %>%
  summarise(x_exp = median(expression))

pt1_100_exp$gene_id <- mapIds(org.Hs.eg.db, keys = pt1_100_exp$gene, keytype = "ENSEMBL", column = "SYMBOL")

depmap_aml_exp <- expression_v_depscore_2 %>% dplyr::filter(lineage_subtype == "AML") %>% dplyr::select(gene_id, expression) %>% group_by(gene_id) %>% summarise(expression = median(expression))

merge(pt1_100_exp, depmap_aml_exp, by = "gene_id") %>% ggplot(mapping = aes(x = x_exp, y = expression)) + geom_point()

# TFRC ENSG00000072274
# BCL2 ENSG00000171791
# BCL2L1 ENSG00000171552
# BCL2L2 ENSG00000129473
# MCL1 ENSG00000143384
# CMYC ENSG00000136997
# HIF1A ENSG00000100644
# MTOR ENSG00000198793
# ACVR1 ENSG00000115170


data_cor <- pt1_100 %>% 
  filter(gene_id == "RUNX1" | gene_id == "BCL2L1") %>% 
  dplyr::select(-gene) %>%
  pivot_longer(cols = -gene_id, 
               names_to = "sample", 
               values_to = "expression") %>%
  pivot_wider(names_from = gene_id, 
              values_from = expression)

ggscatter(data_cor, x = "RUNX1", y = "BCL2L1", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson", cor.coef.size = 8,
          xlab = "RUNX1", ylab = "BCL2L1") +
  theme(axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20))
              


bcl2l1_lo_genes <- c("S100A9", "S100A8", "ELANE", "CFD", "RNASE2", "AZU1", "PRTN3", "CXCR4", "EVI2B")

aml_cells <- expression_v_depscore_2 %>%
  filter(lineage_subtype == "AML", 
         gene_id == "BCL2L1") %>%
  mutate(bcl2l1_essential = ifelse(depscore <= -1, 'yes', 'no')) %>%
  dplyr::select(stripped_cell_line_name, bcl2l1_essential)

bcl2l1_ess_aml <- pull(aml_cells %>% filter(bcl2l1_essential == "yes") %>% dplyr::select(stripped_cell_line_name))

bcl2l1_noness_aml <- pull(aml_cells %>% filter(essential == "no") %>% dplyr::select(stripped_cell_line_name))

lo_gene_aml <- expression_v_depscore_2 %>%
  filter(lineage_subtype == "AML",
         gene_id %in% bcl2l1_lo_genes) %>%
  dplyr::select(stripped_cell_line_name, gene_id, expression)

bcl2l1_panel_plot <- merge(lo_gene_aml, aml_cells, by = "stripped_cell_line_name") %>%
  group_by(gene_id) %>%
  ggplot(mapping = aes(x = gene_id, y = expression, fill = bcl2l1_essential)) + 
  geom_boxplot() +
  facet_wrap(~bcl2l1_essential) +
    xlab("gene_id") +
    ylab("gene expression") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, angle = 30, hjust = 1, vjust = 1, color = "black"),
          axis.text.y = element_text(size = 14))  

```

```{r}

# examine above gene list in TF1 and U937 specifically

expression_v_depscore_2 %>%
  filter(stripped_cell_line_name == "TF1" |
           stripped_cell_line_name == "U937") %>%
  filter(gene_id %in% bcl2l1_lo_genes) %>%
  dplyr::select(stripped_cell_line_name, gene_id, expression)

```


```{r}

# BEAT AML validation of gene expression vs bcl2 inhibitor sensitivity ?

abt737 <- read.csv("~/Documents/R/pedcancer/beat_aml data/abt737_sensitivity.csv")
panel_exp <- read.csv("~/Documents/R/pedcancer/beat_aml data/panel_rnaseq.csv")

abt737 %>% filter(RNAseq == "TRUE") %>% ggplot(mapping = aes(x = AUC)) + geom_histogram(binwidth = 5, fill = "white", color = "black")

lower <- abt737$AUC %>% quantile(probs = 0.10, na.rm = TRUE)
upper <- abt737$AUC %>% quantile(probs = 0.90, na.rm = TRUE)



upper_samples <- pull(abt737 %>% filter(AUC >= upper) %>% dplyr::select(SampleID))
lower_samples <- pull(abt737 %>% filter(AUC <= lower) %>% dplyr::select(SampleID))

beataml_panel_plot <- panel_exp %>% pivot_longer(cols = -gene_id,
                           names_to = "SampleID", 
                           values_to = "expression") %>%
  filter(SampleID %in% upper_samples | SampleID %in% lower_samples) %>%
  mutate(group = ifelse(SampleID %in% upper_samples, 'upper', 'lower')) %>%
  group_by(gene_id) %>%
  ggplot(mapping = aes(x = group, y = expression, fill = group)) +
  geom_boxplot() +
  facet_wrap(~gene_id) +
          xlab("") +
    ylab("Expression") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, color = "black"),
          axis.text.y = element_text(size = 14))


```


```{r}

# Lee et al. AML dataset

lee_expression <- read.csv("~/Documents/R/pedcancer/beat_aml data/lee_panel_rnaseq.csv")
lee_auc <- read.csv("~/Documents/R/pedcancer/beat_aml data/lee_panel_auc.csv")


# find samples with strong or weak ABT263 response

lee_abt263 <- lee_auc %>% filter(UID == "ABT-263") %>%
  dplyr::select(- X53.drugs) %>%
  pivot_longer(cols = 2:31,
               names_to = "sample",
               values_to = "auc") 

lee_abt263 %>%
  ggplot(mapping = aes(x = fct_reorder(sample, -auc), y = auc)) + geom_col() +
      xlab("Sample") +
    ylab("AUC for ABT263") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, angle = 30, hjust = 1, vjust = 1, color = "black"),
          axis.text.y = element_text(size = 14)) 
  
lee_abt263_ends <- lee_abt263 %>% 
  filter(auc >= 600 |auc <= 450) %>%
  arrange(desc(auc)) 

lee_ends_samples <- c("AML069", "AML063", "AML099", "AML039", "AML017", "AML079", "AML012", "AML083", "AML074", "AML013", "AML070", "AML051", "AML059", "AML078", "AML056", "AML055")	
lee_ends_hi <- c("AML069", "AML063", "AML099", "AML039", "AML017", "AML079", "AML012", "AML083")
lee_ends_lo <- c("AML074", "AML013", "AML070", "AML051", "AML059", "AML078", "AML056", "AML055")

# look for genes of interest in these samples

lee_expression %>%
  filter(gene_id %in% bcl2l1_lo_genes) %>%
  pivot_longer(cols = -gene_id, 
               names_to = "samples",
               values_to = "expression") %>%
  filter(samples %in% lee_ends_samples) %>%
  mutate(group = ifelse(samples %in% lee_ends_lo, 'lower', 'upper')) %>%
  ggplot(mapping = aes(x = group, y = expression, fill = group)) + geom_boxplot() +
  facet_wrap(~gene_id) +
        xlab("") +
    ylab("Expression") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, hjust = 1, vjust = 1, color = "black"),
          axis.text.y = element_text(size = 14)) 


# Repeat code chunk for ABT 737

lee_abt737 <- lee_auc %>% filter(UID == "ABT-737") %>%
  dplyr::select(- X53.drugs) %>%
  pivot_longer(cols = 2:31,
               names_to = "sample",
               values_to = "auc") 

lee_abt737 %>%
  ggplot(mapping = aes(x = fct_reorder(sample, -auc), y = auc)) + geom_col() +
      xlab("Sample") +
    ylab("AUC for ABT737") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, hjust = 1, vjust = 1, color = "black"),
          axis.text.y = element_text(size = 14)) 
  
lee_abt737_ends <- lee_abt737 %>% 
  filter(auc >= 600 |auc <= 300) %>%
  arrange(desc(auc)) 

lee_ends_samples_737 <- c("AML069", "AML063", "AML030", "AML012", "AML099", "AML083", "AML059", "AML078", "AML056", "AML070", "AML055")
lee_ends_hi_737 <- c("AML069", "AML063", "AML030", "AML012", "AML099", "AML083")

# look for genes of interest in these samples

lee_expression %>%
  filter(gene_id %in% bcl2l1_lo_genes) %>%
  pivot_longer(cols = -gene_id, 
               names_to = "samples",
               values_to = "expression") %>%
  filter(samples %in% lee_ends_samples_737) %>%
  mutate(group = ifelse(samples %in% lee_ends_hi_737, 'upper', 'lower')) %>%
  ggplot(mapping = aes(x = group, y = expression, fill = group)) + geom_boxplot() +
  facet_wrap(~gene_id) +
        xlab("") +
    ylab("Expression") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, color = "black"),
          axis.text.y = element_text(size = 14)) 

```

```{r}

expression_v_depscore_2 %>%
  filter(lineage_subtype == "AML",
         gene_id == "BCL2L1") %>%
  dplyr::select(stripped_cell_line_name, depscore) %>%
  arrange(desc(-depscore)) %>%
  ggplot(mapping = aes(x = fct_reorder(stripped_cell_line_name, depscore), y = depscore)) + 
  geom_col() +
          xlab("AML Cell Line") +
    ylab("BCL2L1 depscore") +
    theme(axis.title.x = element_text(size = 14), 
          axis.title.y = element_text(size = 16),
          axis.text.x = element_text(size = 10, angle = 30, hjust = 1, vjust = 1, color = "black"),
          axis.text.y = element_text(size = 14)) 
  

```

```{r}

# examine 'amino acid metabolic process' GO genes in BCL2L1 essential and non essential AML cells

# list: STAT5B, GAMT, AADAT, CKM, KYAT1, CKMT1A, MTHFR, MUT, KMO, ASS1, SLC6A8, ALDH6A1, GATM, DDAH2, CPS1, DDAH1, MTHFD1, NOX4, DPEP1, KYAT3, MTRR, HDC, CKMT2, GHR, IYD, CBS, SRR, HNF4A, CKB, ATP7A, STAT5A

aa_metabolic <- c("STAT5B", "GAMT", "AADAT", "CKM", "KYAT1", "CKMT1A", "MTHFR", "MUT", "KMO", "ASS1", "SLC6A8", "ALDH6A1", "GATM", "DDAH2", "CPS1", "DDAH1", "MTHFD1", "NOX4", "DPEP1", "KYAT3", "MTRR", "HDC", "CKMT2", "GHR", "IYD", "CBS", "SRR", "HNF4A", "CKB", "ATP7A", "STAT5A")

bcl2l1_ess <- pull(expression_v_depscore_2 %>% filter(lineage_subtype == "AML", gene_id == "BCL2L1", depscore <= -1.5) %>% dplyr::select(stripped_cell_line_name))

bcl2l1_noness <- pull(expression_v_depscore_2 %>% filter(lineage_subtype == "AML", gene_id == "BCL2L1", depscore >= -0.5) %>% dplyr::select(stripped_cell_line_name))
  
aa_metabolic_expression <- expression_v_depscore_2 %>%
  filter(stripped_cell_line_name %in% bcl2l1_ess |
           stripped_cell_line_name %in% bcl2l1_noness) %>%
  filter(gene_id %in% aa_metabolic) %>%
  mutate(group = ifelse(stripped_cell_line_name %in% bcl2l1_ess, 'essential', 'nonessential')) %>%
  dplyr::select(stripped_cell_line_name, gene_id, expression)

ggplot(aa_metaboilc_expression, mapping = aes(x = group, y = expression)) + 
  geom_boxplot() + 
  facet_wrap(~gene_id)

aa_metabolic_matrix <- as.matrix(aa_metabolic_expression %>%
  pivot_wider(names_from = stripped_cell_line_name, 
              values_from = expression) %>%
  column_to_rownames("gene_id"))

pheatmap(aa_metabolic_matrix, scale = "none", cluster_cols = TRUE)

```


```{r}

# TARGET miRNA data

pt1_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PABGKN-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PABGKN = reads_per_million_miRNA_mapped)

pt2_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PABLDZ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PABLDZ = reads_per_million_miRNA_mapped)

pt3_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PACDZR-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PACDZR = reads_per_million_miRNA_mapped)

pt4_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PADYIR-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PADYIR = reads_per_million_miRNA_mapped)

pt5_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PADZCG-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PADZCG = reads_per_million_miRNA_mapped)

pt6_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PADZKD-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PADZKD = reads_per_million_miRNA_mapped)

pt7_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEAFC-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEAFC = reads_per_million_miRNA_mapped)

pt8_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEAKL-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEAKL = reads_per_million_miRNA_mapped)

pt9_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAECBZ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAECBZ = reads_per_million_miRNA_mapped)

pt10_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAECCE-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAECCE = reads_per_million_miRNA_mapped)

pt11_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEENN-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEENN = reads_per_million_miRNA_mapped)

pt12_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEERJ-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEERJ = reads_per_million_miRNA_mapped)

pt13_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEEYP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEEYP = reads_per_million_miRNA_mapped)

pt14_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEFGR-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEFGR = reads_per_million_miRNA_mapped)

pt15_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEFHC-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEFHC = reads_per_million_miRNA_mapped)

pt16_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEIKD-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEIKD = reads_per_million_miRNA_mapped)

pt17_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAEJBT-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAEJBT = reads_per_million_miRNA_mapped)

pt18_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAERAH-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAERAH = reads_per_million_miRNA_mapped)

pt19_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAESVN-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAESVN = reads_per_million_miRNA_mapped)

pt20_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKERZ-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKERZ = reads_per_million_miRNA_mapped)

pt21_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKHBF-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKHBF = reads_per_million_miRNA_mapped)

pt22_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKIWK-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKIWK = reads_per_million_miRNA_mapped)

pt23_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKIYW-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKIYW = reads_per_million_miRNA_mapped)

pt24_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKKBK-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKKBK = reads_per_million_miRNA_mapped)

pt25_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKLPD-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKLPD = reads_per_million_miRNA_mapped)

pt26_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKRUP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKRUP = reads_per_million_miRNA_mapped)

pt27_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKTCX-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKTCX = reads_per_million_miRNA_mapped)

pt28_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKVGI-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKVGI = reads_per_million_miRNA_mapped)

pt29_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAKVYM-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAKVYM = reads_per_million_miRNA_mapped)

pt30_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALBYP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PALBYP = reads_per_million_miRNA_mapped)

pt31_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALFVW-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PALFVW = reads_per_million_miRNA_mapped)

pt32_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALGKX-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PALGKX = reads_per_million_miRNA_mapped)

pt33_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALHVV-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PALHVV = reads_per_million_miRNA_mapped)

pt34_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PALVKV-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PALVKV = reads_per_million_miRNA_mapped)

pt35_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMVKZ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAMVKZ = reads_per_million_miRNA_mapped)

pt36_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMVVP-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAMVVP = reads_per_million_miRNA_mapped)

pt37_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMYAS-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAMYAS = reads_per_million_miRNA_mapped)

pt38_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMYGX-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAMYGX = reads_per_million_miRNA_mapped)

pt39_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAMYJX-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAMYJX = reads_per_million_miRNA_mapped)

pt40_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANAEV-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANAEV = reads_per_million_miRNA_mapped)

pt41_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANBWF-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANBWF = reads_per_million_miRNA_mapped)

pt42_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANBYS-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANBYS = reads_per_million_miRNA_mapped)

pt43_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANBZH-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANBZH = reads_per_million_miRNA_mapped)

pt44_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANDDC-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANDDC = reads_per_million_miRNA_mapped)

pt45_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANDER-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANDER = reads_per_million_miRNA_mapped)

pt46_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANDIX-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANDIX = reads_per_million_miRNA_mapped)

pt47_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANFMG-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANFMG = reads_per_million_miRNA_mapped)

pt48_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANGDN-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANGDN = reads_per_million_miRNA_mapped)

pt49_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANGJY-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANGJY = reads_per_million_miRNA_mapped)

pt50_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANGTF-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANGTF = reads_per_million_miRNA_mapped)

pt51_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANIGD-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANIGD = reads_per_million_miRNA_mapped)

pt52_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANINI-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANINI = reads_per_million_miRNA_mapped)

pt53_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANJFJ-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANJFJ = reads_per_million_miRNA_mapped)

pt54_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANJGR-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANJGR = reads_per_million_miRNA_mapped)

pt55_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANJIM-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANJIM = reads_per_million_miRNA_mapped)

pt56_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKBX-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKBX = reads_per_million_miRNA_mapped)

pt57_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKCE-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKCE = reads_per_million_miRNA_mapped)

pt58_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKEF-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKEF = reads_per_million_miRNA_mapped)

pt59_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKFG-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKFG = reads_per_million_miRNA_mapped)

pt60_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKFZ-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKFZ = reads_per_million_miRNA_mapped)

pt61_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKKE-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKKE = reads_per_million_miRNA_mapped)

pt62_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANKNB-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANKNB = reads_per_million_miRNA_mapped)

pt63_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLIR-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLIR = reads_per_million_miRNA_mapped)

pt64_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLIZ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLIZ = reads_per_million_miRNA_mapped)

pt65_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLJN-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLJN = reads_per_million_miRNA_mapped)

pt66_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLKB-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLKB = reads_per_million_miRNA_mapped)

pt67_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLLX-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLLX = reads_per_million_miRNA_mapped)

pt68_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLRE-09A-05R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLRE = reads_per_million_miRNA_mapped)

pt69_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLXK-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLXK = reads_per_million_miRNA_mapped)

pt70_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANLXM-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANLXM = reads_per_million_miRNA_mapped)

pt71_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANNHB-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANNHB = reads_per_million_miRNA_mapped)

pt72_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANNJI-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANNJI = reads_per_million_miRNA_mapped)

pt73_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANPTM-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANPTM = reads_per_million_miRNA_mapped)

pt74_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANRWM-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANRWM = reads_per_million_miRNA_mapped)

pt75_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANSBH-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANSBH = reads_per_million_miRNA_mapped)

pt76_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANSDI-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANSDI = reads_per_million_miRNA_mapped)

pt77_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANSJB-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANSJB = reads_per_million_miRNA_mapped)

pt78_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTIV-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANTIV = reads_per_million_miRNA_mapped)

pt79_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTNA-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANTNA = reads_per_million_miRNA_mapped)

pt80_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTPW-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANTPW = reads_per_million_miRNA_mapped)

pt81_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTRL-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANTRL = reads_per_million_miRNA_mapped)

pt82_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANTXR-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANTXR = reads_per_million_miRNA_mapped)

pt83_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUDS-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANUDS = reads_per_million_miRNA_mapped)

pt84_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUNT-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANUNT = reads_per_million_miRNA_mapped)

pt85_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUTB-09A-05R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANUTB = reads_per_million_miRNA_mapped)

pt86_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUTX-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANUTX = reads_per_million_miRNA_mapped)

pt87_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUUA-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANUUA = reads_per_million_miRNA_mapped)

pt88_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANUUU-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANUUU = reads_per_million_miRNA_mapped)

pt89_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANVBP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANVBP = reads_per_million_miRNA_mapped)

pt90_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANVGE-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANVGE = reads_per_million_miRNA_mapped)

pt91_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANWHP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANWHP = reads_per_million_miRNA_mapped)

pt92_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANXPU-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANXPU = reads_per_million_miRNA_mapped)

pt93_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANXWX-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANXWX = reads_per_million_miRNA_mapped)

pt94_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANYCG-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANYCG = reads_per_million_miRNA_mapped)

pt95_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANYGP-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANYGP = reads_per_million_miRNA_mapped)

pt96_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANZKA-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANZKA = reads_per_million_miRNA_mapped)

pt97_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANZWB-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANZWB = reads_per_million_miRNA_mapped)

pt98_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PANZXI-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PANZXI = reads_per_million_miRNA_mapped)

pt99_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPAEG-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPAEG = reads_per_million_miRNA_mapped)

pt100_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPAPD-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPAPD = reads_per_million_miRNA_mapped)

pt101_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPAWN-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPAWN = reads_per_million_miRNA_mapped)

pt102_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPBEJ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPBEJ = reads_per_million_miRNA_mapped)

pt103_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPSCM-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPSCM = reads_per_million_miRNA_mapped)

pt104_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPUNR-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPUNR = reads_per_million_miRNA_mapped)

pt105_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPUSR-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPUSR = reads_per_million_miRNA_mapped)

pt106_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVCN-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPVCN = reads_per_million_miRNA_mapped)

pt107_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVDV-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPVDV = reads_per_million_miRNA_mapped)

pt108_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVET-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPVET = reads_per_million_miRNA_mapped)

pt109_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVGE-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPVGE = reads_per_million_miRNA_mapped)

pt110_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVKG-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPVKG = reads_per_million_miRNA_mapped)

pt111_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPVZK-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPVZK = reads_per_million_miRNA_mapped)

pt112_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWFN-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPWFN = reads_per_million_miRNA_mapped)

pt113_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWHS-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPWHS = reads_per_million_miRNA_mapped)

pt114_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWIU-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPWIU = reads_per_million_miRNA_mapped)

pt115_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWYK-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPWYK = reads_per_million_miRNA_mapped)

pt116_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPWZR-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPWZR = reads_per_million_miRNA_mapped)

pt117_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPXRJ-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPXRJ = reads_per_million_miRNA_mapped)

pt118_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPXUF-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPXUF = reads_per_million_miRNA_mapped)

pt119_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPXWI-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPXWI = reads_per_million_miRNA_mapped)

pt120_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPYUI-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPYUI = reads_per_million_miRNA_mapped)

pt121_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZCL-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZCL = reads_per_million_miRNA_mapped)

pt122_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZFV-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZFV = reads_per_million_miRNA_mapped)

pt123_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZIZ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZIZ = reads_per_million_miRNA_mapped)

pt124_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZLT-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZLT = reads_per_million_miRNA_mapped)

pt125_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZTC-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZTC = reads_per_million_miRNA_mapped)

pt126_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZXN-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZXN = reads_per_million_miRNA_mapped)

pt127_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZYS-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZYS = reads_per_million_miRNA_mapped)

pt128_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PAPZZE-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PAPZZE = reads_per_million_miRNA_mapped)

pt129_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARAEG-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARAEG = reads_per_million_miRNA_mapped)

pt130_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARAFA-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARAFA = reads_per_million_miRNA_mapped)

pt131_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARAIC-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARAIC = reads_per_million_miRNA_mapped)

pt132_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARAJX-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARAJX = reads_per_million_miRNA_mapped)

pt133_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARANT-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARANT = reads_per_million_miRNA_mapped)

pt134_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARAUM-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARAUM = reads_per_million_miRNA_mapped)

pt135_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBCV-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBCV = reads_per_million_miRNA_mapped)

pt136_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBEL-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBEL = reads_per_million_miRNA_mapped)

pt137_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBFJ-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBFJ = reads_per_million_miRNA_mapped)

pt138_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBIU-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBIU = reads_per_million_miRNA_mapped)

pt139_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBRA-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBRA = reads_per_million_miRNA_mapped)

pt140_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBTV-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBTV = reads_per_million_miRNA_mapped)

pt141_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARBXE-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARBXE = reads_per_million_miRNA_mapped)

pt142_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARCUK-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARCUK = reads_per_million_miRNA_mapped)

pt143_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARCZL-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARCZL = reads_per_million_miRNA_mapped)

pt144_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARDDY-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARDDY = reads_per_million_miRNA_mapped)

pt145_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARDMG-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARDMG = reads_per_million_miRNA_mapped)

pt146_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARENB-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARENB = reads_per_million_miRNA_mapped)

pt147_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARFAL-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARFAL = reads_per_million_miRNA_mapped)

pt148_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARHVK-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARHVK = reads_per_million_miRNA_mapped)

pt149_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARJCR-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARJCR = reads_per_million_miRNA_mapped)

pt150_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARKCX-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARKCX = reads_per_million_miRNA_mapped)

pt151_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARMZF-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARMZF = reads_per_million_miRNA_mapped)

pt152_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARPDS-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARPDS = reads_per_million_miRNA_mapped)

pt153_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARPWL-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARPWL = reads_per_million_miRNA_mapped)

pt154_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARTAL-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARTAL = reads_per_million_miRNA_mapped)

pt155_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUBT-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARUBT = reads_per_million_miRNA_mapped)

pt156_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUNX-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARUNX = reads_per_million_miRNA_mapped)

pt157_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUTH-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARUTH = reads_per_million_miRNA_mapped)

pt158_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUUB-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARUUB = reads_per_million_miRNA_mapped)

pt159_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARWDZ-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARWDZ = reads_per_million_miRNA_mapped)

pt160_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARWXU-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARWXU = reads_per_million_miRNA_mapped)

pt161_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARXMP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARXMP = reads_per_million_miRNA_mapped)

pt162_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARUWX-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARUWX = reads_per_million_miRNA_mapped)

pt163_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARXNG-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARXNG = reads_per_million_miRNA_mapped)

pt164_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARYGA-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARYGA = reads_per_million_miRNA_mapped)

pt165_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARYVW-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARYVW = reads_per_million_miRNA_mapped)

pt166_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARZUU-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARZUU = reads_per_million_miRNA_mapped)

pt167_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PARZWH-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PARZWH = reads_per_million_miRNA_mapped)

pt168_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASBHI-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASBHI = reads_per_million_miRNA_mapped)

pt169_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASBPW-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASBPW = reads_per_million_miRNA_mapped)

pt170_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASCCS-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASCCS = reads_per_million_miRNA_mapped)

pt171_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASCFW-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASCFW = reads_per_million_miRNA_mapped)

pt172_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASDGX-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASDGX = reads_per_million_miRNA_mapped)

pt173_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASECW-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASECW = reads_per_million_miRNA_mapped)

pt174_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASEFD-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASEFD = reads_per_million_miRNA_mapped)

pt175_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASFEW-09A-05R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASFEW = reads_per_million_miRNA_mapped)

pt176_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASFNP-09A-02R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASFNP = reads_per_million_miRNA_mapped)

pt177_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASGMZ-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASGMZ = reads_per_million_miRNA_mapped)

pt178_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASGWH-09A-04R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASGWH = reads_per_million_miRNA_mapped)

pt179_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASHBI-09A-01R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASHBI = reads_per_million_miRNA_mapped)

pt180_miRNA <- read.table(url("https://target-data.nci.nih.gov/Public/AML/miRNA-seq/L3/expression/BCCA/TARGET-NCI/TARGET-20-PASHYZ-09A-03R.mirna.quantification.txt"), header = TRUE) %>% dplyr::select(miRNA_ID, reads_per_million_miRNA_mapped) %>% dplyr::rename(PASHYZ = reads_per_million_miRNA_mapped)



# merge dfs

pt1_2miRNA <- merge(pt1_miRNA, pt2_miRNA, by = "miRNA_ID")
pt1_3miRNA <- merge(pt1_2miRNA, pt3_miRNA, by = "miRNA_ID")
pt1_4miRNA <- merge(pt1_3miRNA, pt4_miRNA, by = "miRNA_ID")
pt1_5miRNA <- merge(pt1_4miRNA, pt5_miRNA, by = "miRNA_ID")
pt1_6miRNA <- merge(pt1_5miRNA, pt6_miRNA, by = "miRNA_ID")
pt1_7miRNA <- merge(pt1_6miRNA, pt7_miRNA, by = "miRNA_ID")
pt1_8miRNA <- merge(pt1_7miRNA, pt8_miRNA, by = "miRNA_ID")
pt1_9miRNA <- merge(pt1_8miRNA, pt9_miRNA, by = "miRNA_ID")
pt1_10miRNA <- merge(pt1_9miRNA, pt10_miRNA, by = "miRNA_ID")

pt1_11miRNA <- merge(pt1_10miRNA, pt11_miRNA, by = "miRNA_ID")
pt1_12miRNA <- merge(pt1_11miRNA, pt12_miRNA, by = "miRNA_ID")
pt1_13miRNA <- merge(pt1_12miRNA, pt13_miRNA, by = "miRNA_ID")
pt1_14miRNA <- merge(pt1_13miRNA, pt14_miRNA, by = "miRNA_ID")
pt1_15miRNA <- merge(pt1_14miRNA, pt15_miRNA, by = "miRNA_ID")
pt1_16miRNA <- merge(pt1_15miRNA, pt16_miRNA, by = "miRNA_ID")
pt1_17miRNA <- merge(pt1_16miRNA, pt17_miRNA, by = "miRNA_ID")
pt1_18miRNA <- merge(pt1_17miRNA, pt18_miRNA, by = "miRNA_ID")
pt1_19miRNA <- merge(pt1_18miRNA, pt19_miRNA, by = "miRNA_ID")
pt1_20miRNA <- merge(pt1_19miRNA, pt20_miRNA, by = "miRNA_ID")

pt1_21miRNA <- merge(pt1_20miRNA, pt21_miRNA, by = "miRNA_ID")
pt1_22miRNA <- merge(pt1_21miRNA, pt22_miRNA, by = "miRNA_ID")
pt1_23miRNA <- merge(pt1_22miRNA, pt23_miRNA, by = "miRNA_ID")
pt1_24miRNA <- merge(pt1_23miRNA, pt24_miRNA, by = "miRNA_ID")
pt1_25miRNA <- merge(pt1_24miRNA, pt25_miRNA, by = "miRNA_ID")
pt1_26miRNA <- merge(pt1_25miRNA, pt26_miRNA, by = "miRNA_ID")
pt1_27miRNA <- merge(pt1_26miRNA, pt27_miRNA, by = "miRNA_ID")
pt1_28miRNA <- merge(pt1_27miRNA, pt28_miRNA, by = "miRNA_ID")
pt1_29miRNA <- merge(pt1_28miRNA, pt29_miRNA, by = "miRNA_ID")
pt1_30miRNA <- merge(pt1_29miRNA, pt30_miRNA, by = "miRNA_ID")

pt1_31miRNA <- merge(pt1_30miRNA, pt31_miRNA, by = "miRNA_ID")
pt1_32miRNA <- merge(pt1_31miRNA, pt32_miRNA, by = "miRNA_ID")
pt1_33miRNA <- merge(pt1_32miRNA, pt33_miRNA, by = "miRNA_ID")
pt1_34miRNA <- merge(pt1_33miRNA, pt34_miRNA, by = "miRNA_ID")
pt1_35miRNA <- merge(pt1_34miRNA, pt35_miRNA, by = "miRNA_ID")
pt1_36miRNA <- merge(pt1_35miRNA, pt36_miRNA, by = "miRNA_ID")
pt1_37miRNA <- merge(pt1_36miRNA, pt37_miRNA, by = "miRNA_ID")
pt1_38miRNA <- merge(pt1_37miRNA, pt38_miRNA, by = "miRNA_ID")
pt1_39miRNA <- merge(pt1_38miRNA, pt39_miRNA, by = "miRNA_ID")
pt1_40miRNA <- merge(pt1_39miRNA, pt40_miRNA, by = "miRNA_ID")

pt1_41miRNA <- merge(pt1_40miRNA, pt41_miRNA, by = "miRNA_ID")
pt1_42miRNA <- merge(pt1_41miRNA, pt42_miRNA, by = "miRNA_ID")
pt1_43miRNA <- merge(pt1_42miRNA, pt43_miRNA, by = "miRNA_ID")
pt1_44miRNA <- merge(pt1_43miRNA, pt44_miRNA, by = "miRNA_ID")
pt1_45miRNA <- merge(pt1_44miRNA, pt45_miRNA, by = "miRNA_ID")
pt1_46miRNA <- merge(pt1_45miRNA, pt46_miRNA, by = "miRNA_ID")
pt1_47miRNA <- merge(pt1_46miRNA, pt47_miRNA, by = "miRNA_ID")
pt1_48miRNA <- merge(pt1_47miRNA, pt48_miRNA, by = "miRNA_ID")
pt1_49miRNA <- merge(pt1_48miRNA, pt49_miRNA, by = "miRNA_ID")
pt1_50miRNA <- merge(pt1_49miRNA, pt50_miRNA, by = "miRNA_ID")

pt1_51miRNA <- merge(pt1_50miRNA, pt51_miRNA, by = "miRNA_ID")
pt1_52miRNA <- merge(pt1_51miRNA, pt52_miRNA, by = "miRNA_ID")
pt1_53miRNA <- merge(pt1_52miRNA, pt53_miRNA, by = "miRNA_ID")
pt1_54miRNA <- merge(pt1_53miRNA, pt54_miRNA, by = "miRNA_ID")
pt1_55miRNA <- merge(pt1_54miRNA, pt55_miRNA, by = "miRNA_ID")
pt1_56miRNA <- merge(pt1_55miRNA, pt56_miRNA, by = "miRNA_ID")
pt1_57miRNA <- merge(pt1_56miRNA, pt57_miRNA, by = "miRNA_ID")
pt1_58miRNA <- merge(pt1_57miRNA, pt58_miRNA, by = "miRNA_ID")
pt1_59miRNA <- merge(pt1_58miRNA, pt59_miRNA, by = "miRNA_ID")
pt1_60miRNA <- merge(pt1_59miRNA, pt60_miRNA, by = "miRNA_ID")

pt1_61miRNA <- merge(pt1_60miRNA, pt61_miRNA, by = "miRNA_ID")
pt1_62miRNA <- merge(pt1_61miRNA, pt62_miRNA, by = "miRNA_ID")
pt1_63miRNA <- merge(pt1_62miRNA, pt63_miRNA, by = "miRNA_ID")
pt1_64miRNA <- merge(pt1_63miRNA, pt64_miRNA, by = "miRNA_ID")
pt1_65miRNA <- merge(pt1_64miRNA, pt65_miRNA, by = "miRNA_ID")
pt1_66miRNA <- merge(pt1_65miRNA, pt66_miRNA, by = "miRNA_ID")
pt1_67miRNA <- merge(pt1_66miRNA, pt67_miRNA, by = "miRNA_ID")
pt1_68miRNA <- merge(pt1_67miRNA, pt68_miRNA, by = "miRNA_ID")
pt1_69miRNA <- merge(pt1_68miRNA, pt69_miRNA, by = "miRNA_ID")
pt1_70miRNA <- merge(pt1_69miRNA, pt70_miRNA, by = "miRNA_ID")

pt1_71miRNA <- merge(pt1_70miRNA, pt71_miRNA, by = "miRNA_ID")
pt1_72miRNA <- merge(pt1_71miRNA, pt72_miRNA, by = "miRNA_ID")
pt1_73miRNA <- merge(pt1_72miRNA, pt73_miRNA, by = "miRNA_ID")
pt1_74miRNA <- merge(pt1_73miRNA, pt74_miRNA, by = "miRNA_ID")
pt1_75miRNA <- merge(pt1_74miRNA, pt75_miRNA, by = "miRNA_ID")
pt1_76miRNA <- merge(pt1_75miRNA, pt76_miRNA, by = "miRNA_ID")
pt1_77miRNA <- merge(pt1_76miRNA, pt77_miRNA, by = "miRNA_ID")
pt1_78miRNA <- merge(pt1_77miRNA, pt78_miRNA, by = "miRNA_ID")
pt1_79miRNA <- merge(pt1_78miRNA, pt79_miRNA, by = "miRNA_ID")
pt1_80miRNA <- merge(pt1_79miRNA, pt80_miRNA, by = "miRNA_ID")

pt1_81miRNA <- merge(pt1_80miRNA, pt81_miRNA, by = "miRNA_ID")
pt1_82miRNA <- merge(pt1_81miRNA, pt82_miRNA, by = "miRNA_ID")
pt1_83miRNA <- merge(pt1_82miRNA, pt83_miRNA, by = "miRNA_ID")
pt1_84miRNA <- merge(pt1_83miRNA, pt84_miRNA, by = "miRNA_ID")
pt1_85miRNA <- merge(pt1_84miRNA, pt85_miRNA, by = "miRNA_ID")
pt1_86miRNA <- merge(pt1_85miRNA, pt86_miRNA, by = "miRNA_ID")
pt1_87miRNA <- merge(pt1_86miRNA, pt87_miRNA, by = "miRNA_ID")
pt1_88miRNA <- merge(pt1_87miRNA, pt88_miRNA, by = "miRNA_ID")
pt1_89miRNA <- merge(pt1_88miRNA, pt89_miRNA, by = "miRNA_ID")
pt1_90miRNA <- merge(pt1_89miRNA, pt90_miRNA, by = "miRNA_ID")

pt1_91miRNA <- merge(pt1_90miRNA, pt91_miRNA, by = "miRNA_ID")
pt1_92miRNA <- merge(pt1_91miRNA, pt92_miRNA, by = "miRNA_ID")
pt1_93miRNA <- merge(pt1_92miRNA, pt93_miRNA, by = "miRNA_ID")
pt1_94miRNA <- merge(pt1_93miRNA, pt94_miRNA, by = "miRNA_ID")
pt1_95miRNA <- merge(pt1_94miRNA, pt95_miRNA, by = "miRNA_ID")
pt1_96miRNA <- merge(pt1_95miRNA, pt96_miRNA, by = "miRNA_ID")
pt1_97miRNA <- merge(pt1_96miRNA, pt97_miRNA, by = "miRNA_ID")
pt1_98miRNA <- merge(pt1_97miRNA, pt98_miRNA, by = "miRNA_ID")
pt1_99miRNA <- merge(pt1_98miRNA, pt99_miRNA, by = "miRNA_ID")
pt1_100miRNA <- merge(pt1_99miRNA, pt100_miRNA, by = "miRNA_ID")

pt1_101miRNA <- merge(pt1_100miRNA, pt101_miRNA, by = "miRNA_ID")
pt1_102miRNA <- merge(pt1_101miRNA, pt102_miRNA, by = "miRNA_ID")
pt1_103miRNA <- merge(pt1_102miRNA, pt103_miRNA, by = "miRNA_ID")
pt1_104miRNA <- merge(pt1_103miRNA, pt104_miRNA, by = "miRNA_ID")
pt1_105miRNA <- merge(pt1_104miRNA, pt105_miRNA, by = "miRNA_ID")
pt1_106miRNA <- merge(pt1_105miRNA, pt106_miRNA, by = "miRNA_ID")
pt1_107miRNA <- merge(pt1_106miRNA, pt107_miRNA, by = "miRNA_ID")
pt1_108miRNA <- merge(pt1_107miRNA, pt108_miRNA, by = "miRNA_ID")
pt1_109miRNA <- merge(pt1_108miRNA, pt109_miRNA, by = "miRNA_ID")
pt1_110miRNA <- merge(pt1_109miRNA, pt110_miRNA, by = "miRNA_ID")

pt1_111miRNA <- merge(pt1_110miRNA, pt111_miRNA, by = "miRNA_ID")
pt1_112miRNA <- merge(pt1_111miRNA, pt112_miRNA, by = "miRNA_ID")
pt1_113miRNA <- merge(pt1_112miRNA, pt113_miRNA, by = "miRNA_ID")
pt1_114miRNA <- merge(pt1_113miRNA, pt114_miRNA, by = "miRNA_ID")
pt1_115miRNA <- merge(pt1_114miRNA, pt115_miRNA, by = "miRNA_ID")
pt1_116miRNA <- merge(pt1_115miRNA, pt116_miRNA, by = "miRNA_ID")
pt1_117miRNA <- merge(pt1_116miRNA, pt117_miRNA, by = "miRNA_ID")
pt1_118miRNA <- merge(pt1_117miRNA, pt118_miRNA, by = "miRNA_ID")
pt1_119miRNA <- merge(pt1_118miRNA, pt119_miRNA, by = "miRNA_ID")
pt1_120miRNA <- merge(pt1_119miRNA, pt120_miRNA, by = "miRNA_ID")

pt1_121miRNA <- merge(pt1_120miRNA, pt121_miRNA, by = "miRNA_ID")
pt1_122miRNA <- merge(pt1_121miRNA, pt122_miRNA, by = "miRNA_ID")
pt1_123miRNA <- merge(pt1_122miRNA, pt123_miRNA, by = "miRNA_ID")
pt1_124miRNA <- merge(pt1_123miRNA, pt124_miRNA, by = "miRNA_ID")
pt1_125miRNA <- merge(pt1_124miRNA, pt125_miRNA, by = "miRNA_ID")
pt1_126miRNA <- merge(pt1_125miRNA, pt126_miRNA, by = "miRNA_ID")
pt1_127miRNA <- merge(pt1_126miRNA, pt127_miRNA, by = "miRNA_ID")
pt1_128miRNA <- merge(pt1_127miRNA, pt128_miRNA, by = "miRNA_ID")
pt1_129miRNA <- merge(pt1_128miRNA, pt129_miRNA, by = "miRNA_ID")
pt1_130miRNA <- merge(pt1_129miRNA, pt130_miRNA, by = "miRNA_ID")

pt1_131miRNA <- merge(pt1_130miRNA, pt131_miRNA, by = "miRNA_ID")
pt1_132miRNA <- merge(pt1_131miRNA, pt132_miRNA, by = "miRNA_ID")
pt1_133miRNA <- merge(pt1_132miRNA, pt133_miRNA, by = "miRNA_ID")
pt1_134miRNA <- merge(pt1_133miRNA, pt134_miRNA, by = "miRNA_ID")
pt1_135miRNA <- merge(pt1_134miRNA, pt135_miRNA, by = "miRNA_ID")
pt1_136miRNA <- merge(pt1_135miRNA, pt136_miRNA, by = "miRNA_ID")
pt1_137miRNA <- merge(pt1_136miRNA, pt137_miRNA, by = "miRNA_ID")
pt1_138miRNA <- merge(pt1_137miRNA, pt138_miRNA, by = "miRNA_ID")
pt1_139miRNA <- merge(pt1_138miRNA, pt139_miRNA, by = "miRNA_ID")
pt1_140miRNA <- merge(pt1_139miRNA, pt140_miRNA, by = "miRNA_ID")

pt1_141miRNA <- merge(pt1_140miRNA, pt141_miRNA, by = "miRNA_ID")
pt1_142miRNA <- merge(pt1_141miRNA, pt142_miRNA, by = "miRNA_ID")
pt1_143miRNA <- merge(pt1_142miRNA, pt143_miRNA, by = "miRNA_ID")
pt1_144miRNA <- merge(pt1_143miRNA, pt144_miRNA, by = "miRNA_ID")
pt1_145miRNA <- merge(pt1_144miRNA, pt145_miRNA, by = "miRNA_ID")
pt1_146miRNA <- merge(pt1_145miRNA, pt146_miRNA, by = "miRNA_ID")
pt1_147miRNA <- merge(pt1_146miRNA, pt147_miRNA, by = "miRNA_ID")
pt1_148miRNA <- merge(pt1_147miRNA, pt148_miRNA, by = "miRNA_ID")
pt1_149miRNA <- merge(pt1_148miRNA, pt149_miRNA, by = "miRNA_ID")

pt1_150miRNA <- merge(pt1_149miRNA, pt150_miRNA, by = "miRNA_ID")
pt1_151miRNA <- merge(pt1_150miRNA, pt151_miRNA, by = "miRNA_ID")
pt1_152miRNA <- merge(pt1_151miRNA, pt152_miRNA, by = "miRNA_ID")
pt1_153miRNA <- merge(pt1_152miRNA, pt153_miRNA, by = "miRNA_ID")
pt1_154miRNA <- merge(pt1_153miRNA, pt154_miRNA, by = "miRNA_ID")
pt1_155miRNA <- merge(pt1_154miRNA, pt155_miRNA, by = "miRNA_ID")
pt1_156miRNA <- merge(pt1_155miRNA, pt156_miRNA, by = "miRNA_ID")
pt1_157miRNA <- merge(pt1_156miRNA, pt157_miRNA, by = "miRNA_ID")
pt1_158miRNA <- merge(pt1_157miRNA, pt158_miRNA, by = "miRNA_ID")
pt1_159miRNA <- merge(pt1_158miRNA, pt159_miRNA, by = "miRNA_ID")

pt1_160miRNA <- merge(pt1_159miRNA, pt160_miRNA, by = "miRNA_ID")
pt1_161miRNA <- merge(pt1_160miRNA, pt161_miRNA, by = "miRNA_ID")
pt1_162miRNA <- merge(pt1_161miRNA, pt162_miRNA, by = "miRNA_ID")
pt1_163miRNA <- merge(pt1_162miRNA, pt163_miRNA, by = "miRNA_ID")
pt1_164miRNA <- merge(pt1_163miRNA, pt164_miRNA, by = "miRNA_ID")
pt1_165miRNA <- merge(pt1_164miRNA, pt165_miRNA, by = "miRNA_ID")
pt1_166miRNA <- merge(pt1_165miRNA, pt166_miRNA, by = "miRNA_ID")
pt1_167miRNA <- merge(pt1_166miRNA, pt167_miRNA, by = "miRNA_ID")
pt1_168miRNA <- merge(pt1_167miRNA, pt168_miRNA, by = "miRNA_ID")
pt1_169miRNA <- merge(pt1_168miRNA, pt169_miRNA, by = "miRNA_ID")

pt1_170miRNA <- merge(pt1_169miRNA, pt170_miRNA, by = "miRNA_ID")
pt1_171miRNA <- merge(pt1_170miRNA, pt171_miRNA, by = "miRNA_ID")
pt1_172miRNA <- merge(pt1_171miRNA, pt172_miRNA, by = "miRNA_ID")
pt1_173miRNA <- merge(pt1_172miRNA, pt173_miRNA, by = "miRNA_ID")
pt1_174miRNA <- merge(pt1_173miRNA, pt174_miRNA, by = "miRNA_ID")
pt1_175miRNA <- merge(pt1_174miRNA, pt174_miRNA, by = "miRNA_ID")
pt1_176miRNA <- merge(pt1_175miRNA, pt176_miRNA, by = "miRNA_ID")
pt1_177miRNA <- merge(pt1_176miRNA, pt177_miRNA, by = "miRNA_ID")
pt1_178miRNA <- merge(pt1_177miRNA, pt178_miRNA, by = "miRNA_ID")
pt1_179miRNA <- merge(pt1_178miRNA, pt179_miRNA, by = "miRNA_ID")

pt1_180miRNA <- merge(pt1_179miRNA, pt180_miRNA, by = "miRNA_ID")


# redo at the with all samples
TARGET_176_mirna <- pt1_176miRNA

TARGET_176_mirna_2 <- as.data.frame(TARGET_176_mirna %>%
  column_to_rownames("miRNA_ID") %>%
  t())

target_mirna_pilot <- TARGET_176_mirna_2 %>% rownames_to_column("sample")

write.csv(target_mirna_pilot, "~/Documents/R/pedcancer/target_data/TARGET_100/target_mirna_pilot.csv", row.names = FALSE)

mirna_samples <- pull(target_mirna_pilot %>% dplyr::select(sample))



```


```{r}

# TARGET methylation data

# need to find tss probes from illumina website for Infinium MethylationEPIC v1.0 B5 Manifest File (CSV Format)
# https://support.illumina.com/array/array_kits/infinium-methylationepic-beadchip-kit/downloads.html

illumina_manifest <- read.csv("~/Documents/R/pedcancer/target_data/infinium-methylationepic-v-1-0-b5-manifest-file.csv", header = TRUE)

# pull out all Probe.IDs labeled with 'tss'
# some of the results were also labeled with 'body'

illumina_tss <- pull(illumina_manifest %>%
  filter(grepl('TSS', UCSC_RefGene_Group)) %>%
  dplyr::select(Name))

# number of total cp probes that contain tss
illumina_tss <- illumina_manifest %>%
  filter(grepl('TSS', UCSC_RefGene_Group)) %>%
  dplyr::select(Name)

# number of total cp probes that contain tss but NOT Body, Exon, 5'UTR, 3'UTR...
illumina_tss_2 <- pull(illumina_manifest %>%
  filter(grepl('TSS', UCSC_RefGene_Group) &
           !grepl('Body', UCSC_RefGene_Group) &
           !grepl('Exon', UCSC_RefGene_Group) &
           !grepl('3.UTR', UCSC_RefGene_Group) &
           !grepl('5.UTR', UCSC_RefGene_Group)) %>%
  dplyr::select(Name))

illumina_tss_25k_id <- pt1_cpg %>%
  filter(Probe.Name %in% illumina_tss_2)


# some files were entered weirdly in TARGET. choosing files with only this 'standard' organization

pt1_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PADZKD.csv", header = TRUE)
pt1_cpg$sample <- str_replace_all(pt1_cpg$Sample.ID, "TARGET-20-PADZKD-09A-01D", "PADZKD")
pt1_cpg <- pt1_cpg %>% dplyr::select(-Sample.ID)
pt1_tss <- pt1_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PADZKD = median(PADZKD))

pt2_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PADZYC.csv", header = TRUE)
pt2_cpg$sample <- str_replace_all(pt2_cpg$Sample.ID, "TARGET-20-PADZYC-09A-01D", "PADZYC")
pt2_cpg <- pt2_cpg %>% dplyr::select(-Sample.ID)
pt2_tss <- pt2_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PADZYC = median(PADZYC))

pt3_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAEABM.csv", header = TRUE)
pt3_cpg$sample <- str_replace_all(pt3_cpg$Sample.ID, "TARGET-20-PAEABM-09A-01D", "PAEABM")
pt3_cpg <- pt3_cpg %>% dplyr::select(-Sample.ID)
pt3_tss <- pt3_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEABM = median(PAEABM))

pt4_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAEAKL.csv", header = TRUE)
pt4_cpg$sample <- str_replace_all(pt4_cpg$Sample.ID, "TARGET-20-PAEAKL-09A-01D", "PAEAKL")
pt4_cpg <- pt4_cpg %>% dplyr::select(-Sample.ID)
pt4_tss <- pt4_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEAKL = median(PAEAKL))

pt5_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAECBZ.csv", header = TRUE)
pt5_cpg$sample <- str_replace_all(pt5_cpg$Sample.ID, "TARGET-20-PAECBZ-09A-01D", "PAECBZ")
pt5_cpg <- pt5_cpg %>% dplyr::select(-Sample.ID)
pt5_tss <- pt5_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAECBZ = median(PAECBZ))

pt6_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAECRF.csv", header = TRUE)
pt6_cpg$sample <- str_replace_all(pt6_cpg$Sample.ID, "TARGET-20-PAECRF-09A-01D", "PAECRF")
pt6_cpg <- pt6_cpg %>% dplyr::select(-Sample.ID)
pt6_tss <- pt6_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAECRF = median(PAECRF))

pt7_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAEEYP.csv", header = TRUE)
pt7_cpg$sample <- str_replace_all(pt7_cpg$Sample.ID, "TARGET-20-PAEEYP-09A-01D", "PAEEYP")
pt7_cpg <- pt7_cpg %>% dplyr::select(-Sample.ID)
pt7_tss <- pt7_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEEYP = median(PAEEYP))

pt8_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAEFHC.csv", header = TRUE)
pt8_cpg$sample <- str_replace_all(pt8_cpg$Sample.ID, "TARGET-20-PAEFHC-09A-01D", "PAEFHC")
pt8_cpg <- pt8_cpg %>% dplyr::select(-Sample.ID)
pt8_tss <- pt8_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEFHC = median(PAEFHC))

pt9_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAEGRE.csv", header = TRUE)
pt9_cpg$sample <- str_replace_all(pt9_cpg$Sample.ID, "TARGET-20-PAEGRE-09A-01D", "PAEGRE")
pt9_cpg <- pt9_cpg %>% dplyr::select(-Sample.ID)
pt9_tss <- pt9_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEGRE = median(PAEGRE))

pt10_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAEGRI.csv", header = TRUE)
pt10_cpg$sample <- str_replace_all(pt10_cpg$Sample.ID, "TARGET-20-PAEGRI-09A-01D", "PAEGRI")
pt10_cpg <- pt10_cpg %>% dplyr::select(-Sample.ID)
pt10_tss <- pt10_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEGRI = median(PAEGRI))

pt11_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAERAH.csv", header = TRUE)
pt11_cpg$sample <- str_replace_all(pt11_cpg$Sample.ID, "TARGET-20-PAERAH-09A-01D", "PAERAH")
pt11_cpg <- pt11_cpg %>% dplyr::select(-Sample.ID)
pt11_tss <- pt11_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAERAH = median(PAERAH))

pt12_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAERXF.csv", header = TRUE)
pt12_cpg$sample <- str_replace_all(pt12_cpg$Sample.ID, "TARGET-20-PAERXF-09A-01D", "PAERXF")
pt12_cpg <- pt12_cpg %>% dplyr::select(-Sample.ID)
pt12_tss <- pt12_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAERXF = median(PAERXF))

pt13_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKHLK.csv", header = TRUE)
pt13_cpg$sample <- str_replace_all(pt13_cpg$Sample.ID, "TARGET-20-PAKHLK-09A-01D", "PAKHLK")
pt13_cpg <- pt13_cpg %>% dplyr::select(-Sample.ID)
pt13_tss <- pt13_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKHLK = median(PAKHLK))

pt14_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKHZW.csv", header = TRUE)
pt14_cpg$sample <- str_replace_all(pt14_cpg$Sample.ID, "TARGET-20-PAKHZW-09A-01D", "PAKHZW")
pt14_cpg <- pt14_cpg %>% dplyr::select(-Sample.ID)
pt14_tss <- pt14_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKHZW = median(PAKHZW))

pt15_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKIYW.csv", header = TRUE)
pt15_cpg$sample <- str_replace_all(pt15_cpg$Sample.ID, "TARGET-20-PAKIYW-09A-01D", "PAKIYW")
pt15_cpg <- pt15_cpg %>% dplyr::select(-Sample.ID)
pt15_tss <- pt15_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKIYW = median(PAKIYW))

pt16_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKKBK.csv", header = TRUE)
pt16_cpg$sample <- str_replace_all(pt16_cpg$Sample.ID, "TARGET-20-PAKKBK-09A-01D", "PAKKBK")
pt16_cpg <- pt16_cpg %>% dplyr::select(-Sample.ID)
pt16_tss <- pt16_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKKBK = median(PAKKBK))

pt17_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKKIM.csv", header = TRUE)
pt17_cpg$sample <- str_replace_all(pt17_cpg$Sample.ID, "TARGET-20-PAKKIM-09A-01D", "PAKKIM")
pt17_cpg <- pt17_cpg %>% dplyr::select(-Sample.ID)
pt17_tss <- pt17_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKKIM = median(PAKKIM))

pt18_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKLPD.csv", header = TRUE)
pt18_cpg$sample <- str_replace_all(pt18_cpg$Sample.ID, "TARGET-20-PAKLPD-09A-01D", "PAKLPD")
pt18_cpg <- pt18_cpg %>% dplyr::select(-Sample.ID)
pt18_tss <- pt18_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKLPD = median(PAKLPD))

pt19_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKRZG.csv", header = TRUE)
pt19_cpg$sample <- str_replace_all(pt19_cpg$Sample.ID, "TARGET-20-PAKRZG-09A-01D", "PAKRZG")
pt19_cpg <- pt19_cpg %>% dplyr::select(-Sample.ID)
pt19_tss <- pt19_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKRZG = median(PAKRZG))

pt20_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKXJR.csv", header = TRUE)
pt20_cpg$sample <- str_replace_all(pt20_cpg$Sample.ID, "TARGET-20-PAKXJR-09A-01D", "PAKXJR")
pt20_cpg <- pt20_cpg %>% dplyr::select(-Sample.ID)
pt20_tss <- pt20_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKXJR = median(PAKXJR))

pt21_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAKXVC.csv", header = TRUE)
pt21_cpg$sample <- str_replace_all(pt21_cpg$Sample.ID, "TARGET-20-PAKXVC-09A-01D", "PAKXVC")
pt21_cpg <- pt21_cpg %>% dplyr::select(-Sample.ID)
pt21_tss <- pt21_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKXVC = median(PAKXVC))

#pt21_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/mirna/PAKXVC.csv", header = TRUE)
#pt21_cpg$sample <- str_replace_all(pt21_cpg$Sample.ID, "TARGET-20-PAKXVC-09A-01D", "PAKXVC")
#pt21_cpg <- pt21_cpg %>% dplyr::select(-Sample.ID)

pt22_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PALGKX.csv", header = TRUE)
pt22_cpg$sample <- str_replace_all(pt22_cpg$Sample.ID, "TARGET-20-PALGKX-09A-01D", "PALGKX")
pt22_cpg <- pt22_cpg %>% dplyr::select(-Sample.ID)
pt22_tss <- pt22_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PALGKX = median(PALGKX))

pt23_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAMVKZ.csv", header = TRUE)
pt23_cpg$sample <- str_replace_all(pt23_cpg$Sample.ID, "TARGET-20-PAMVKZ-09A-01D", "PAMVKZ")
pt23_cpg <- pt23_cpg %>% dplyr::select(-Sample.ID)
pt23_tss <- pt23_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAMVKZ = median(PAMVKZ))

pt24_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAMVVP.csv", header = TRUE)
pt24_cpg$sample <- str_replace_all(pt24_cpg$Sample.ID, "TARGET-20-PAMVVP-09A-01D", "PAMVVP")
pt24_cpg <- pt24_cpg %>% dplyr::select(-Sample.ID)
pt24_tss <- pt24_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAMVVP = median(PAMVVP))

pt25_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAMYAS.csv", header = TRUE)
pt25_cpg$sample <- str_replace_all(pt25_cpg$Sample.ID, "TARGET-20-PAMYAS-09A-01D", "PAMYAS")
pt25_cpg <- pt25_cpg %>% dplyr::select(-Sample.ID)
pt25_tss <- pt25_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAMYAS = median(PAMYAS))

pt26_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAMYGX.csv", header = TRUE)
pt26_cpg$sample <- str_replace_all(pt26_cpg$Sample.ID, "TARGET-20-PAMYGX-09A-03D", "PAMYGX")
pt26_cpg <- pt26_cpg %>% dplyr::select(-Sample.ID)
pt26_tss <- pt26_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAMYGX = median(PAMYGX))

pt27_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAMYVU.csv", header = TRUE)
pt27_cpg$sample <- str_replace_all(pt27_cpg$Sample.ID, "TARGET-20-PAMYVU-09A-03D", "PAMYVU")
pt27_cpg <- pt27_cpg %>% dplyr::select(-Sample.ID)
pt27_tss <- pt27_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAMYVU = median(PAMYVU))

pt28_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANAEV.csv", header = TRUE)
pt28_cpg$sample <- str_replace_all(pt28_cpg$Sample.ID, "TARGET-20-PANAEV-09A-01D", "PANAEV")
pt28_cpg <- pt28_cpg %>% dplyr::select(-Sample.ID)
pt28_tss <- pt28_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANAEV = median(PANAEV))

pt29_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANASY.csv", header = TRUE)
pt29_cpg$sample <- str_replace_all(pt29_cpg$Sample.ID, "TARGET-20-PANASY-09A-01D", "PANASY")
pt29_cpg <- pt29_cpg %>% dplyr::select(-Sample.ID)
pt29_tss <- pt29_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANASY = median(PANASY))

pt30_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANBYS.csv", header = TRUE)
pt30_cpg$sample <- str_replace_all(pt30_cpg$Sample.ID, "TARGET-20-PANBYS-09A-01D", "PANBYS")
pt30_cpg <- pt30_cpg %>% dplyr::select(-Sample.ID)
pt30_tss <- pt30_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANBYS = median(PANBYS))

pt31_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANBZH.csv", header = TRUE)
pt31_cpg$sample <- str_replace_all(pt31_cpg$Sample.ID, "TARGET-20-PANBZH-09A-01D", "PANBZH")
pt31_cpg <- pt31_cpg %>% dplyr::select(-Sample.ID)
pt31_tss <- pt31_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANBZH = median(PANBZH))

pt32_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANCSC.csv", header = TRUE)
pt32_cpg$sample <- str_replace_all(pt32_cpg$Sample.ID, "TARGET-20-PANCSC-09A-01D", "PANCSC")
pt32_cpg <- pt32_cpg %>% dplyr::select(-Sample.ID)
pt32_tss <- pt32_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANCSC = median(PANCSC))

pt33_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANDDC.csv", header = TRUE)
pt33_cpg$sample <- str_replace_all(pt33_cpg$Sample.ID, "TARGET-20-PANDDC-09A-01D", "PANDDC")
pt33_cpg <- pt33_cpg %>% dplyr::select(-Sample.ID)
pt33_tss <- pt33_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANDDC = median(PANDDC))

pt34_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANFMG.csv", header = TRUE)
pt34_cpg$sample <- str_replace_all(pt34_cpg$Sample.ID, "TARGET-20-PANFMG-09A-01D", "PANFMG")
pt34_cpg <- pt34_cpg %>% dplyr::select(-Sample.ID)
pt34_tss <- pt34_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANFMG = median(PANFMG))

pt35_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANGDN.csv", header = TRUE)
pt35_cpg$sample <- str_replace_all(pt35_cpg$Sample.ID, "TARGET-20-PANGDN-09A-01D", "PANGDN")
pt35_cpg <- pt35_cpg %>% dplyr::select(-Sample.ID)
pt35_tss <- pt35_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANGDN = median(PANGDN))

pt36_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANGJY.csv", header = TRUE)
pt36_cpg$sample <- str_replace_all(pt36_cpg$Sample.ID, "TARGET-20-PANGJY-09A-01D", "PANGJY")
pt36_cpg <- pt36_cpg %>% dplyr::select(-Sample.ID)
pt36_tss <- pt36_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANGJY = median(PANGJY))

pt37_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANHWD.csv", header = TRUE)
pt37_cpg$sample <- str_replace_all(pt37_cpg$Sample.ID, "TARGET-20-PANHWD-09A-01D", "PANHWD")
pt37_cpg <- pt37_cpg %>% dplyr::select(-Sample.ID)
pt37_tss <- pt37_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANHWD = median(PANHWD))

pt38_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANHYK.csv", header = TRUE)
pt38_cpg$sample <- str_replace_all(pt38_cpg$Sample.ID, "TARGET-20-PANHYK-09A-02D", "PANHYK")
pt38_cpg <- pt38_cpg %>% dplyr::select(-Sample.ID)
pt38_tss <- pt38_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANHYK = median(PANHYK))

pt39_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANILV.csv", header = TRUE)
pt39_cpg$sample <- str_replace_all(pt39_cpg$Sample.ID, "TARGET-20-PANILV-09A-01D", "PANILV")
pt39_cpg <- pt39_cpg %>% dplyr::select(-Sample.ID)
pt39_tss <- pt39_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANILV = median(PANILV))

pt40_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANINI.csv", header = TRUE)
pt40_cpg$sample <- str_replace_all(pt40_cpg$Sample.ID, "TARGET-20-PANINI-09A-01D", "PANINI")
pt40_cpg <- pt40_cpg %>% dplyr::select(-Sample.ID)
pt40_tss <- pt40_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANINI = median(PANINI))

pt41_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANJGR.csv", header = TRUE)
pt41_cpg$sample <- str_replace_all(pt41_cpg$Sample.ID, "TARGET-20-PANJGR-09A-02D", "PANJGR")
pt41_cpg <- pt41_cpg %>% dplyr::select(-Sample.ID)
pt41_tss <- pt41_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANJGR = median(PANJGR))

pt42_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANKFG.csv", header = TRUE)
pt42_cpg$sample <- str_replace_all(pt42_cpg$Sample.ID, "TARGET-20-PANKFG-09A-01D", "PANKFG")
pt42_cpg <- pt42_cpg %>% dplyr::select(-Sample.ID)
pt42_tss <- pt42_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANKFG = median(PANKFG))

pt43_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANKFZ.csv", header = TRUE)
pt43_cpg$sample <- str_replace_all(pt43_cpg$Sample.ID, "TARGET-20-PANKFZ-09A-01D", "PANKFZ")
pt43_cpg <- pt43_cpg %>% dplyr::select(-Sample.ID)
pt43_tss <- pt43_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANKFZ = median(PANKFZ))

pt44_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANKKE.csv", header = TRUE)
pt44_cpg$sample <- str_replace_all(pt44_cpg$Sample.ID, "TARGET-20-PANKKE-09A-01D", "PANKKE")
pt44_cpg <- pt44_cpg %>% dplyr::select(-Sample.ID)
pt44_tss <- pt44_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANKKE = median(PANKKE))

pt45_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANKNB.csv", header = TRUE)
pt45_cpg$sample <- str_replace_all(pt45_cpg$Sample.ID, "TARGET-20-PANKNB-09A-01D", "PANKNB")
pt45_cpg <- pt45_cpg %>% dplyr::select(-Sample.ID)
pt45_tss <- pt45_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANKNB = median(PANKNB))

pt46_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANLIZ.csv", header = TRUE)
pt46_cpg$sample <- str_replace_all(pt46_cpg$Sample.ID, "TARGET-20-PANLIZ-09A-01D", "PANLIZ")
pt46_cpg <- pt46_cpg %>% dplyr::select(-Sample.ID)
pt46_tss <- pt46_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLIZ = median(PANLIZ))

pt47_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANLJK.csv", header = TRUE)
pt47_cpg$sample <- str_replace_all(pt47_cpg$Sample.ID, "TARGET-20-PANLJK-09A-01D", "PANLJK")
pt47_cpg <- pt47_cpg %>% dplyr::select(-Sample.ID)
pt47_tss <- pt47_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLJK = median(PANLJK))

pt48_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANLJN.csv", header = TRUE)
pt48_cpg$sample <- str_replace_all(pt48_cpg$Sample.ID, "TARGET-20-PANLJN-09A-01D", "PANLJN")
pt48_cpg <- pt48_cpg %>% dplyr::select(-Sample.ID)
pt48_tss <- pt48_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLJN = median(PANLJN))

pt49_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANLKB.csv", header = TRUE)
pt49_cpg$sample <- str_replace_all(pt49_cpg$Sample.ID, "TARGET-20-PANLKB-09A-01D", "PANLKB")
pt49_cpg <- pt49_cpg %>% dplyr::select(-Sample.ID)
pt49_tss <- pt49_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLKB = median(PANLKB))

pt50_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANLXK.csv", header = TRUE)
pt50_cpg$sample <- str_replace_all(pt50_cpg$Sample.ID, "TARGET-20-PANLXK-09A-01D", "PANLXK")
pt50_cpg <- pt50_cpg %>% dplyr::select(-Sample.ID)
pt50_tss <- pt50_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLXK = median(PANLXK))


# alternate format download -- from Pol -- for the weirdly formatted samples
#path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PADZCG-09A-01D_450k_L3.txt"
#data <- vroom::vroom(path, delim = "\t", skip = 1) 
#colnames(data) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
#data$sample <- ("PADZCG")

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PABLDZ-09A-03D_450k_L3.txt"
pt51_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt51_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt51_cpg$sample <- ("PABLDZ")
pt51_tss <- pt51_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PABLDZ, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PABLDZ = median(PABLDZ))


path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PADZCG-09A-01D_450k_L3.txt"
pt52_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt52_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt52_cpg$sample <- ("PADZCG")
pt52_tss <- pt52_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PADZCG, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PADZCG = median(PADZCG))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAECCE-09A-01D_450k_L3.txt"
pt53_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt53_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt53_cpg$sample <- ("PAECCE")
pt53_tss <- pt53_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAECCE, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAECCE = median(PAECCE))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAEFGR-09A-02D_450k_L3.txt"
pt54_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt54_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt54_cpg$sample <- ("PAEFGR")
pt54_tss <- pt54_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAEFGR, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEFGR = median(PAEFGR))


path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAEIKD-09A-01D_450k_L3.txt"
pt55_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt55_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt55_cpg$sample <- ("PAEIKD")
pt55_tss <- pt55_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAEIKD, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAEIKD = median(PAEIKD))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAKERZ-09A-01D_450k_L3.txt"
pt56_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt56_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt56_cpg$sample <- ("PAKERZ")
pt56_tss <- pt56_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAKERZ, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKERZ = median(PAKERZ))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAKIWK-09A-01D_450k_L3.txt"
pt57_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt57_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt57_cpg$sample <- ("PAKIWK")
pt57_tss <- pt57_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAKIWK, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKIWK = median(PAKIWK))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAKTCX-09A-02D_450k_L3.txt"
pt58_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt58_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt58_cpg$sample <- ("PAKTCX")
pt58_tss <- pt58_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAKTCX, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKTCX = median(PAKTCX))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAKVGI-09A-01D_450k_L3.txt"
pt59_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt59_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt59_cpg$sample <- ("PAKVGI")
pt59_tss <- pt59_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAKVGI, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAKVGI = median(PAKVGI))


path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAMYJX-09A-03D_450k_L3.txt"
pt60_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt60_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt60_cpg$sample <- ("PAMYJX")
pt60_tss <- pt60_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAMYJX, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAMYJX = median(PAMYJX))


path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANDER-09A-01D_450k_L3.txt"
pt61_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt61_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt61_cpg$sample <- ("PANDER")
pt61_tss <- pt61_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANDER, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANDER = median(PANDER))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANDIX-09A-02D_450k_L3.txt"
pt62_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt62_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt62_cpg$sample <- ("PANDIX")
pt62_tss <- pt62_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANDIX, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANDIX = median(PANDIX))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANGTF-09A-02D_450k_L3.txt"
pt63_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt63_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt63_cpg$sample <- ("PANGTF")
pt63_tss <- pt63_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANGTF, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANGTF = median(PANGTF))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANLIR-09A-02D_450k_L3.txt"
pt64_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt64_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt64_cpg$sample <- ("PANLIR")
pt64_tss <- pt64_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANLIR, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLIR = median(PANLIR))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANLLX-09A-01D_450k_L3.txt"
pt65_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt65_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt65_cpg$sample <- ("PANLLX")
pt65_tss <- pt65_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANLLX, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLLX = median(PANLLX))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANLRE-09A-01D_450k_L3.txt"
pt66_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt66_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt66_cpg$sample <- ("PANLRE")
pt66_tss <- pt66_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANLRE, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLRE = median(PANLRE))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANLXM-09A-03D_450k_L3.txt"
pt67_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt67_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt67_cpg$sample <- ("PANLXM")
pt67_tss <- pt67_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANLXM, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANLXM = median(PANLXM))

pt68_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANNHB.csv", header = TRUE)
pt68_cpg$sample <- str_replace_all(pt68_cpg$Sample.ID, "TARGET-20-PANNHB-09A-01D", "PANNHB")
pt68_cpg <- pt68_cpg %>% dplyr::select(-Sample.ID)
pt68_tss <- pt68_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANNHB = median(PANNHB))

pt69_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANNJI.csv", header = TRUE)
pt69_cpg$sample <- str_replace_all(pt69_cpg$Sample.ID, "TARGET-20-PANNJI-09A-02D", "PANNJI")
pt69_cpg <- pt69_cpg %>% dplyr::select(-Sample.ID)
pt69_tss <- pt69_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANNJI = median(PANNJI))

pt70_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANPTM.csv", header = TRUE)
pt70_cpg$sample <- str_replace_all(pt70_cpg$Sample.ID, "TARGET-20-PANPTM-09A-01D", "PANPTM")
pt70_cpg <- pt70_cpg %>% dplyr::select(-Sample.ID)
pt70_tss <- pt70_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANPTM = median(PANPTM))

pt71_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANSBH.csv", header = TRUE)
pt71_cpg$sample <- str_replace_all(pt71_cpg$Sample.ID, "TARGET-20-PANSBH-09A-02D", "PANSBH")
pt71_cpg <- pt71_cpg %>% dplyr::select(-Sample.ID)
pt71_tss <- pt71_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANSBH = median(PANSBH))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PANSJB-09A-02D_450k_L3.txt"
pt72_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt72_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt72_cpg$sample <- ("PANSJB")
pt72_tss <- pt72_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PANSJB, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PANSJB = median(PANSJB))

pt73_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANTNA.csv", header = TRUE)
pt73_cpg$sample <- str_replace_all(pt73_cpg$Sample.ID, "TARGET-20-PANTNA-09A-01D", "PANTNA")
pt73_cpg <- pt73_cpg %>% dplyr::select(-Sample.ID)
pt73_tss <- pt73_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANTNA = median(PANTNA))

pt74_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANTPW.csv", header = TRUE)
pt74_cpg$sample <- str_replace_all(pt74_cpg$Sample.ID, "TARGET-20-PANTPW-09A-01D", "PANTPW")
pt74_cpg <- pt74_cpg %>% dplyr::select(-Sample.ID)
pt74_tss <- pt74_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANTPW = median(PANTPW))

pt75_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANUTB.csv", header = TRUE)
pt75_cpg$sample <- str_replace_all(pt75_cpg$Sample.ID, "TARGET-20-PANUTB-09A-01D", "PANUTB")
pt75_cpg <- pt75_cpg %>% dplyr::select(-Sample.ID)
pt75_tss <- pt75_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANUTB = median(PANUTB))

pt76_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANUUA.csv", header = TRUE)
pt76_cpg$sample <- str_replace_all(pt76_cpg$Sample.ID, "TARGET-20-PANUUA-09A-01D", "PANUUA")
pt76_cpg <- pt76_cpg %>% dplyr::select(-Sample.ID)
pt76_tss <- pt76_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANUUA = median(PANUUA))

pt77_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANVBP.csv", header = TRUE)
pt77_cpg$sample <- str_replace_all(pt77_cpg$Sample.ID, "TARGET-20-PANVBP-09A-01D", "PANVBP")
pt77_cpg <- pt77_cpg %>% dplyr::select(-Sample.ID)
pt77_tss <- pt77_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANVBP = median(PANVBP))

pt78_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANVGE.csv", header = TRUE)
pt78_cpg$sample <- str_replace_all(pt78_cpg$Sample.ID, "TARGET-20-PANVGE-09A-01D", "PANVGE")
pt78_cpg <- pt78_cpg %>% dplyr::select(-Sample.ID)
pt78_tss <- pt78_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANVGE = median(PANVGE))

pt79_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANWHP.csv", header = TRUE)
pt79_cpg$sample <- str_replace_all(pt79_cpg$Sample.ID, "TARGET-20-PANWHP-09A-01D", "PANWHP")
pt79_cpg <- pt79_cpg %>% dplyr::select(-Sample.ID)
pt79_tss <- pt79_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANWHP = median(PANWHP))

pt80_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANYGP.csv", header = TRUE)
pt80_cpg$sample <- str_replace_all(pt80_cpg$Sample.ID, "TARGET-20-PANYGP-09A-01D", "PANYGP")
pt80_cpg <- pt80_cpg %>% dplyr::select(-Sample.ID)
pt80_tss <- pt80_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANYGP = median(PANYGP))

pt81_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANZKA.csv", header = TRUE)
pt81_cpg$sample <- str_replace_all(pt81_cpg$Sample.ID, "TARGET-20-PANZKA-09A-01D", "PANZKA")
pt81_cpg <- pt81_cpg %>% dplyr::select(-Sample.ID)
pt81_tss <- pt81_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANZKA = median(PANZKA))

pt82_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PANZWB.csv", header = TRUE)
pt82_cpg$sample <- str_replace_all(pt82_cpg$Sample.ID, "TARGET-20-PANZWB-09A-01D", "PANZWB")
pt82_cpg <- pt82_cpg %>% dplyr::select(-Sample.ID)
pt82_tss <- pt82_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PANZWB = median(PANZWB))

pt83_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPAWN.csv", header = TRUE)
pt83_cpg$sample <- str_replace_all(pt83_cpg$Sample.ID, "TARGET-20-PAPAWN-09A-01D", "PAPAWN")
pt83_cpg <- pt83_cpg %>% dplyr::select(-Sample.ID)
pt83_tss <- pt83_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPAWN = median(PAPAWN))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAPBEJ-09A-03D_450k_L3.txt"
pt84_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt84_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt84_cpg$sample <- ("PAPBEJ")
pt84_tss <- pt84_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAPBEJ, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPBEJ = median(PAPBEJ))

pt85_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPVCN.csv", header = TRUE)
pt85_cpg$sample <- str_replace_all(pt85_cpg$Sample.ID, "TARGET-20-PAPVCN-09A-01D", "PAPVCN")
pt85_cpg <- pt85_cpg %>% dplyr::select(-Sample.ID)
pt85_tss <- pt85_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPVCN = median(PAPVCN))

pt86_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPVGE.csv", header = TRUE)
pt86_cpg$sample <- str_replace_all(pt86_cpg$Sample.ID, "TARGET-20-PAPVGE-09A-02D", "PAPVGE")
pt86_cpg <- pt86_cpg %>% dplyr::select(-Sample.ID)
pt86_tss <- pt86_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPVGE = median(PAPVGE))

pt87_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPVZK.csv", header = TRUE)
pt87_cpg$sample <- str_replace_all(pt87_cpg$Sample.ID, "TARGET-20-PAPVZK-09A-01D", "PAPVZK")
pt87_cpg <- pt87_cpg %>% dplyr::select(-Sample.ID)
pt87_tss <- pt87_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPVZK = median(PAPVZK))


pt88_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPWHS.csv", header = TRUE)
pt88_cpg$sample <- str_replace_all(pt88_cpg$Sample.ID, "TARGET-20-PAPWHS-09A-02D", "PAPWHS")
pt88_cpg <- pt88_cpg %>% dplyr::select(-Sample.ID)
pt88_tss <- pt88_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPWHS = median(PAPWHS))

pt89_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPWIU.csv", header = TRUE)
pt89_cpg$sample <- str_replace_all(pt89_cpg$Sample.ID, "TARGET-20-PAPWIU-09A-01D", "PAPWIU")
pt89_cpg <- pt89_cpg %>% dplyr::select(-Sample.ID)
pt89_tss <- pt89_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPWIU = median(PAPWIU))

pt90_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPWYK.csv", header = TRUE)
pt90_cpg$sample <- str_replace_all(pt90_cpg$Sample.ID, "TARGET-20-PAPWYK-09A-01D", "PAPWYK")
pt90_cpg <- pt90_cpg %>% dplyr::select(-Sample.ID)
pt90_tss <- pt90_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPWYK = median(PAPWYK))

#pt91_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPWYK.csv", header = TRUE)
#pt91_cpg$sample <- str_replace_all(pt91_cpg$Sample.ID, "TARGET-20-PAPWYK-09A-01D", "PAPWYK")
#pt91_cpg <- pt91_cpg %>% dplyr::select(-Sample.ID)
#pt91_tss <- pt91_cpg %>%
 # filter(Probe.Name %in% illumina_tss) %>%
#  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
#  pivot_wider(names_from = sample,
 #             values_from = AVG_Beta) %>% unnest() 

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PAPXRJ-09A-03D_450k_L3.txt"
pt92_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt92_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt92_cpg$sample <- ("PAPXRJ")
pt92_tss <- pt92_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PAPXRJ, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPXRJ = median(PAPXRJ))

pt93_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPXWI.csv", header = TRUE)
pt93_cpg$sample <- str_replace_all(pt93_cpg$Sample.ID, "TARGET-20-PAPXWI-09A-01D", "PAPXWI")
pt93_cpg <- pt93_cpg %>% dplyr::select(-Sample.ID)
pt93_tss <- pt93_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPXWI = median(PAPXWI))


pt94_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPZIZ.csv", header = TRUE)
pt94_cpg$sample <- str_replace_all(pt94_cpg$Sample.ID, "TARGET-20-PAPZIZ-09A-01D", "PAPZIZ")
pt94_cpg <- pt94_cpg %>% dplyr::select(-Sample.ID)
pt94_tss <- pt94_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest()  %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPZIZ = median(PAPZIZ))

pt95_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PAPZLT.csv", header = TRUE)
pt95_cpg$sample <- str_replace_all(pt95_cpg$Sample.ID, "TARGET-20-PAPZLT-09A-01D", "PAPZLT")
pt95_cpg <- pt95_cpg %>% dplyr::select(-Sample.ID)
pt95_tss <- pt95_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PAPZLT = median(PAPZLT))

pt96_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARAJX.csv", header = TRUE)
pt96_cpg$sample <- str_replace_all(pt96_cpg$Sample.ID, "TARGET-20-PARAJX-09A-01D", "PARAJX")
pt96_cpg <- pt96_cpg %>% dplyr::select(-Sample.ID)
pt96_tss <- pt96_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARAJX = median(PARAJX))

pt97_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARANT.csv", header = TRUE)
pt97_cpg$sample <- str_replace_all(pt97_cpg$Sample.ID, "TARGET-20-PARANT-09A-01D", "PARANT")
pt97_cpg <- pt97_cpg %>% dplyr::select(-Sample.ID)
pt97_tss <- pt97_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARANT = median(PARANT))

pt98_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARBFJ.csv", header = TRUE)
pt98_cpg$sample <- str_replace_all(pt98_cpg$Sample.ID, "TARGET-20-PARBFJ-09A-01D", "PARBFJ")
pt98_cpg <- pt98_cpg %>% dplyr::select(-Sample.ID)
pt98_tss <- pt98_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARBFJ = median(PARBFJ))

pt99_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARBIU.csv", header = TRUE)
pt99_cpg$sample <- str_replace_all(pt99_cpg$Sample.ID, "TARGET-20-PARBIU-09A-01D", "PARBIU")
pt99_cpg <- pt99_cpg %>% dplyr::select(-Sample.ID)
pt99_tss <- pt99_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARBIU = median(PARBIU))

pt100_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARBRA.csv", header = TRUE)
pt100_cpg$sample <- str_replace_all(pt100_cpg$Sample.ID, "TARGET-20-PARBRA-09A-01D", "PARBRA")
pt100_cpg <- pt100_cpg %>% dplyr::select(-Sample.ID)
pt100_tss <- pt100_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARBRA = median(PARBRA))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-21-PARBTV-09A-03D_450k_L3.txt"
pt101_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt101_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt101_cpg$sample <- ("PARBTV")
pt101_tss <- pt101_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARBTV, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARBTV = median(PARBTV))

pt102_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARBXE.csv", header = TRUE)
pt102_cpg$sample <- str_replace_all(pt102_cpg$Sample.ID, "TARGET-20-PARBXE-09A-01D", "PARBXE")
pt102_cpg <- pt102_cpg %>% dplyr::select(-Sample.ID)
pt102_tss <- pt102_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARBXE = median(PARBXE))

pt103_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARCUK.csv", header = TRUE)
pt103_cpg$sample <- str_replace_all(pt103_cpg$Sample.ID, "TARGET-20-PARCUK-09A-02D", "PARCUK")
pt103_cpg <- pt103_cpg %>% dplyr::select(-Sample.ID)
pt103_tss <- pt103_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARCUK = median(PARCUK))

pt104_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARCZL.csv", header = TRUE)
pt104_cpg$sample <- str_replace_all(pt104_cpg$Sample.ID, "TARGET-20-PARCZL-09A-01D", "PARCZL")
pt104_cpg <- pt104_cpg %>% dplyr::select(-Sample.ID)
pt104_tss <- pt104_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARCZL = median(PARCZL))

pt105_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARDDY.csv", header = TRUE)
pt105_cpg$sample <- str_replace_all(pt105_cpg$Sample.ID, "TARGET-20-PARDDY-09A-01D", "PARDDY")
pt105_cpg <- pt105_cpg %>% dplyr::select(-Sample.ID)
pt105_tss <- pt105_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARDDY = median(PARDDY))

pt106_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARDMG.csv", header = TRUE)
pt106_cpg$sample <- str_replace_all(pt106_cpg$Sample.ID, "TARGET-20-PARDMG-09A-01D", "PARDMG")
pt106_cpg <- pt106_cpg %>% dplyr::select(-Sample.ID)
pt106_tss <- pt106_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARDMG = median(PARDMG))

pt107_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARENB.csv", header = TRUE)
pt107_cpg$sample <- str_replace_all(pt107_cpg$Sample.ID, "TARGET-20-PARENB-09A-01D", "PARENB")
pt107_cpg <- pt107_cpg %>% dplyr::select(-Sample.ID)
pt107_tss <- pt107_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARENB = median(PARENB))

pt108_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARFAL.csv", header = TRUE)
pt108_cpg$sample <- str_replace_all(pt108_cpg$Sample.ID, "TARGET-20-PARFAL-09A-02D", "PARFAL")
pt108_cpg <- pt108_cpg %>% dplyr::select(-Sample.ID)
pt108_tss <- pt108_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARFAL = median(PARFAL))

pt109_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARHVK.csv", header = TRUE)
pt109_cpg$sample <- str_replace_all(pt109_cpg$Sample.ID, "TARGET-20-PARHVK-09A-01D", "PARHVK")
pt109_cpg <- pt109_cpg %>% dplyr::select(-Sample.ID)
pt109_tss <- pt109_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARHVK = median(PARHVK))

pt110_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARJCR.csv", header = TRUE)
pt110_cpg$sample <- str_replace_all(pt110_cpg$Sample.ID, "TARGET-20-PARJCR-09A-01D", "PARJCR")
pt110_cpg <- pt110_cpg %>% dplyr::select(-Sample.ID)
pt110_tss <- pt110_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARJCR = median(PARJCR))

pt111_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARKCX.csv", header = TRUE)
pt111_cpg$sample <- str_replace_all(pt111_cpg$Sample.ID, "TARGET-20-PARKCX-09A-01D", "PARKCX")
pt111_cpg <- pt111_cpg %>% dplyr::select(-Sample.ID)
pt111_tss <- pt111_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARKCX = median(PARKCX))

pt112_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARMZF.csv", header = TRUE)
pt112_cpg$sample <- str_replace_all(pt112_cpg$Sample.ID, "TARGET-20-PARMZF-09A-03D", "PARMZF")
pt112_cpg <- pt112_cpg %>% dplyr::select(-Sample.ID)
pt112_tss <- pt112_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARMZF = median(PARMZF))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PARPDS-09A-01D_450k_L3.txt"
pt113_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt113_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt113_cpg$sample <- ("PARPDS")
pt113_tss <- pt113_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARPDS, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARPDS = median(PARPDS))

pt114_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARPWL.csv", header = TRUE)
pt114_cpg$sample <- str_replace_all(pt114_cpg$Sample.ID, "TARGET-20-PARPWL-09A-01D", "PARPWL")
pt114_cpg <- pt114_cpg %>% dplyr::select(-Sample.ID)
pt114_tss <- pt114_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARPWL = median(PARPWL))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PARTAL-09A-02D_450k_L3.txt"
pt115_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt115_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt115_cpg$sample <- ("PARTAL")
pt115_tss <- pt115_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARTAL, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARTAL = median(PARTAL))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PARUBT-09A-03D_450k_L3.txt"
pt116_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt116_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt116_cpg$sample <- ("PARUBT")
pt116_tss <- pt116_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARUBT, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARUBT = median(PARUBT))

pt117_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARUNX.csv", header = TRUE)
pt117_cpg$sample <- str_replace_all(pt117_cpg$Sample.ID, "TARGET-20-PARUNX-09A-01D", "PARUNX")
pt117_cpg <- pt117_cpg %>% dplyr::select(-Sample.ID)
pt117_tss <- pt117_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARUNX = median(PARUNX))

pt118_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARUTH.csv", header = TRUE)
pt118_cpg$sample <- str_replace_all(pt118_cpg$Sample.ID, "TARGET-20-PARUTH-09A-02D", "PARUTH")
pt118_cpg <- pt118_cpg %>% dplyr::select(-Sample.ID)
pt118_tss <- pt118_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARUTH = median(PARUTH))

pt119_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARUUB.csv", header = TRUE)
pt119_cpg$sample <- str_replace_all(pt119_cpg$Sample.ID, "TARGET-20-PARUUB-09A-01D", "PARUUB")
pt119_cpg <- pt119_cpg %>% dplyr::select(-Sample.ID)
pt119_tss <- pt119_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARUUB = median(PARUUB))

pt120_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARWDZ.csv", header = TRUE)
pt120_cpg$sample <- str_replace_all(pt120_cpg$Sample.ID, "TARGET-20-PARWDZ-09A-01D", "PARWDZ")
pt120_cpg <- pt120_cpg %>% dplyr::select(-Sample.ID)
pt120_tss <- pt120_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARWDZ = median(PARWDZ))

pt121_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARWXU.csv", header = TRUE)
pt121_cpg$sample <- str_replace_all(pt121_cpg$Sample.ID, "TARGET-20-PARWXU-09A-01D", "PARWXU")
pt121_cpg <- pt121_cpg %>% dplyr::select(-Sample.ID)
pt121_tss <- pt121_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARWXU = median(PARWXU))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PARXMP-09A-02D_450k_L3.txt"
pt122_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt122_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt122_cpg$sample <- ("PARXMP")
pt122_tss <- pt122_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARXMP, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARXMP = median(PARXMP))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PARUWX-09A-03D_450k_L3.txt"
pt123_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt123_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt123_cpg$sample <- ("PARUWX")
pt123_tss <- pt123_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARUWX, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARUWX = median(PARUWX))

pt124_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARXNG.csv", header = TRUE)
pt124_cpg$sample <- str_replace_all(pt124_cpg$Sample.ID, "TARGET-20-PARXNG-09A-01D", "PARXNG")
pt124_cpg <- pt124_cpg %>% dplyr::select(-Sample.ID)
pt124_tss <- pt124_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARXNG = median(PARXNG))

pt125_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARYGA.csv", header = TRUE)
pt125_cpg$sample <- str_replace_all(pt125_cpg$Sample.ID, "TARGET-20-PARYGA-09A-01D", "PARYGA")
pt125_cpg <- pt125_cpg %>% dplyr::select(-Sample.ID)
pt125_tss <- pt125_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARYGA = median(PARYGA))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PARYVW-09A-01D_450k_L3.txt"
pt126_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt126_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt126_cpg$sample <- ("PARYVW")
pt126_tss <- pt126_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PARYVW, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PARYVW = median(PARYVW))

pt127_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PARZWH.csv", header = TRUE)
pt127_cpg$sample <- str_replace_all(pt127_cpg$Sample.ID, "TARGET-20-PARZWH-09A-01D", "PARZWH")
pt127_cpg <- pt127_cpg %>% dplyr::select(-Sample.ID)
pt127_tss <- pt127_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PARZWH = median(PARZWH))

pt128_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASBHI.csv", header = TRUE)
pt128_cpg$sample <- str_replace_all(pt128_cpg$Sample.ID, "TARGET-20-PASBHI-09A-01D", "PASBHI")
pt128_cpg <- pt128_cpg %>% dplyr::select(-Sample.ID)
pt128_tss <- pt128_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASBHI = median(PASBHI))

pt129_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASBPW.csv", header = TRUE)
pt129_cpg$sample <- str_replace_all(pt129_cpg$Sample.ID, "TARGET-20-PASBPW-09A-01D", "PASBPW")
pt129_cpg <- pt129_cpg %>% dplyr::select(-Sample.ID)
pt129_tss <- pt129_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASBPW = median(PASBPW))

pt130_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASCCS.csv", header = TRUE)
pt130_cpg$sample <- str_replace_all(pt130_cpg$Sample.ID, "TARGET-20-PASCCS-09A-01D", "PASCCS")
pt130_cpg <- pt130_cpg %>% dplyr::select(-Sample.ID)
pt130_tss <- pt130_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASCCS = median(PASCCS))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PASCFW-09A-03D_450k_L3.txt"
pt131_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt131_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt131_cpg$sample <- ("PASCFW")
pt131_tss <- pt131_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PASCFW, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PASCFW = median(PASCFW))

pt132_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASDGX.csv", header = TRUE)
pt132_cpg$sample <- str_replace_all(pt132_cpg$Sample.ID, "TARGET-20-PASDGX-09A-01D", "PASDGX")
pt132_cpg <- pt132_cpg %>% dplyr::select(-Sample.ID)
pt132_tss <- pt132_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASDGX = median(PASDGX))

pt133_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASECW.csv", header = TRUE)
pt133_cpg$sample <- str_replace_all(pt133_cpg$Sample.ID, "TARGET-20-PASECW-09A-01D", "PASECW")
pt133_cpg <- pt133_cpg %>% dplyr::select(-Sample.ID)
pt133_tss <- pt133_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASECW = median(PASECW))

pt134_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASEFD.csv", header = TRUE)
pt134_cpg$sample <- str_replace_all(pt134_cpg$Sample.ID, "TARGET-20-PASEFD-09A-01D", "PASEFD")
pt134_cpg <- pt134_cpg %>% dplyr::select(-Sample.ID)
pt134_tss <- pt134_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASEFD = median(PASEFD))

pt135_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASFEW.csv", header = TRUE)
pt135_cpg$sample <- str_replace_all(pt135_cpg$Sample.ID, "TARGET-20-PASFEW-09A-02D", "PASFEW")
pt135_cpg <- pt135_cpg %>% dplyr::select(-Sample.ID)
pt135_tss <- pt135_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASFEW = median(PASFEW))

pt136_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASFNP.csv", header = TRUE)
pt136_cpg$sample <- str_replace_all(pt136_cpg$Sample.ID, "TARGET-20-PASFNP-09A-01D", "PASFNP")
pt136_cpg <- pt136_cpg %>% dplyr::select(-Sample.ID)
pt136_tss <- pt136_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASFNP = median(PASFNP))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PASGMZ-09A-02D_450k_L3.txt"
pt137_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt137_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt137_cpg$sample <- ("PASGMZ")
pt137_tss <- pt137_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PASGMZ, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PASGMZ = median(PASGMZ))

pt138_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASGWH.csv", header = TRUE)
pt138_cpg$sample <- str_replace_all(pt138_cpg$Sample.ID, "TARGET-20-PASGWH-09A-02D", "PASGWH")
pt138_cpg <- pt138_cpg %>% dplyr::select(-Sample.ID)
pt138_tss <- pt138_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASGWH = median(PASGWH))

path <- "https://target-data.nci.nih.gov/Public/AML/methylation_array/L3/TARGET-20-PASHBI-09A-02D_450k_L3.txt"
pt139_cpg <- vroom::vroom(path, delim = "\t", skip = 1) 
colnames(pt139_cpg) <- c("Probe.Name", "AVG_Beta", "Gene.Symbols", "Chromosome", "Position")
pt139_cpg$sample <- ("PASHBI")
pt139_tss <- pt139_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  mutate(Gene.Symbols = gsub("\\;.*", "", Gene.Symbols)) %>%
  dplyr:: select(PASHBI, Gene.Symbols) %>%
  group_by(Gene.Symbols) %>%
  summarise(PASHBI = median(PASHBI))

pt140_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/cpg/PASHYZ.csv", header = TRUE)
pt140_cpg$sample <- str_replace_all(pt140_cpg$Sample.ID, "TARGET-20-PASHYZ-09A-01D", "PASHYZ")
pt140_cpg <- pt140_cpg %>% dplyr::select(-Sample.ID)
pt140_tss <- pt140_cpg %>%
  filter(Probe.Name %in% illumina_tss_2) %>%
  dplyr::select(AVG_Beta, Gene.Symbols, sample) %>%
  pivot_wider(names_from = sample,
              values_from = AVG_Beta) %>% unnest() %>%
  group_by(Gene.Symbols) %>%
  summarise(PASHYZ = median(PASHYZ))


# merge tss files

pt1_2tss <- merge(pt1_tss, pt2_tss, by = "Gene.Symbols")  
pt1_3tss <- merge(pt1_2tss, pt3_tss, by = "Gene.Symbols")
pt1_4tss <- merge(pt1_3tss, pt4_tss, by = "Gene.Symbols")
pt1_5tss <- merge(pt1_4tss, pt5_tss, by = "Gene.Symbols")
pt1_6tss <- merge(pt1_5tss, pt6_tss, by = "Gene.Symbols")
pt1_7tss <- merge(pt1_6tss, pt7_tss, by = "Gene.Symbols")
pt1_8tss <- merge(pt1_7tss, pt8_tss, by = "Gene.Symbols")
pt1_9tss <- merge(pt1_8tss, pt9_tss, by = "Gene.Symbols")
pt1_10tss <- merge(pt1_9tss, pt10_tss, by = "Gene.Symbols")

pt1_11tss <- merge(pt1_10tss, pt11_tss, by = "Gene.Symbols")
pt1_12tss <- merge(pt1_11tss, pt12_tss, by = "Gene.Symbols")
pt1_13tss <- merge(pt1_12tss, pt13_tss, by = "Gene.Symbols")
pt1_14tss <- merge(pt1_13tss, pt14_tss, by = "Gene.Symbols")
pt1_15tss <- merge(pt1_14tss, pt15_tss, by = "Gene.Symbols")
pt1_16tss <- merge(pt1_15tss, pt16_tss, by = "Gene.Symbols")
pt1_17tss <- merge(pt1_16tss, pt17_tss, by = "Gene.Symbols")
pt1_18tss <- merge(pt1_17tss, pt18_tss, by = "Gene.Symbols")
pt1_19tss <- merge(pt1_18tss, pt19_tss, by = "Gene.Symbols")

pt1_20tss <- merge(pt1_19tss, pt20_tss, by = "Gene.Symbols")
pt1_21tss <- merge(pt1_20tss, pt21_tss, by = "Gene.Symbols")
pt1_22tss <- merge(pt1_21tss, pt22_tss, by = "Gene.Symbols")
pt1_23tss <- merge(pt1_22tss, pt23_tss, by = "Gene.Symbols")
pt1_24tss <- merge(pt1_23tss, pt24_tss, by = "Gene.Symbols")
pt1_25tss <- merge(pt1_24tss, pt25_tss, by = "Gene.Symbols")
pt1_26tss <- merge(pt1_25tss, pt26_tss, by = "Gene.Symbols")
pt1_27tss <- merge(pt1_26tss, pt27_tss, by = "Gene.Symbols")
pt1_28tss <- merge(pt1_27tss, pt28_tss, by = "Gene.Symbols")
pt1_29tss <- merge(pt1_28tss, pt29_tss, by = "Gene.Symbols")

pt1_30tss <- merge(pt1_29tss, pt30_tss, by = "Gene.Symbols")
pt1_31tss <- merge(pt1_30tss, pt31_tss, by = "Gene.Symbols")
pt1_32tss <- merge(pt1_31tss, pt32_tss, by = "Gene.Symbols")
pt1_33tss <- merge(pt1_32tss, pt33_tss, by = "Gene.Symbols")
pt1_34tss <- merge(pt1_33tss, pt34_tss, by = "Gene.Symbols")
pt1_35tss <- merge(pt1_34tss, pt35_tss, by = "Gene.Symbols")
pt1_36tss <- merge(pt1_35tss, pt36_tss, by = "Gene.Symbols")
pt1_37tss <- merge(pt1_36tss, pt37_tss, by = "Gene.Symbols")
pt1_38tss <- merge(pt1_37tss, pt38_tss, by = "Gene.Symbols")
pt1_39tss <- merge(pt1_38tss, pt39_tss, by = "Gene.Symbols")

pt1_30tss <- merge(pt1_29tss, pt30_tss, by = "Gene.Symbols")
pt1_31tss <- merge(pt1_30tss, pt31_tss, by = "Gene.Symbols")
pt1_32tss <- merge(pt1_31tss, pt32_tss, by = "Gene.Symbols")
pt1_33tss <- merge(pt1_32tss, pt33_tss, by = "Gene.Symbols")
pt1_34tss <- merge(pt1_33tss, pt34_tss, by = "Gene.Symbols")
pt1_35tss <- merge(pt1_34tss, pt35_tss, by = "Gene.Symbols")
pt1_36tss <- merge(pt1_35tss, pt36_tss, by = "Gene.Symbols")
pt1_37tss <- merge(pt1_36tss, pt37_tss, by = "Gene.Symbols")
pt1_38tss <- merge(pt1_37tss, pt38_tss, by = "Gene.Symbols")
pt1_39tss <- merge(pt1_38tss, pt39_tss, by = "Gene.Symbols")

pt1_40tss <- merge(pt1_39tss, pt40_tss, by = "Gene.Symbols")
pt1_41tss <- merge(pt1_40tss, pt41_tss, by = "Gene.Symbols")
pt1_42tss <- merge(pt1_41tss, pt42_tss, by = "Gene.Symbols")
pt1_43tss <- merge(pt1_42tss, pt43_tss, by = "Gene.Symbols")
pt1_44tss <- merge(pt1_43tss, pt44_tss, by = "Gene.Symbols")
pt1_45tss <- merge(pt1_44tss, pt45_tss, by = "Gene.Symbols")
pt1_46tss <- merge(pt1_45tss, pt46_tss, by = "Gene.Symbols")
pt1_47tss <- merge(pt1_46tss, pt47_tss, by = "Gene.Symbols")
pt1_48tss <- merge(pt1_47tss, pt48_tss, by = "Gene.Symbols")
pt1_49tss <- merge(pt1_48tss, pt49_tss, by = "Gene.Symbols")

pt1_50tss <- merge(pt1_49tss, pt50_tss, by = "Gene.Symbols")
pt1_51tss <- merge(pt1_50tss, pt51_tss, by = "Gene.Symbols")
pt1_52tss <- merge(pt1_51tss, pt52_tss, by = "Gene.Symbols")
pt1_53tss <- merge(pt1_52tss, pt53_tss, by = "Gene.Symbols")
pt1_54tss <- merge(pt1_53tss, pt54_tss, by = "Gene.Symbols")
pt1_55tss <- merge(pt1_54tss, pt55_tss, by = "Gene.Symbols")
pt1_56tss <- merge(pt1_55tss, pt56_tss, by = "Gene.Symbols")
pt1_57tss <- merge(pt1_56tss, pt57_tss, by = "Gene.Symbols")
pt1_58tss <- merge(pt1_57tss, pt58_tss, by = "Gene.Symbols")
pt1_59tss <- merge(pt1_58tss, pt59_tss, by = "Gene.Symbols")

pt1_60tss <- merge(pt1_59tss, pt60_tss, by = "Gene.Symbols")
pt1_61tss <- merge(pt1_60tss, pt61_tss, by = "Gene.Symbols")
pt1_62tss <- merge(pt1_61tss, pt62_tss, by = "Gene.Symbols")
pt1_63tss <- merge(pt1_62tss, pt63_tss, by = "Gene.Symbols")
pt1_64tss <- merge(pt1_63tss, pt64_tss, by = "Gene.Symbols")
pt1_65tss <- merge(pt1_64tss, pt65_tss, by = "Gene.Symbols")
pt1_66tss <- merge(pt1_65tss, pt66_tss, by = "Gene.Symbols")
pt1_67tss <- merge(pt1_66tss, pt67_tss, by = "Gene.Symbols")
pt1_68tss <- merge(pt1_67tss, pt68_tss, by = "Gene.Symbols")
pt1_69tss <- merge(pt1_68tss, pt69_tss, by = "Gene.Symbols")

pt1_70tss <- merge(pt1_69tss, pt70_tss, by = "Gene.Symbols")
pt1_71tss <- merge(pt1_70tss, pt71_tss, by = "Gene.Symbols")
pt1_72tss <- merge(pt1_71tss, pt72_tss, by = "Gene.Symbols")
pt1_73tss <- merge(pt1_72tss, pt73_tss, by = "Gene.Symbols")
pt1_74tss <- merge(pt1_73tss, pt74_tss, by = "Gene.Symbols")
pt1_75tss <- merge(pt1_74tss, pt75_tss, by = "Gene.Symbols")
pt1_76tss <- merge(pt1_75tss, pt76_tss, by = "Gene.Symbols")
pt1_77tss <- merge(pt1_76tss, pt77_tss, by = "Gene.Symbols")
pt1_78tss <- merge(pt1_77tss, pt78_tss, by = "Gene.Symbols")
pt1_79tss <- merge(pt1_78tss, pt79_tss, by = "Gene.Symbols")

pt1_80tss <- merge(pt1_79tss, pt80_tss, by = "Gene.Symbols")
pt1_81tss <- merge(pt1_80tss, pt81_tss, by = "Gene.Symbols")
pt1_82tss <- merge(pt1_81tss, pt82_tss, by = "Gene.Symbols")
pt1_83tss <- merge(pt1_82tss, pt83_tss, by = "Gene.Symbols")
pt1_84tss <- merge(pt1_83tss, pt84_tss, by = "Gene.Symbols")
pt1_85tss <- merge(pt1_84tss, pt85_tss, by = "Gene.Symbols")
pt1_86tss <- merge(pt1_85tss, pt86_tss, by = "Gene.Symbols")
pt1_87tss <- merge(pt1_86tss, pt87_tss, by = "Gene.Symbols")
pt1_88tss <- merge(pt1_87tss, pt88_tss, by = "Gene.Symbols")
pt1_89tss <- merge(pt1_88tss, pt89_tss, by = "Gene.Symbols")

pt1_90tss <- merge(pt1_89tss, pt90_tss, by = "Gene.Symbols")

pt1_92tss <- merge(pt1_90tss, pt92_tss, by = "Gene.Symbols")
pt1_93tss <- merge(pt1_92tss, pt93_tss, by = "Gene.Symbols")
pt1_94tss <- merge(pt1_93tss, pt94_tss, by = "Gene.Symbols")
pt1_95tss <- merge(pt1_94tss, pt95_tss, by = "Gene.Symbols")
pt1_96tss <- merge(pt1_95tss, pt96_tss, by = "Gene.Symbols")
pt1_97tss <- merge(pt1_96tss, pt97_tss, by = "Gene.Symbols")
pt1_98tss <- merge(pt1_97tss, pt98_tss, by = "Gene.Symbols")
pt1_99tss <- merge(pt1_98tss, pt99_tss, by = "Gene.Symbols")

pt1_100tss <- merge(pt1_99tss, pt100_tss, by = "Gene.Symbols")
pt1_101tss <- merge(pt1_100tss, pt101_tss, by = "Gene.Symbols")
pt1_102tss <- merge(pt1_101tss, pt102_tss, by = "Gene.Symbols")
pt1_103tss <- merge(pt1_102tss, pt103_tss, by = "Gene.Symbols")
pt1_104tss <- merge(pt1_103tss, pt104_tss, by = "Gene.Symbols")
pt1_105tss <- merge(pt1_104tss, pt105_tss, by = "Gene.Symbols")
pt1_106tss <- merge(pt1_105tss, pt106_tss, by = "Gene.Symbols")
pt1_107tss <- merge(pt1_106tss, pt107_tss, by = "Gene.Symbols")
pt1_108tss <- merge(pt1_107tss, pt108_tss, by = "Gene.Symbols")
pt1_109tss <- merge(pt1_108tss, pt109_tss, by = "Gene.Symbols")

pt1_110tss <- merge(pt1_109tss, pt110_tss, by = "Gene.Symbols")
pt1_111tss <- merge(pt1_110tss, pt111_tss, by = "Gene.Symbols")
pt1_112tss <- merge(pt1_111tss, pt112_tss, by = "Gene.Symbols")
pt1_113tss <- merge(pt1_112tss, pt113_tss, by = "Gene.Symbols")
pt1_114tss <- merge(pt1_113tss, pt114_tss, by = "Gene.Symbols")
pt1_115tss <- merge(pt1_114tss, pt115_tss, by = "Gene.Symbols")
pt1_116tss <- merge(pt1_115tss, pt116_tss, by = "Gene.Symbols")
pt1_117tss <- merge(pt1_116tss, pt117_tss, by = "Gene.Symbols")
pt1_118tss <- merge(pt1_117tss, pt118_tss, by = "Gene.Symbols")
pt1_119tss <- merge(pt1_118tss, pt119_tss, by = "Gene.Symbols")

pt1_120tss <- merge(pt1_119tss, pt120_tss, by = "Gene.Symbols")
pt1_121tss <- merge(pt1_120tss, pt121_tss, by = "Gene.Symbols")
pt1_122tss <- merge(pt1_121tss, pt122_tss, by = "Gene.Symbols")
pt1_123tss <- merge(pt1_122tss, pt123_tss, by = "Gene.Symbols")
pt1_124tss <- merge(pt1_123tss, pt124_tss, by = "Gene.Symbols")
pt1_125tss <- merge(pt1_124tss, pt125_tss, by = "Gene.Symbols")
pt1_126tss <- merge(pt1_125tss, pt126_tss, by = "Gene.Symbols")
pt1_127tss <- merge(pt1_126tss, pt127_tss, by = "Gene.Symbols")
pt1_128tss <- merge(pt1_127tss, pt128_tss, by = "Gene.Symbols")
pt1_129tss <- merge(pt1_128tss, pt129_tss, by = "Gene.Symbols")

pt1_130tss <- merge(pt1_129tss, pt130_tss, by = "Gene.Symbols")
pt1_131tss <- merge(pt1_130tss, pt131_tss, by = "Gene.Symbols")
pt1_132tss <- merge(pt1_131tss, pt132_tss, by = "Gene.Symbols")
pt1_133tss <- merge(pt1_132tss, pt133_tss, by = "Gene.Symbols")
pt1_134tss <- merge(pt1_133tss, pt134_tss, by = "Gene.Symbols")
pt1_135tss <- merge(pt1_134tss, pt135_tss, by = "Gene.Symbols")
pt1_136tss <- merge(pt1_135tss, pt136_tss, by = "Gene.Symbols")
pt1_137tss <- merge(pt1_136tss, pt137_tss, by = "Gene.Symbols")
pt1_138tss <- merge(pt1_137tss, pt138_tss, by = "Gene.Symbols")
pt1_139tss <- merge(pt1_138tss, pt139_tss, by = "Gene.Symbols")

pt1_140tss <- merge(pt1_139tss, pt140_tss, by = "Gene.Symbols")


pt1_136tss_target <- as.data.frame(t(pt1_136tss %>% column_to_rownames("Gene.Symbols"))) %>% rownames_to_column("sample")

target_cpg_pilot <- pt1_136tss_target

write.csv(target_cpg_pilot, "~/Documents/R/pedcancer/target_data/TARGET_100/target_cpg_pilot.csv", row.names = FALSE)

cpg_samples <- pull(target_cpg_pilot %>% dplyr::select(sample))


```

```{r}

# filter expression, mirna and cpg dataframe in a circle to find samples with all 3 multiomics

# start with cpg samples in mir
target_pilot_mir <- target_mirna_pilot %>% filter(sample %in% cpg_samples)
target_pilot_mir_samples <- pull(target_pilot_mir %>% dplyr::select(sample))
write.csv(target_pilot_mir, "~/Documents/R/pedcancer/target_data/TARGET_100/target_pilot_mir.csv", row.names = FALSE)

sample_overlap <- target_mirna_pilot %>% 
  mutate(exp = ifelse(sample %in% expression_samples, 'yes', 'no'), 
         cpg = ifelse(sample %in% cpg_samples, 'yes', 'no')) %>%
  dplyr::select(sample, exp, cpg)

# need better list for mulitomcs samples
expression_and_methylation_samples <- target_discovery_sample_matrix %>% 
  dplyr::select(Case.USI,
                mRNA.seq..BCCA.and.Hudson.Alpha..HA..,
                Methylation) %>%
  filter(mRNA.seq..BCCA.and.Hudson.Alpha..HA.. != "F" &
          mRNA.seq..BCCA.and.Hudson.Alpha..HA.. != "N" &
         Methylation == "Y") %>%
  mutate(sample = str_replace_all(Case.USI, "TARGET-20-","")) %>%
  dplyr::select(sample, mRNA.seq..BCCA.and.Hudson.Alpha..HA.., Methylation) %>%
  mutate(mirna = ifelse(sample %in% mirna_samples, 'yes', 'no'), 
         exp = ifelse(sample %in% expression_samples, 'yes', 'no'), 
         cpg = ifelse(sample %in% cpg_samples, 'yes', 'no'))



target_pilot_exp <- target_expression_pilot %>% filter(sample %in% target_pilot_mir_samples)
target_pilot_exp_samples <- pull(target_pilot_exp %>% dplyr::select(sample))
write.csv(target_pilot_exp, "~/Documents/R/pedcancer/target_data/TARGET_100/target_pilot_exp.csv", row.names = FALSE)



target_pilot_cpg <- target_pilot_cpg %>% filter(sample %in% target_pilot_exp_samples)
target_pilot_cpg_samples <- pull(target_pilot_cpg %>% dplyr::select(sample))
write.csv(target_pilot_cpg, "~/Documents/R/pedcancer/target_data/TARGET_100/target_pilot_cpg.csv", row.names = FALSE)


#### 

cpg_mirna_samples <- pull(target_cpg_pilot %>%
  filter(sample %in% mirna_samples) %>%
  dplyr::select(sample))

trios <- pull(target_expression_pilot %>%
  filter(sample %in% cpg_mirna_samples) %>%
  dplyr::select(sample))


target_cpg_pilot_115 <- target_cpg_pilot %>%
  filter(sample %in% trios)
write.csv(target_cpg_pilot_115, "~/Documents/R/pedcancer/target_data/TARGET_100/target_cpg_pilot_115.csv", row.names = FALSE)

target_mirna_pilot_115 <- target_mirna_pilot %>%
  filter(sample %in% trios)
write.csv(target_mirna_pilot_115, "~/Documents/R/pedcancer/target_data/TARGET_100/target_mirna_pilot_115.csv", row.names = FALSE)

target_expression_pilot_115 <- target_expression_pilot %>%
  filter(sample %in% trios)
write.csv(target_expression_pilot_115, "~/Documents/R/pedcancer/target_data/TARGET_100/target_expression_pilot_115.csv", row.names = FALSE)


```

```{r}
# multiomics clusters

#download target data
target_expression <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/target_expression_pilot_115.csv")
target_mirna <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/target_mirna_pilot_115.csv")
target_cpg <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/target_cpg_pilot_115.csv")
mitocarta <- read.csv("~/Documents/R/ddh/data/mitocarta.csv")

# download UMAP clusters = multiomics_clusters_target_114_4_4_2_CORRECT
multiomics_clusters_target_114_4_4_2_CORRECT <- readRDS("~/Documents/R/pedcancer/target_data/TARGET_100/multiomics_clusters_target_114_4_4_2_CORRECT.Rds")

multiomics_clusters_092822 <- readRDS("~/Documents/R/pedcancer/target_data/TARGET_100/multiomics_clusters092822.Rds")

## most recent target-AML umap clustering Feb 2023
target_umap <- readRDS("~/Documents/R/pedcancer/target_ubmi_clusters.Rds")


# merge with clinical dataset
target_discovery_clinical <- read.xlsx("~/Documents/R/pedcancer/target_data/TARGET_AML_ClinicalData_Discovery_20211201.xlsx", 1)  # read first sheet

# combine clinical and umap files
target_clinical <- target_discovery_clinical %>%
    mutate(id = gsub("TARGET-20-", "", TARGET.USI))

target_combined <- merge(target_umap, target_clinical, by = "id") %>%
    mutate(relapse = ifelse(First.Event == 'Relapse', 'yes', 
                          ifelse(First.Event == "Censored", 'no',
                                 'other')))

#need to choose correct clustering data
multiomics_114_4_4_2_samples <- pull(multiomics_clusters_092822 %>% dplyr::select(id))

multiomics_samples <- pull(multiomics_clusters_092822 %>% dplyr::select(id))

cluster_0 <- pull(multiomics_clusters_092822 %>%
                    filter(clust == "cluster0") %>%
                    dplyr::select(id))

cluster_1 <- pull(multiomics_clusters_092822 %>%
                    filter(clust == "cluster1") %>%
                    dplyr::select(id))
  
cluster_2 <- pull(multiomics_clusters_092822 %>%
                    filter(clust == "cluster2") %>%
                    dplyr::select(id))

cluster_3 <- pull(multiomics_clusters_092822 %>%
                    filter(clust == "cluster3") %>%
                    dplyr::select(id))

cluster_4 <- pull(multiomics_clusters_092822 %>%
                    filter(clust == "cluster4") %>%
                    dplyr::select(id))

cluster_5 <- pull(multiomics_clusters_092822 %>%
                    filter(clust == "cluster5") %>%
                    dplyr::select(id))


# mark clusters in clinical dataset

target_discovery_clinical_3 <- target_discovery_clinical %>% 
                                mutate(sample = str_replace_all(TARGET.USI, "TARGET-20-", ""))

target_discovery_clinical_clusters <- target_discovery_clinical_3 %>%
              filter(sample %in% multiomics_114_4_4_2_samples) %>%
  mutate(cluster = ifelse(sample %in% cluster_1, '1', 
                          ifelse(sample %in% cluster_2, '2', 
                                 ifelse(sample %in% cluster_3, '3', '4')))) %>%
  mutate(fav_status = ifelse(Primary.Cytogenetic.Code == "inv(16)" |
                              Primary.Cytogenetic.Code == "t(8;21)" |
                              NPM.mutation == "Yes" |
                              CEBPA.mutation == "Yes", 'favorable', 
                            ifelse(Primary.Cytogenetic.Code == "MLL" |
                                     monosomy.5 == "Yes" |
                                     monosomy.7 == "Yes" |
                                     FLT3.ITD.positive. == "Yes" |
                                     FLT3.PM == "Yes", 'unfavorable', 'na')))

```




```{r}

## CLUSTER ANALYSIS WITH CLINICAL DATA



# samples in each cluster after filtering out unknown cytogenetic code
target_combined %>%
  group_by(clust) %>%
  summarise(length(clust))

target_combined %>%
  group_by(clust, Primary.Cytogenetic.Code) %>%
  summarise(number = length(id)) %>%
  pivot_wider(names_from = clust, values_from = number)

target_combined %>%
  group_by(clust, Gene.Fusion) %>%
  summarise(number = length(id)) %>%
  pivot_wider(names_from = clust, values_from = number) %>%
  arrange(Gene.Fusion)

# cluster plot
target_combined %>% ggplot(mapping = aes(x = X1, y = X2, color = clust)) + geom_point(size = 4) +
    guides(color = guide_legend(override.aes = list(size = 3))) +
  theme(panel.background = element_rect(fill = "transparent"),
        axis.line = element_line(linetype = "solid", size = 0.5),
        axis.title.x = element_text(family = "Arial", size = 16),
        axis.text.x = element_text(family = "Arial", size = 16), 
        axis.title.y = element_text(family = "Arial", size = 16),
        axis.text.y = element_text(family = "Arial",, size = 16),
        legend.title = element_text(family = "Arial", size = 12),
        legend.text = element_text(family = "Arial", size = 12),
        legend.position = c(0.8, 0.28),
        legend.background = element_rect(linetype = "solid", color = "gray"))


# age at diagnosis 
target_combined %>%
mutate(dx_age_yr = Age.at.Diagnosis.in.Days / 365) %>%
  group_by(clust) %>%
  dplyr::select(clust, dx_age_yr) %>%
  arrange(clust) %>%
  ggplot(mapping = aes(x = clust, y = dx_age_yr, fill = clust)) +
  geom_boxplot()

# gender
target_combined %>%
  group_by(clust) %>%
  mutate(cluster_length = length(clust)) %>%
  group_by(Gender, clust) %>%
  mutate(gender_length = length(Gender)) %>%
  dplyr::select(id, Gender, clust, cluster_length, gender_length) %>%
  mutate(gender_prop = gender_length / cluster_length) %>%
  ggplot(mapping = aes(x = fct_reorder(Gender, -gender_prop), y = gender_prop, group = clust, color = clust)) + 
  geom_point() +
          theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16))

# clusters by race  
target_combined %>%
  group_by(clust) %>%
  mutate(cluster_length = length(clust)) %>%
  group_by(Race, clust) %>%
  mutate(race_length = length(Race)) %>%
  dplyr::select(id, Race, clust, cluster_length, race_length) %>%
  mutate(race_prop = race_length / cluster_length) %>%
    ggplot(mapping = aes(x = fct_reorder(Race, -race_prop), y = race_prop, group = clust, color = clust)) + 
  geom_point() +
          theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16))

# clusters by primary cytogenetic code
target_combined %>%
  group_by(clust) %>%
  mutate(cluster_length = length(clust)) %>%
  group_by(Primary.Cytogenetic.Code, clust) %>%
  mutate(cyto_length = length(Primary.Cytogenetic.Code)) %>%
  dplyr::select(id, Primary.Cytogenetic.Code, clust, cluster_length, cyto_length) %>%
  mutate(proportion = cyto_length / cluster_length) %>%
    ggplot(mapping = aes(x = fct_reorder(Primary.Cytogenetic.Code, -proportion), y = proportion, group = clust, color = clust)) + 
  geom_point(size = 5) +
          theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16))

#how do clusters break down by fusion status ?
target_combined %>%
  group_by(clust) %>%
  mutate(cluster_length = length(clust)) %>%
  group_by(Gene.Fusion, clust) %>%
  mutate(fusion_length = length(Gene.Fusion)) %>%
  dplyr::select(id, Gene.Fusion, clust, cluster_length, fusion_length) %>%
  mutate(proportion = fusion_length / cluster_length) %>%
  ggplot(mapping = aes(x = fct_reorder(Gene.Fusion, -proportion), y = proportion, group = clust, color = clust)) + 
  geom_point(size = 5) +
          theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16))



# clusters by FLT3 status 
target_combined %>%
  group_by(clust) %>%
  mutate(cluster_length = length(clust)) %>%
  group_by(FLT3.ITD.positive., clust) %>%
  mutate(Flt3_length = length(FLT3.ITD.positive.)) %>%
  dplyr::select(id, FLT3.ITD.positive., clust, cluster_length, Flt3_length) %>%
  mutate(Flt3_prop = Flt3_length / cluster_length) %>%
    ggplot(mapping = aes(x = fct_reorder(FLT3.ITD.positive., -Flt3_prop), y = Flt3_prop, group = clust, color = clust)) + 
  geom_point() +
          theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16))


# clusters by vital status  
target_combined %>%
  group_by(clust) %>%
  mutate(cluster_length = length(clust)) %>%
  group_by(Vital.Status, clust) %>%
  mutate(vital_length = length(Vital.Status)) %>%
  dplyr::select(id, Vital.Status, clust, cluster_length, vital_length) %>%
  mutate(vital_prop = vital_length / cluster_length) %>%
    ggplot(mapping = aes(x = fct_reorder(Vital.Status, -vital_prop), y = vital_prop, group = clust, color = clust)) + 
  geom_point() +
          theme(plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, color = "black", angle = 45, hjust = 0.90, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16))

######
######
##### overall survival length and event free survival length
target_combined %>%
  group_by(clust) %>%
  ggplot(mapping = aes(x = clust, y = Event.Free.Survival.Time.in.Days, group = clust, fill = clust)) +
  geom_violin(width = 1.5, trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white") +
  guides(color = guide_legend(override.aes = list(size = 8))) +
  xlab("Multiomics Cluster") + 
  ylab("Event Free Survival (Days)") +
            theme(panel.background = element_rect(fill = "transparent"),
                  plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, angle = -30, hjust = 0, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
               axis.line = element_line(linetype = "solid", size = 0.5),
        legend.background = element_rect(linetype = "solid", color = "gray"))

target_combined %>%
  dplyr::select(id, clust, First.Event) %>%
  group_by(First.Event, clust) %>%
  summarise(length(First.Event)) %>%
  pivot_wider(names_from = First.Event,
              values_from = `length(First.Event)`)

target_combined %>%
  group_by(clust) %>%
  ggplot(mapping = aes(x = clust, y = Overall.Survival.Time.in.Days, group = clust, fill = clust)) +
  geom_violin(width = 1.5, trim = FALSE) +
  geom_boxplot(width = 0.1, fill = "white") +
  guides(color = guide_legend(override.aes = list(size = 8))) +
  xlab("Multiomics Cluster") + 
  ylab("Event Free Survival (Days)") +
            theme(panel.background = element_rect(fill = "transparent"),
                  plot.title = element_text(family = "Arial", size = 20),
              axis.title.x = element_text(family = "Arial", size = 16),
              axis.title.y = element_text(family = "Arial", size = 16),
              axis.text.x = element_text(family = "Arial", size = 16, angle = -30, hjust = 0, vjust = 0.95), 
              axis.text.y = element_text(family = "Arial",, size = 16),
               axis.line = element_line(linetype = "solid", size = 0.5),
        legend.background = element_rect(linetype = "solid", color = "gray"))

target_combined %>%
  group_by(clust) %>%
  ggplot(mapping = aes(x = clust, y = Overall.Survival.Time.in.Days, group = clust, fill = clust)) +
  geom_boxplot() +
  facet_wrap(~Primary.Cytogenetic.Code)

target_combined %>%
  group_by(clust) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  ggplot(mapping = aes(x = clust, y = Overall.Survival.Time.in.Days, group = clust, fill = clust)) +
  geom_boxplot()

target_combined %>%
  group_by(clust) %>%
  ggplot(mapping = aes(x = clust, y = Event.Free.Survival.Time.in.Days, group = clust, fill = clust)) +
  geom_boxplot() +
  facet_wrap(~Primary.Cytogenetic.Code)


# WBC at diagnosis
target_combined %>%
  group_by(clust) %>%
  dplyr::select(clust, WBC.at.Diagnosis) %>%
  arrange(clust) %>%
  ggplot(mapping = aes(x = clust, y = WBC.at.Diagnosis, group = clust)) +
  geom_boxplot()




```

```{r}

# more data to viz

# clusters by cytogenetics
target_combined %>%
  ggplot(mapping = aes(x = X1, y = X2, color = Primary.Cytogenetic.Code)) +
  geom_point(size = 3) +
    guides(color = guide_legend(override.aes = list(size = 8))) +
  theme(panel.background = element_rect(fill = "transparent"),
        axis.line = element_line(linetype = "solid", size = 0.5),
        axis.title.x = element_text(family = "Arial", size = 16),
        axis.text.x = element_text(family = "Arial", size = 16), 
        axis.title.y = element_text(family = "Arial", size = 16),
        axis.text.y = element_text(family = "Arial",, size = 16), 
        legend.position = c(0.8, 0.28),
        legend.background = element_rect(linetype = "solid", color = "gray"))
  
#clusters by gene fusion
target_combined %>%
  ggplot(mapping = aes(x = X1, y = X2, color = Gene.Fusion)) +
  geom_point(size = 3) +
    guides(color = guide_legend(override.aes = list(size = 6)))

target_combined %>%
  ggplot(mapping = aes(x = X1, y = X2, color = Primary.Cytogenetic.Code, size = Event.Free.Survival.Time.in.Days)) +
  geom_point() +
  guides(color = guide_legend(override.aes = list(size = 8)))

target_combined %>%
  ggplot(mapping = aes(x = X1, y = X2, color = Primary.Cytogenetic.Code, size = Overall.Survival.Time.in.Days)) +
  geom_point() +
  guides(color = guide_legend(override.aes = list(size = 8)))

# cluster by favorable status


# favorable status EFS 



target_combined %>%
  group_by(clust, Risk.group) %>%
  ggplot(mapping = aes(x = clust, y = Event.Free.Survival.Time.in.Days, fill = Risk.group)) + 
  geom_boxplot() +
  guides(color = guide_legend(override.aes = list(size = 6))) +
  ylab("Event Free Survival (Days)") +
  xlab("Multiomic Cluster") +
  theme(panel.background = element_rect(fill = "transparent"),
        axis.line = element_line(linetype = "solid", size = 0.5),
        axis.title.x = element_text(family = "Arial", size = 16),
        axis.text.x = element_text(family = "Arial", size = 16), 
        axis.title.y = element_text(family = "Arial", size = 16),
        axis.text.y = element_text(family = "Arial",, size = 16))

target_combined %>%
  filter(First.Event == "Relapse") %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  group_by(clust, Primary.Cytogenetic.Code) %>%
  ggplot(mapping = aes(x = clust, y = Event.Free.Survival.Time.in.Days, fill = clust)) + 
  geom_boxplot() +
  guides(color = guide_legend(override.aes = list(size = 6))) +
  ylab("Overall Survival (Days)") +
  xlab("Multiomic Cluster") +
  theme(panel.background = element_rect(fill = "transparent"),
        axis.line = element_line(linetype = "solid", size = 0.5),
        axis.title.x = element_text(family = "Arial", size = 16),
        axis.text.x = element_text(family = "Arial", size = 16), 
        axis.title.y = element_text(family = "Arial", size = 16),
        axis.text.y = element_text(family = "Arial",, size = 16),
        legend.position = "none")

relapse_ids <- target_combined %>%
  filter(First.Event == 'Relapse') %>%
    dplyr::select(id)

has_both <- pull(relapse_ids %>%
  mutate(has04 = ifelse(id %in% relapse_id2, 'yes', 'no')) %>%
  filter(has04 == 'yes') %>%
  dplyr::select(id))

nonrelapse_ids <- pull(target_combined %>%
  filter(First.Event == 'Censored') %>%
    dplyr::select(id))

target_combined %>%
  group_by(relapse) %>%
  summarise(OS = median(Overall.Survival.Time.in.Days),
            EFS = median(Event.Free.Survival.Time.in.Days))

target_combined %>%
  filter(relapse != 'other') %>%
  ggplot(mapping = aes(x = relapse, y = Overall.Survival.Time.in.Days)) + 
  geom_boxplot()

target_combined %>%
  mutate(flt3_status = ifelse(FLT3.PM == 'No' | FLT3.PM == 'NO', 'no_flt3', ifelse(FLT3.PM == 'Yes', 'yes_flt3', 'other'))) %>%
  dplyr::select(relapse, flt3_status)  %>%
  group_by(relapse, flt3_status) %>%
  ggplot(mapping = aes(x = relapse, y = flt3_status)) + geom_point()


  


```

```{r}


# number of samples in each cytogenetic group

umap_target_clin %>%
  group_by(clust, Primary.Cytogenetic.Code) %>%
  summarise(number = length(sample)) %>%
  pivot_wider(names_from = clust, values_from = number)


umap_target_clin %>%
  group_by(clust, Gene.Fusion) %>%
  summarise(number = length(sample)) %>%
  pivot_wider(names_from = clust, values_from = number) %>%
  arrange(Gene.Fusion)

umap_target_clin %>%
  group_by(clust, fav_status) %>%
  summarise(number = length(sample)) %>%
  pivot_wider(names_from = clust, values_from = number) 

umap_target_clin %>%
  group_by(clust, Risk.group) %>%
  summarise(number = length(sample)) %>%
  pivot_wider(names_from = clust, values_from = number) 


```

```{r}

# cytogenetically 'other' ??

umap_target_clin %>%
  filter(Primary.Cytogenetic.Code == "Other") %>%
  group_by(Gene.Fusion) %>%
  summarise(length(Gene.Fusion))


```


```{r}

# clinical differences in cytogenetically normal clusters 1 vs 4 ?

umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  group_by(cluster) %>%
  summarise(mean_age = mean(Age.at.Diagnosis.in.Days/365), 
            sd = sd(Age.at.Diagnosis.in.Days/365)) 

umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  group_by(cluster) %>%
  summarise(mean_wbc = mean(WBC.at.Diagnosis), 
            sd = sd(WBC.at.Diagnosis)) 

umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  arrange(cluster) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  dplyr::select(sample, cluster, FLT3.ITD.positive., FLT3.PM, NPM.mutation, CEBPA.mutation, WT1.mutation, c.Kit.Mutation.Exon.8, c.Kit.Mutation.Exon.17)

umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  group_by(cluster) %>%
  summarise(efs = mean(Event.Free.Survival.Time.in.Days),
            sd = sd(Event.Free.Survival.Time.in.Days),
            sem = sd/sqrt(length(cluster)))

umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  group_by(cluster) %>%
  summarise(os = mean(Overall.Survival.Time.in.Days),
            sd = sd(Overall.Survival.Time.in.Days),
            sem = sd/sqrt(length(cluster)))

umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  arrange(cluster) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  dplyr::select(sample, cluster, First.Event)


#############
#############

# standard and low risk between clusters 2 and 4

umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  group_by(cluster) %>%
  summarise(mean_age = mean(Age.at.Diagnosis.in.Days/365), 
            sd = sd(Age.at.Diagnosis.in.Days/365)) 

umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  group_by(cluster) %>%
  summarise(mean_wbc = mean(WBC.at.Diagnosis), 
            sd = sd(WBC.at.Diagnosis)) 

umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  arrange(cluster) %>%
  dplyr::select(sample, cluster, Risk.group, FLT3.ITD.positive., FLT3.PM, NPM.mutation, CEBPA.mutation, WT1.mutation, c.Kit.Mutation.Exon.8, c.Kit.Mutation.Exon.17)

umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  group_by(cluster) %>%
  summarise(efs = mean(Event.Free.Survival.Time.in.Days),
            sd = sd(Event.Free.Survival.Time.in.Days),
            sem = sd/sqrt(length(cluster)))

umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  group_by(cluster) %>%
  summarise(os = mean(Overall.Survival.Time.in.Days),
            sd = sd(Overall.Survival.Time.in.Days),
            sem = sd/sqrt(length(cluster)))

umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  arrange(cluster) %>%
  dplyr::select(sample, cluster, First.Event, Event.Free.Survival.Time.in.Days)



```


```{r}

# limma comparing normal cytogenetics, clusters 1-4

# collecting all samples for clusters 1-4 normal cyto

# model matrix with samples and cluster designation -- using only 1, 5 and normal cytogenetics
mod_mat <- umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  dplyr::select(sample, cluster)

mod_samp <- pull(mod_mat %>%
  dplyr::select(sample))

# gather omics data from normal cytogenetics / clusters 1,3,5 samples

# expression -- commented out bc did not have to upload again
# expression <- (read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/target_expression_pilot_115.csv"))

exp_limma <- as.data.frame(target_expression %>% filter(sample %in% mod_samp) %>%
  column_to_rownames("sample") %>%
  t())

# run limma with model matrix and expression data

design <- model.matrix(~cluster, mod_mat)
    fit <- lmFit(exp_limma, design)
    fit <- eBayes(fit)
    results <- decideTests(fit, method = "separate", adjust.method = "fdr", p.value = 0.05, lfc = FALSE)
    summary(results)
    topTable(fit)
  tab <- topTable(fit, sort = "none", n = Inf, coef = 1)
  
limma_exp_cn_1.4 <- tab %>% rownames_to_column("gene_id")
write.csv(limma_exp_cn_1.4, "~/Documents/R/pedcancer/target_data/TARGET_100/limma_exp_cn_1.4.csv", row.names = FALSE)

limma_exp_cn_1.4 %>%
  mutate(neglogP = -log(adj.P.Val)) %>%
  ggplot(mapping = aes(x = logFC, y = neglogP)) + geom_point() + xlim(-10, 10)





###################
###################
# limma comparing SR/LR clusters 2 and 4

# sample assignment
cl2 <- pull(umap_target_clin %>% 
              filter(cluster == 2) %>%
              filter(Risk.group == "Standard Risk" |
                       Risk.group == "Low Risk") %>%
              dplyr::select(sample))

cl4 <- pull(umap_target_clin %>% 
              filter(cluster == 4) %>%
              filter(Risk.group == "Standard Risk" |
                       Risk.group == "Low Risk") %>%
              dplyr::select(sample))

# model matrix with samples and cluster designation -- using only 1, 5 and normal cytogenetics
mod_mat <- umap_target_clin %>%
  filter(cluster == 2 |
           cluster == 4) %>%
  filter(Risk.group == "Standard Risk" |
           Risk.group == "Low Risk") %>%
  dplyr::select(sample, cluster)

mod_samp <- pull(mod_mat %>%
  dplyr::select(sample))

# gather omics data from normal cytogenetics / clusters 1,3,5 samples

# expression -- commented out bc did not have to upload again
# need to uncomment for new analysis
# expression <- (read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/target_expression_pilot_115.csv"))

exp_limma <- as.data.frame(target_expression %>% filter(sample %in% mod_samp) %>%
  column_to_rownames("sample") %>%
  t())

# run limma with model matrix and expression data

design <- model.matrix(~cluster, mod_mat)
    fit <- lmFit(exp_limma, design)
    fit <- eBayes(fit)
    results <- decideTests(fit, method = "separate", adjust.method = "fdr", p.value = 0.05, lfc = FALSE)
    summary(results)
    topTable(fit)
  tab <- topTable(fit, sort = "none", n = Inf, coef = 1)
  
limma_exp_sr.lr_2.4 <- tab %>% rownames_to_column("gene_id")
write.csv(limma_exp_sr.lr_2.4, "~/Documents/R/pedcancer/target_data/TARGET_100/limma_exp_sr.lr_2.4.csv", row.names = FALSE)

limma_sig <- limma_exp_sr.lr_2.4 %>%
  filter(adj.P.Val <= 0.05, 
         logFC >= 2) %>%
  arrange(-logFC)

exp_mat <- merge(target_expression, mod_mat, by = "sample") 
exp_mat <- exp_mat[, c(1, 18457, 2:18456)]

mitocarta_oxphos <- pull(mitocarta %>% filter(grepl("OXPHOS", MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))

limma_sig_oxphos <- pull(limma_sig %>% filter(gene_id %in% mitocarta_oxphos) %>% dplyr::select(gene_id))

exp_mat %>%
  pivot_longer(cols = 3: 18457,
               names_to = "gene_id", 
               values_to = "expression") %>%
  filter(gene_id %in% limma_sig_oxphos) %>%
  filter(expression > 0) %>%
  filter(gene_id %in% cV) %>%
  group_by(cluster, gene_id) %>%
  mutate(med_exp = median(expression)) %>%
  filter(med_exp > 10) %>%
  ggplot(mapping = aes(x = fct_reorder(gene_id, -med_exp), y = expression, fill = cluster)) + 
  geom_boxplot() +
  ylab("Gene Expression") +
  xlab("") +
  theme(panel.background = element_rect(fill = "transparent"),
        axis.line = element_line(linetype = "solid", size = 0.5),
        axis.title.x = element_text(family = "Arial", size = 16),
        axis.text.x = element_text(family = "Arial", size = 7, angle = 30, vjust = 0.9, hjust = 1, color = "black"), 
        axis.title.y = element_text(family = "Arial", size = 16),
        axis.text.y = element_text(family = "Arial",, size = 16))
  
```

```{r}
mitocarta <- read.csv("~/Documents/R/ddh/data/mitocarta.csv")

mitocarta_oxphos <- pull(mitocarta %>%
  filter(grepl('OXPHOS', MitoCarta3.0_MitoPathways)) %>%
  dplyr::select(Symbol))

# mark each oxphos component
cI <- pull(mitocarta %>% filter(grepl('\\bComplex I\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cII <- pull(mitocarta %>% filter(grepl('\\bComplex II\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cIII <- pull(mitocarta %>% filter(grepl('\\bComplex III\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cIV <- pull(mitocarta %>% filter(grepl('\\bComplex IV\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cV <- pull(mitocarta %>% filter(grepl('\\bComplex V\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))

oxphos_subunit <- as.data.frame(target_expression %>% 
  column_to_rownames("sample") %>%
  t()) %>%
  rownames_to_column("gene_id") %>%
  filter(gene_id %in% mitocarta_oxphos) %>%
  mutate(oxphos_subunit = ifelse(gene_id %in% cI, 'cI', 
                          ifelse(gene_id %in% cII, 'cII', 
                                 ifelse(gene_id %in% cIII, 'cIII', 
                                        ifelse(gene_id %in% cIV, 'cIV', 
                                               ifelse(gene_id %in% cV, 'cV', 
                                                      'none')))))) %>%
  filter(oxphos_subunit != 'none') %>%
  dplyr::select(gene_id, oxphos_subunit) %>%
  column_to_rownames("gene_id")
```


```{r}
sample_cluster <- umap_target_clin %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  filter(cluster == '1' | cluster == '4') %>%
  dplyr::select(sample, cluster) %>%
  column_to_rownames("sample")

oxphos_pheatmap <- as.data.frame(target_expression %>% 
  filter(sample %in% mod_samp) %>%
  column_to_rownames("sample") %>%
  t()) %>%
  rownames_to_column("gene_id") %>%
  filter(gene_id %in% mitocarta_oxphos) %>%
  column_to_rownames("gene_id") %>%
  pheatmap(color = colorRampPalette(rev(brewer.pal(n = 11, name =
  "RdYlBu")))(100), scale = "row", cellwidth = 10, show_rownames = F, annotation_col = sample_cluster, annotation_row = oxphos_subunit, border_color = "black", treeheight_row = 10, treeheight_col = 25, show_colnames = F, annotation_names_col = F, annotation_names_row = T,)
  

```


```{r}

limma_exp_sr.lr_2.4_oxphos <- read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/limma_exp_sr.lr_2.4_oxphos.csv")

sr.lr_oxphos <- pull(limma_exp_sr.lr_2.4_oxphos %>% dplyr::select(gene_id))


  

```


```{r}

# cpg between clusters 1, 4 cytogentic normal

mod_mat <- umap_target_clin %>%
  filter(cluster == 1 |
           cluster == 4) %>%
  filter(Primary.Cytogenetic.Code == "Normal") %>%
  dplyr::select(sample, cluster)

mod_samp <- pull(mod_mat %>%
  dplyr::select(sample))

# gather omics data from normal cytogenetics / clusters 1,3,5 samples

# cpg df -- commented out bc did not have to upload again
# cpg <- (read.csv("~/Documents/R/pedcancer/target_data/TARGET_100/target_cpg_pilot_115.csv"))

cpg_limma <- as.data.frame(target_cpg %>% filter(sample %in% mod_samp) %>%
  column_to_rownames("sample") %>%
  t())

# run limma with model matrix and cpg data

design <- model.matrix(~cluster, mod_mat)
    fit <- lmFit(cpg_limma, design)
    fit <- eBayes(fit)
    results <- decideTests(fit, method = "separate", adjust.method = "fdr", p.value = 0.05, lfc = FALSE)
    summary(results)
    topTable(fit)
  tab <- topTable(fit, sort = "none", n = Inf, coef = 1)
  
  limma_cpg_cn_1.4 <- tab %>% rownames_to_column("gene_id")
write.csv(limma_cpg_cn_1.4, "~/Documents/R/pedcancer/target_data/TARGET_100/limma_cpg_cn_1.4.csv", row.names = FALSE)

limma_cpg_cn_1.4 %>%
  mutate(neglogP = -log(adj.P.Val)) %>%
  ggplot(mapping = aes(x = logFC, y = neglogP)) + geom_point()

```



```{r}

# oxphos

mitocarta <- read.csv("~/Documents/R/ddh/data/mitocarta.csv")

# mitocarta pathways
mitocarta_groups <- mitocarta %>% group_by(MitoCarta3.0_MitoPathways) %>% summarise(length(MitoCarta3.0_MitoPathways))

### pull mitocarta genes of interest

mitocarta_oxphos <- pull(mitocarta %>%
  filter(grepl('OXPHOS', MitoCarta3.0_MitoPathways)) %>%
  dplyr::select(Symbol))

# mark each oxphos component
cI <- pull(mitocarta %>% filter(grepl('\\bComplex I\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cII <- pull(mitocarta %>% filter(grepl('\\bComplex II\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cIII <- pull(mitocarta %>% filter(grepl('\\bComplex III\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cIV <- pull(mitocarta %>% filter(grepl('\\bComplex IV\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))
cV <- pull(mitocarta %>% filter(grepl('\\bComplex V\\b', MitoCarta3.0_MitoPathways)) %>% dplyr::select(Symbol))

```

```{r}

# find omics with biggest OXPHOS differences in clusters 1,4 cytogenetically normal

exp_cpg_merge <- merge(limma_exp_cn_1.4, limma_cpg_cn_1.4, by = "gene_id") %>%
  dplyr::select(gene_id, logFC.x, adj.P.Val.x,logFC.y, adj.P.Val.y) %>%
    mutate(complex = ifelse(gene_id %in% cI, 'cI', 
                          ifelse(gene_id %in% cII, 'cII', 
                                 ifelse(gene_id %in% cIII, 'cIII', 
                                        ifelse(gene_id %in% cIV, 'cIV', 
                                               ifelse(gene_id %in% cV, 'cV', 
                                                      'none'))))))

exp_cpg_top <- exp_cpg_merge %>%
  filter(adj.P.Val.x <= 0.05 &
           adj.P.Val.y <= 0.05) %>%
  filter(complex != 'none')
  
exp_cpg_top %>%
  ggplot(mapping = aes(x = logFC.x, y = logFC.y, color = complex)) + geom_point() +
  xlab("expression") + 
  ylab("methylation") +
  geom_label_repel(mapping = aes(x = logFC.x, y = logFC.y, label = gene_id), max.overlaps = 30)


# expression
limma_exp_cn_1.4 %>%
  filter(gene_id %in% mitocarta_oxphos) %>%
  filter(adj.P.Val <= 0.05 &
           logFC >= 25) %>%
  ggplot(mapping = aes(x = fct_reorder(gene_id, -logFC), y = logFC)) +
  geom_col() +  theme(axis.text.x = element_text(family = "Arial", size = 12, color = "black", angle = 45, hjust = 0.90, vjust = 0.95))

limma_exp_cn_1.4_oxphos <- pull(limma_exp_cn_1.4 %>%
  filter(gene_id %in% mitocarta_oxphos) %>%
  filter(adj.P.Val <= 0.05) %>%
  arrange(-logFC) %>%
  dplyr::select(gene_id))

write.csv(limma_exp_cn_1.4_oxphos, "~/Documents/R/pedcancer/target_data/TARGET_100/limma_exp_cn_1.4_oxphos.csv")

# cpg -- all significant FDR are hypomethylated at TSS ==  higher expression ?
limma_cpg_cn_1.5 %>%
  filter(gene_id %in% mitocarta_oxphos) %>%
  filter(adj.P.Val <= 0.05) %>%
  arrange(-logFC)

hypometh_oxphos <- pull(limma_cpg_cn_1.4 %>%
  filter(gene_id %in% mitocarta_oxphos) %>%
  filter(adj.P.Val <= 0.05) %>%
  arrange(-logFC) %>%
    dplyr::select(gene_id))


# find expression of significantly hypomethylated genes

limma_exp_cn_1.4 %>%
  filter(gene_id %in% hypometh_oxphos) %>%
  arrange(logFC) %>%
  ggplot(mapping = aes(x = fct_reorder(gene_id, -logFC), y = logFC)) + geom_col() +
  theme(axis.text.x = element_text(family = "Arial", size = 12, color = "black", angle = 45, hjust = 0.90, vjust = 0.95))
  

# examine relationship between tss methylation and expression of hypomethylated oxphos genes

hypometh_oxphos_cpg <- limma_cpg_cn_1.5 %>%
  filter(gene_id %in% hypometh_oxphos) %>%
  dplyr::select(gene_id, logFC)

hypometh_oxpho_exp <- limma_exp_cn_1.5 %>%
  filter(gene_id %in% hypometh_oxphos) %>%
  dplyr::select(gene_id, logFC)

hypometh_oxphos_cor <- merge(hypometh_oxpho_exp, hypometh_oxphos_cpg, by = "gene_id")

hypometh_oxphos_cor %>% ggplot(mapping = aes(x = logFC.x, y = logFC.y)) + geom_point() +
  xlab("expression") + 
  ylab("methylation") +
  geom_text(mapping = aes(x = logFC.x, y = logFC.y, label = gene_id))

# determine subsets of oxphos complexes most effected


  
```



```{r}

target_cpg %>% 
  pivot_longer(cols = 2:8162, names_to = "gene_id", values_to = "methylation") %>% 
  filter(gene_id %in% mitocarta_oxphos) %>%
  mutate(cluster = ifelse(sample %in% cl2, '2',
                          ifelse(sample %in% cl4, '4', 'other'))) %>%
  filter(cluster == '2' | cluster == '4') %>%
  filter(gene_id %in% cII) %>%
  mutate(med_tss = median(methylation)) %>%
  group_by(cluster) %>%
  ggplot(mapping = aes(x = fct_reorder(gene_id, -med_tss), y = methylation, fill = cluster)) + 
  geom_boxplot() +
  ylab("TSS methylation") +
  xlab("") +
  theme(panel.background = element_rect(fill = "transparent"),
        axis.line = element_line(linetype = "solid", size = 0.5),
        axis.title.x = element_text(family = "Arial", size = 16),
        axis.text.x = element_text(family = "Arial", size = 8, angle = 30, vjust = 0.9), 
        axis.title.y = element_text(family = "Arial", size = 16),
        axis.text.y = element_text(family = "Arial",, size = 16))


  mutate(med_exp = median(expression)) %>%
  filter(med_exp > 10) %>%
  ggplot(mapping = aes(x = fct_reorder(gene_id, -med_exp), y = expression, fill = cluster)) + 

```

```{r}

# goal is to download all target-aml files automatically, then filter based on desired analysis

url <- "https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-FHCRC/"

txts <- XML::getHTMLLinks(
  url, 
  xpQuery = "//a/@href['.txt' = substring(.)]"
)

['.txt' = substring(., string-length(.) - 3)]"

dir.create("mytxts")
wd <- getwd()
setwd("mytxts")
file.create(txts)
lapply(paste0(url, txts), function(x) download.file(x, basename(x)))

a <- readLines(url)

loc.txt <- grep(".txt", a)

convfn <- function(line, marker, page){
  i <- unlist(gregexpr(pattern ='href="', line)) + 6
  i2<- unlist(gregexpr(pattern =, "TARGET",  line)) + 3
  #target file
  .destfile <- substring(line, i[1], i2[1])
  #target url
  .url      <- paste("https://target-data.nci.nih.gov/Public/AML/mRNA-seq/L3/expression/BCCA/TARGET-FHCRC/", .destfile, sep = "/")
  #print targets
  cat(.url, '\n', .destfile, '\n')
  #the workhorse function
  download.file(url=.url, destfile=.destfile)
}

sapply(a[loc.txt], 
       FUN = convfn, 
       marker = '.txt"', #this is key part, locates text file name
       page = url)

```


